---
title: 'The spatially resolved transcriptome signatures of glomeruli in chronic kidney disease'
author: Geremy Clair
date: June 1, 2021
output:
  html_document:
    toc: true
    toc_depth: 2
    number_sections: true
    theme : cerulean
    code_folding: hide
---

# Directory setting
We will set the working directory
```{r set_directory_and_load_file}
knitr::opts_chunk$set(fig.width=10, fig.height=6, echo = TRUE, results = TRUE, warning = FALSE, message=FALSE) 
options(repos=structure(c(CRAN="https://cloud.r-project.org/")))
setwd("~/Documents/GitHub/Effect_of_glomerular_disease_on_the_podocyte_cell_cycle/")
library(RomicsProcessor)
library(proteinminon)
library(DT)
```

#data loading and Uniprot Mapping
```{r }
data<-read.csv(file="01_source_files/Norm_TargetCountMatrix.csv")
metadata<-read.csv(file="01_source_files/metadata_ROI.csv")
data<-data[9:nrow(data),]
data[,-1]<-sapply(data[,-1],as.numeric)

uniprot<-read.csv("01_source_files/uniprot_Mapping_table.csv")

if(!file.exists("./03_output_files/Uniprot_Homo_sapiens_proteome_UP000005640_2020_06_28.fasta")){
    download_UniProtFasta(proteomeID = "UP000005640",reviewed = T,export = TRUE, file="./03_output_files/Uniprot_Homo_sapiens_proteome_UP000005640_2020_06_28.fasta")
}

if(file.exists("./03_output_files//UniprotTable_Homo_sapiens_proteome_UP000005640_2020_06_28.csv")){
  UniProtTable<-read.csv("./03_output_files/UniprotTable_Homo_sapiens_proteome_UP000005640_2020_06_28.csv")
  }else{
  download_UniProtTable(proteomeID = "UP000005640",reviewed = T)
  write.csv(UniProtTable,("./03_output_files/UniprotTable_Homo_sapiens_proteome_UP000005640_2020_06_28.csv"),row.names=FALSE)
  }

if(file.exists("./03_output_files/UniProtTable_GO.csv")){
  UniProtTable_GO<-read.csv(file="./03_output_files/UniProtTable_GO.csv")
}else{
generate_UniProtTable_GO()
write.csv(UniProtTable_GO,file="./03_output_files/UniProtTable_GO.csv",row.names=FALSE)
}

if(file.exists("./03_output_files/UniProtTable_KEGG.csv")){
  UniProtTable_KEGG<-read.csv(file="./03_output_files/UniProtTable_KEGG.csv")
}else{
generate_UniProtTable_KEGG()
write.csv(UniProtTable_KEGG,file="./03_output_files/UniProtTable_KEGG.csv",row.names=FALSE)
}

uniprot<-cbind(uniprot,Reviewed=uniprot$Entry %in% UniProtTable$Uniprot_Accession, ncharGO=nchar(uniprot$Gene.ontology.IDs))

uniprot<-uniprot[with(uniprot, order(uniprot[1], uniprot$Reviewed, uniprot$ncharGO,decreasing = T)),]
uniprot<-uniprot[!duplicated(uniprot[1]),]
uniprot<-uniprot[order(uniprot[1]),]

data$slide.name[data$slide.name %in% uniprot$yourlist.M202106285BF3C56A578D7D6DFD1FC81EE5DA77300A323DH]<-paste0(data$slide.name[data$slide.name %in% uniprot$yourlist.M202106285BF3C56A578D7D6DFD1FC81EE5DA77300A323DH],"@",uniprot[match(data$slide.name[data$slide.name %in% uniprot$yourlist.M202106285BF3C56A578D7D6DFD1FC81EE5DA77300A323DH],uniprot$yourlist.M202106285BF3C56A578D7D6DFD1FC81EE5DA77300A323DH),3])

data$slide.name[duplicated(data[1])]<-paste0(data$slide.name[duplicated(data[1])],2)

IDs<-data$slide.name

data$slide.name<-gsub(".*\\@","",data$slide.name)
write.csv(metadata,file="./03_output_files/metadata_ROI.csv")
```

#RomicsObject Generation

The table containing the data and metadata was loaded, an identifier column was added, and the data was loaded in an romics_object
```{r load}
romics_rna<-romicsCreateObject(data,metadata,main_factor = "disease_gen_structure")
romics_rna<-romicsSampleNameFromFactor(romics_rna,factor = "sample_name")
```

#Check data distribution
After checking the data distribution it seems that the data require a log transformation both a log2 and log10 were tested
```{r data_grouping}
print("First we will test a log2 transformation")
distribHistogramGlobal(log2transform(romics_rna),bin = 0.5)
print("Then a log10")
distribHistogramGlobal(log10transform(romics_rna),bin = 0.1)
print("Both solution appeared equivalent the data was log2 transformed")
romics_rna<-log2transform(romics_rna)
print("The boxplot distribution per sample was plotted")
distribBoxplot(romics_rna)
```

#Subsetting
Different Subsets were created from the global spatial rna dataset
First only the glomeruli regions will be kept in the Subset named romics_glomeruli
then this Subset will be further Subseted in Subset named Subset01->Subset24

Here is the list of Subsets to be generated:
1.	Yng AS male only
2.	Yng AS female only
3.	Yng AS male + Yng normal 
4.	Yng AS female + Yng normal 
5.	Yng AS male + Yng AS female 
6.	Yng AS male + Yng AS female + Yng normal
7.	Adult AS female only
8.	Adult AS female + normal 
9.	Adult FSGS male only
10.	Adult FSGS female only
11.	Adult FSGS male + normal
12.	Adult FSGS female + normal
13.	Adult FSGS male + FSGS female
14.	Adult FSGS male + FSGS female + normal
15.	Adult MN male only
16.	Adult MN female only
17.	Adult MN male + normal 
18.	Adult MN female +normal 
19.	Adult MN male + MN female
20.	Adult MN male + MN female + normal
21.	Adult AS + FSGS + MN 
22.	Adult AS + FSGS + MN + control 
23. Adult AS + FSGS + control
24.	Adult  + FSGS 

```{r Subsetting}
romics_glomeruli<-romicsSubset(romics_rna, subset_vector ="Glomeruli" ,type = "keep", factor = "structure",by = "level")
Subset01<-romicsSubset(romics_glomeruli, subset_vector ="Alport(M)" ,type = "keep", factor = "disease_gen",by = "level")
Subset02<-romicsSubset(romics_glomeruli, subset_vector ="Alport(F)(Young)" ,type = "keep", factor = "disease_gen",by = "level")
Subset03<-romicsSubset(romics_glomeruli, subset_vector =c("Alport(M)","Normal (Youth)") ,type = "keep", factor = "disease_gen",by = "level")
Subset04<-romicsSubset(romics_glomeruli, subset_vector =c("Alport(F)(Young)","Normal (Youth)") ,type = "keep", factor = "disease_gen",by = "level")
Subset05<-romicsSubset(romics_glomeruli, subset_vector =c("Alport(M)","Alport(F)(Young)") ,type = "keep", factor = "disease_gen",by = "level")
Subset06<-romicsSubset(romics_glomeruli, subset_vector =c("Alport(M)","Alport(F)(Young)","Normal (Youth)") ,type = "keep", factor = "disease_gen",by = "level")
Subset07<-romicsSubset(romics_glomeruli, subset_vector =c("Alport(F)(Adult)") ,type = "keep", factor = "disease_gen",by = "level")
Subset08<-romicsSubset(romics_glomeruli, subset_vector =c("Alport(F)(Adult)","Normal (Adult)") ,type = "keep", factor = "disease_gen",by = "level")
Subset09<-romicsSubset(romics_glomeruli, subset_vector =c("FSGS (M)") ,type = "keep", factor = "disease_gen",by = "level")
Subset10<-romicsSubset(romics_glomeruli, subset_vector =c("FSGS(F)") ,type = "keep", factor = "disease_gen",by = "level")
Subset11<-romicsSubset(romics_glomeruli, subset_vector =c("FSGS (M)","Normal (Adult)") ,type = "keep", factor = "disease_gen",by = "level")
Subset12<-romicsSubset(romics_glomeruli, subset_vector =c("FSGS(F)","Normal (Adult)") ,type = "keep", factor = "disease_gen",by = "level")
Subset13<-romicsSubset(romics_glomeruli, subset_vector =c("FSGS(F)","FSGS (M)") ,type = "keep", factor = "disease_gen",by = "level")
Subset14<-romicsSubset(romics_glomeruli, subset_vector =c("FSGS(F)","FSGS (M)","Normal (Adult)") ,type = "keep", factor = "disease_gen",by = "level")
Subset15<-romicsSubset(romics_glomeruli, subset_vector =c("MN(M)") ,type = "keep", factor = "disease_gen",by = "level")
Subset16<-romicsSubset(romics_glomeruli, subset_vector =c("MN(F)") ,type = "keep", factor = "disease_gen",by = "level")
Subset17<-romicsSubset(romics_glomeruli, subset_vector =c("MN(M)","Normal (Adult)") ,type = "keep", factor = "disease_gen",by = "level")
Subset18<-romicsSubset(romics_glomeruli, subset_vector =c("MN(F)","Normal (Adult)") ,type = "keep", factor = "disease_gen",by = "level")
Subset19<-romicsSubset(romics_glomeruli, subset_vector =c("MN(F)","MN(M)") ,type = "keep", factor = "disease_gen",by = "level")
Subset20<-romicsSubset(romics_glomeruli, subset_vector =c("MN(F)","MN(M)","Normal (Adult)") ,type = "keep", factor = "disease_gen",by = "level")
Subset21<-romicsSubset(romics_glomeruli, subset_vector =c("Alport(F)(Adult)","MN(F)","MN(M)","FSGS (M)", "FSGS(F)") ,type = "keep", factor = "disease_gen",by = "level")
Subset22<-romicsSubset(romics_glomeruli, subset_vector =c("Alport(F)(Adult)","MN(M)","MN(F)","FSGS (M)","FSGS(F)","Normal (Adult)") ,type = "keep", factor = "disease_gen",by = "level")
Subset23<-romicsSubset(romics_glomeruli, subset_vector =c("Alport(F)(Adult)", "FSGS (M)","FSGS(F)","Normal (Adult)") ,type = "keep", factor = "disease_gen",by = "level")
Subset24<-romicsSubset(romics_glomeruli, subset_vector =c("Alport(F)(Adult)","FSGS (M)","FSGS(F)") ,type = "keep", factor = "disease_gen",by = "level")
```

# hclust and Principal component analysis

## global data

```{r Global_Hclust}
romicsHclust(romics_rna)
```

and then by PCA
```{r Global_PCA}
#Slide_19.B.01500_Tub_ROI_9 was found to be an outlier using PmartR outlier detection algorithm and was therefore removed from the analysis
romics_rna<-romicsSubset(romics_rna,subset_vector = "Slide_19.B.01500_Tub_ROI_9",type = "drop",by = "colnames")
indPCAplot(romics_rna,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(romics_rna,Xcomp = 1,Ycomp = 2,Zcomp = 3)
indPCA3D(romicsChangeFactor(romics_rna,main_factor = "structure"),Xcomp = 1,Ycomp = 2,Zcomp = 3)
indPCA3D(romicsChangeFactor(romics_rna,main_factor = "disease_gen"),Xcomp = 1,Ycomp = 2,Zcomp = 3)
print("The PCA results were recorded")
PCA_rna<-romicsPCA(romics_rna)
write.csv(PCA_rna$var$contrib,file="03_output_files/PCA_golbal_contrib.csv")
```

## Expression change in glomeruli vs tubule

ANOVA was performed, Heatmap was plotted, expected markers of Glomeruli and Tubules were checked for their localization.
```{r}
romics_rna_gt<-romicsChangeFactor(romics_rna,main_factor = "structure")
romics_rna_gt<-romicsANOVA(romics_rna_gt)

pFrequencyPlot(romics_rna_gt)
  
romicsHeatmap(romics_rna_gt,sample_hclust = T,ANOVA_filter = "padj" ,p = 0.05,variable_hclust_number = 2)
romics_rna_gt<- romicsVariableHclust(romics_rna_gt,clusters = 2,ANOVA_filter = "padj" ,p = 0.05,plot=F)

library("ggpubr")

#PODXL
singleVariablePlot(romics_rna_gt,variable ="PODXL_HUMAN")+ggtitle(label = "PODXL")+
  geom_bracket(xmin = "Glomeruli", xmax = "Tubules", y.position = 12, label =formatC(romics_rna_gt$statistics[grepl(paste0(".*","PODXL_HUMAN",".*"),rownames(romics_rna_gt$statistics)),colnames(romics_rna_gt$statistics)=="ANOVA_padj"], format = "e"))

#SYNPO
singleVariablePlot(romics_rna_gt,variable ="SYNPO_HUMAN")+ggtitle(label = "SYNPO")+
  geom_bracket(xmin = "Glomeruli", xmax = "Tubules", y.position = 8, label =formatC(romics_rna_gt$statistics[grepl(paste0(".*","SYNPO_HUMAN",".*"),rownames(romics_rna_gt$statistics)),colnames(romics_rna_gt$statistics)=="ANOVA_padj"], format = "e"))

#EHD3
singleVariablePlot(romics_rna_gt,variable ="EHD3")+ggtitle(label = "EHD3")+
  geom_bracket(xmin = "Glomeruli", xmax = "Tubules", y.position = 8, label =formatC(romics_rna_gt$statistics[grepl(paste0(".*","EHD3",".*"),rownames(romics_rna_gt$statistics)),colnames(romics_rna_gt$statistics)=="ANOVA_padj"], format = "e"))


print(paste0("EDH3 ANOVA adjusted pvalue is ", romics_rna_gt$statistics[rownames(romics_rna_gt$statistics)=="EHD3",2]))
singleVariablePlot(romics_rna_gt,variable = "EHD3",type = "jb",title = "EDH3")

print(paste0("APOE_HUMAN ANOVA adjusted pvalue is ", romics_rna_gt$statistics[rownames(romics_rna_gt$statistics)=="APOE_HUMAN",2]))
singleVariablePlot(romics_rna_gt,variable = "APOE_HUMAN",type = "jb",title = "APOE")

print(paste0("SLC17A3 ANOVA adjusted pvalue is ", romics_rna_gt$statistics[rownames(romics_rna_gt$statistics)=="NPT4_HUMAN",2]))
singleVariablePlot(romics_rna_gt,variable = "NPT4_HUMAN",type = "jb",title = "SLC17A3")

print(paste0("DCXR ANOVA adjusted pvalue is ", romics_rna_gt$statistics[rownames(romics_rna_gt$statistics)=="DCXR_HUMAN",2]))
singleVariablePlot(romics_rna_gt,variable = "DCXR_HUMAN",type = "jb",title = "DCXR")

romics_rna_gt<-romicsZscores(romics_rna_gt)
enrichment_rna_gt<-romicsEnrichement(romics_rna_gt,cluster_filter = "hclust_clusters")
enrichment_rna_gt<-enrichment_rna_gt[enrichment_rna_gt$pval<0.05&enrichment_rna_gt$fold_change>1,]
```

## Glom_only

The sample grouping was checked by h-clustering
```{r Glom_Hclust}
romicsHclust(romics_glomeruli)
```

and then by PCA
```{r Glom_PCA}
indPCAplot(romics_glomeruli,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(romics_glomeruli,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_glomeruli<-romicsPCA(romics_rna)
write.csv(PCA_glomeruli$var$contrib,file="03_output_files/PCA_glomeruli_contrib.csv")

universe <- rownames(PCA_glomeruli$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_glomeruli$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_glomeruli$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_glomeruli_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_glomeruli_enr<-PCA_glomeruli_enr[PCA_glomeruli_enr$pval<0.05 & PCA_glomeruli_enr$fold_change>1,]

datatable(PCA_glomeruli_enr)

write.table(PCA_glomeruli_enr, file="./03_output_files/PCA_glomeruli_enr.txt",sep = "\t",row.names = F)
```


For all glomeruli we performed a trajectory analysis using the package SLICER
```{r glomeruli_Trajectory}
library("SLICER")
library("lle")

table_Glomeruli<-t(romics_glomeruli$data)
genes <- select_genes(table_Glomeruli)
k = select_k(table_Glomeruli[,genes], kmin=5)
traj_lle = lle(table_Glomeruli[,genes], m=2, k)$Y

print("the points of the trajectory are plotted")

traj_graph = conn_knn_graph(traj_lle,5)
ends = find_extreme_cells(traj_graph, traj_lle)
start = 1

print("The trajectory connections were plotted")

distances = graph_process_distance(traj_graph,embedding = traj_lle,start = 1)

traj_lle_for_graph = as.data.frame(traj_lle)
colnames(traj_lle_for_graph) = c("Manifold_dim_1","Manifold_dim_2")
traj_lle_for_graph$Manifold_dim_1 = as.numeric(traj_lle_for_graph$Manifold_dim_1)
traj_lle_for_graph$Manifold_dim_2 = as.numeric(traj_lle_for_graph$Manifold_dim_2)
traj_lle_for_graph<- cbind(ROI= rownames(table_Glomeruli), disease_gen= t(romics_glomeruli$metadata[5,]), traj_lle_for_graph)#, branches =  assign_branches(traj_graph,start = 1))

print("The plots is colored by condition")

ggplot(traj_lle_for_graph,aes(x=Manifold_dim_1,Manifold_dim_2,color=disease_gen))+ geom_point(size=1,)+ geom_text(label=traj_lle_for_graph$ROI)+theme_ROP()

DT::datatable(traj_lle_for_graph)
```


## Subset01

The sample grouping was checked by h-clustering
```{r Subset01_Hclust}
romicsHclust(Subset01)
```

and then by PCA
```{r Subset01_PCA}
indPCAplot(Subset01,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset01,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset01<-romicsPCA(Subset01)
write.csv(PCA_Subset01$var$contrib,file="03_output_files/PCA_Subset01_contrib.csv")

universe <- rownames(PCA_Subset01$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset01$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset01$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset01_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset01_enr<-PCA_Subset01_enr[PCA_Subset01_enr$pval<0.05 & PCA_Subset01_enr$fold_change>1,]

datatable(PCA_Subset01_enr)

write.table(PCA_Subset01_enr, file="./03_output_files/PCA_Subset01_enr.txt",sep = "\t",row.names = F)

```

## Subset02

The sample grouping was checked by h-clustering
```{r Subset02_Hclust}
romicsHclust(Subset02)
```

and then by PCA
```{r Subset02_PCA}
indPCAplot(Subset02,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset02,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset02<-romicsPCA(Subset02)
write.csv(PCA_Subset02$var$contrib,file="03_output_files/PCA_Subset02_contrib.csv")

universe <- rownames(PCA_Subset02$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset02$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset02$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset02_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset02_enr<-PCA_Subset02_enr[PCA_Subset02_enr$pval<0.05 & PCA_Subset02_enr$fold_change>1,]

datatable(PCA_Subset02_enr)

write.table(PCA_Subset02_enr, file="./03_output_files/PCA_Subset02_enr.txt",sep = "\t",row.names = F)
```

## Subset03

The sample grouping was checked by h-clustering
```{r Subset03_Hclust}
romicsHclust(Subset03)
```

and then by PCA
```{r Subset03_PCA}
indPCAplot(Subset03,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset03,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset03<-romicsPCA(Subset03)
write.csv(PCA_Subset03$var$contrib,file="03_output_files/PCA_Subset03_contrib.csv")

universe <- rownames(PCA_Subset03$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset03$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset03$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset03_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset03_enr<-PCA_Subset03_enr[PCA_Subset03_enr$pval<0.05 & PCA_Subset03_enr$fold_change>1,]

datatable(PCA_Subset03_enr)

write.table(PCA_Subset03_enr, file="./03_output_files/PCA_Subset03_enr.txt",sep = "\t",row.names = F)
```

## Subset04

The sample grouping was checked by h-clustering
```{r Subset04_Hclust}
romicsHclust(Subset04)
```

and then by PCA
```{r Subset04_PCA}
indPCAplot(Subset04,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset04,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset04<-romicsPCA(Subset04)
write.csv(PCA_Subset04$var$contrib,file="03_output_files/PCA_Subset04_contrib.csv")

universe <- rownames(PCA_Subset04$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset04$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset04$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset04_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset04_enr<-PCA_Subset04_enr[PCA_Subset04_enr$pval<0.05 & PCA_Subset04_enr$fold_change>1,]

datatable(PCA_Subset04_enr)

write.table(PCA_Subset04_enr, file="./03_output_files/PCA_Subset04_enr.txt",sep = "\t",row.names = F)
```

## Subset05

The sample grouping was checked by h-clustering
```{r Subset05_Hclust}
romicsHclust(Subset05)
```

and then by PCA
```{r Subset05_PCA}
indPCAplot(Subset05,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset05,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset05<-romicsPCA(Subset05)
write.csv(PCA_Subset05$var$contrib,file="03_output_files/PCA_Subset05_contrib.csv")

universe <- rownames(PCA_Subset05$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset05$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset05$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset05_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset05_enr<-PCA_Subset05_enr[PCA_Subset05_enr$pval<0.05 & PCA_Subset05_enr$fold_change>1,]

datatable(PCA_Subset05_enr)

write.table(PCA_Subset05_enr, file="./03_output_files/PCA_Subset05_enr.txt",sep = "\t",row.names = F)
```
## Subset06

The sample grouping was checked by h-clustering
```{r Subset06_Hclust}
romicsHclust(Subset06)
```

and then by PCA
```{r Subset06_PCA}
indPCAplot(Subset06,label = T,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset06,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset06<-romicsPCA(Subset06)
write.csv(PCA_Subset06$var$contrib,file="03_output_files/PCA_Subset06_contrib.csv")

universe <- rownames(PCA_Subset06$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset06$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset06$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset06_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset06_enr<-PCA_Subset06_enr[PCA_Subset06_enr$pval<0.05 & PCA_Subset06_enr$fold_change>1,]

datatable(PCA_Subset06_enr)

write.table(PCA_Subset06_enr, file="./03_output_files/PCA_Subset06_enr.txt",sep = "\t",row.names = F)
```

For this group of Samples the sample clustering was evaluated
```{r Subset06_sampleClust}
Subset06_samples<-Subset06$data
Subset06_samples<-t(Subset06_samples)

print("First the distance between the samples will be evaluated")
distance <- get_dist(Subset06_samples)
fviz_dist(distance, gradient = list(low = "#00AFBB", mid = "white", high = "#FC4E07"))

print("K-mean will be use to establish the clusters for a number of clusters of 2,3,4,5,6, and 7")
k2Subset06<-kmeans(Subset06_samples,centers=2,nstart=25)
fviz_cluster(k2Subset06, data = Subset06_samples)

k3Subset06<-kmeans(Subset06_samples,centers=3,nstart=25)
fviz_cluster(k3Subset06, data = Subset06_samples)

k4Subset06<-kmeans(Subset06_samples,centers=4,nstart=25)
fviz_cluster(k4Subset06, data = Subset06_samples)

k5Subset06<-kmeans(Subset06_samples,centers=5,nstart=25)
fviz_cluster(k5Subset06, data = Subset06_samples)

k6Subset06<-kmeans(Subset06_samples,centers=6,nstart=25)
fviz_cluster(k6Subset06, data = Subset06_samples)

k7Subset06<-kmeans(Subset06_samples,centers=7,nstart=25)
fviz_cluster(k7Subset06, data = Subset06_samples)

k8Subset06<-kmeans(Subset06_samples,centers=8,nstart=25)
fviz_cluster(k8Subset06, data = Subset06_samples)

k9Subset06<-kmeans(Subset06_samples,centers=9,nstart=25)
fviz_cluster(k9Subset06, data = Subset06_samples)

groups06<- t(Subset06$metadata[5,])
groups06<- cbind(group=groups06,group_number=as.integer(as.factor(groups06[1:length(groups06)])))
groups06<- data.frame(cbind(groups06,rownames(Subset06_samples)))

print("The best number of clusters will be evaluated by the Elbow method")

fviz_nbclust(Subset06_samples, kmeans, method = "wss")

print("Using the Silhouette function")

fviz_nbclust(Subset06_samples, kmeans, method = "silhouette")

print("and using the Gap Statistics")

fviz_nbclust(Subset06_samples, kmeans, method = "gap_stat")

print("Finally for the 3 clusters corresponding to the three biological sample types intracluster and extracluster distances were calculated using the package clv")

library("clv")
clust_dist<-cls.scatt.data(Subset06_samples,k3Subset06$cluster)
group_dist<-cls.scatt.data(Subset06_samples,groups06$group_number)


print("Average intracluster (3 clusters) distances are :")
clust_dist$intracls.average

print("Average extracluster_distances (3 clusters) are :")
clust_dist$intercls.average

print("Average intragroups (AS(F)=c1, AS(M)=c2, Ctrl=c3)) distances are :")
group_dist$intracls.average

print("Average intergroups (AS(F)=c1, AS(M)=c2, Ctrl=c3)) distances are :")
group_dist$intercls.average

```
## Subset07

The sample grouping was checked by h-clustering
```{r Subset07_Hclust}
romicsHclust(Subset07)
```

and then by PCA
```{r Subset07_PCA}
indPCAplot(Subset07,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset07,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset07<-romicsPCA(Subset07)
write.csv(PCA_Subset07$var$contrib,file="03_output_files/PCA_Subset07_contrib.csv")

universe <- rownames(PCA_Subset07$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset07$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset07$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset07_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset07_enr<-PCA_Subset07_enr[PCA_Subset07_enr$pval<0.05 & PCA_Subset07_enr$fold_change>1,]

datatable(PCA_Subset07_enr)

write.table(PCA_Subset07_enr, file="./03_output_files/PCA_Subset07_enr.txt",sep = "\t",row.names = F)
```
## Subset08

The sample grouping was checked by h-clustering
```{r Subset08_Hclust}
romicsHclust(Subset08)
```

and then by PCA
```{r Subset08_PCA}
indPCAplot(Subset08,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset08,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset08<-romicsPCA(Subset08)
write.csv(PCA_Subset08$var$contrib,file="03_output_files/PCA_Subset08_contrib.csv")

universe <- rownames(PCA_Subset08$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset08$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset08$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset08_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset08_enr<-PCA_Subset08_enr[PCA_Subset08_enr$pval<0.05 & PCA_Subset08_enr$fold_change>1,]

datatable(PCA_Subset08_enr)

write.table(PCA_Subset08_enr, file="./03_output_files/PCA_Subset08_enr.txt",sep = "\t",row.names = F)
```

## Subset09

The sample grouping was checked by h-clustering
```{r Subset09_Hclust}
romicsHclust(Subset09)
```

and then by PCA
```{r Subset09_PCA}
indPCAplot(Subset09,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
#indPCA3D(Subset09,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset09<-romicsPCA(Subset09)
write.csv(PCA_Subset09$var$contrib,file="03_output_files/PCA_Subset09_contrib.csv")

universe <- rownames(PCA_Subset09$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset09$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset09$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset09_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset09_enr<-PCA_Subset09_enr[PCA_Subset09_enr$pval<0.05 & PCA_Subset09_enr$fold_change>1,]

datatable(PCA_Subset09_enr)

write.table(PCA_Subset09_enr, file="./03_output_files/PCA_Subset09_enr.txt",sep = "\t",row.names = F)
```

## Subset10

The sample grouping was checked by h-clustering
```{r Subset10_Hclust}
romicsHclust(Subset10)
```

and then by PCA
```{r Subset10_PCA}
indPCAplot(Subset10,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset10,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset10<-romicsPCA(Subset10)
write.csv(PCA_Subset10$var$contrib,file="03_output_files/PCA_Subset10_contrib.csv")

universe <- rownames(PCA_Subset10$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset10$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset10$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset10_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset10_enr<-PCA_Subset10_enr[PCA_Subset10_enr$pval<0.05 & PCA_Subset10_enr$fold_change>1,]

datatable(PCA_Subset10_enr)

write.table(PCA_Subset10_enr, file="./03_output_files/PCA_Subset10_enr.txt",sep = "\t",row.names = F)
```

## Subset11

The sample grouping was checked by h-clustering
```{r Subset11_Hclust}
romicsHclust(Subset11)
```

and then by PCA
```{r Subset11_PCA}
indPCAplot(Subset11,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset11,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset11<-romicsPCA(Subset11)
write.csv(PCA_Subset11$var$contrib,file="03_output_files/PCA_Subset11_contrib.csv")

universe <- rownames(PCA_Subset11$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset11$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset11$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset11_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset11_enr<-PCA_Subset11_enr[PCA_Subset11_enr$pval<0.05 & PCA_Subset11_enr$fold_change>1,]

datatable(PCA_Subset11_enr)

write.table(PCA_Subset11_enr, file="./03_output_files/PCA_Subset11_enr.txt",sep = "\t",row.names = F)
```

## Subset12

The sample grouping was checked by h-clustering
```{r Subset12_Hclust}
romicsHclust(Subset12)
```

and then by PCA
```{r Subset12_PCA}
indPCAplot(Subset12,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset12,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset12<-romicsPCA(Subset12)
write.csv(PCA_Subset12$var$contrib,file="03_output_files/PCA_Subset12_contrib.csv")

universe <- rownames(PCA_Subset12$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset12$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset12$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset12_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset12_enr<-PCA_Subset12_enr[PCA_Subset12_enr$pval<0.05 & PCA_Subset12_enr$fold_change>1,]

datatable(PCA_Subset12_enr)

write.table(PCA_Subset12_enr, file="./03_output_files/PCA_Subset12_enr.txt",sep = "\t",row.names = F)
```

## Subset13

The sample grouping was checked by h-clustering
```{r Subset13_Hclust}
romicsHclust(Subset13)
```

and then by PCA
```{r Subset13_PCA}
indPCAplot(Subset13,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset13,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset13<-romicsPCA(Subset13)
write.csv(PCA_Subset13$var$contrib,file="03_output_files/PCA_Subset13_contrib.csv")

universe <- rownames(PCA_Subset13$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset13$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset13$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset13_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset13_enr<-PCA_Subset13_enr[PCA_Subset13_enr$pval<0.05 & PCA_Subset13_enr$fold_change>1,]

datatable(PCA_Subset13_enr)

write.table(PCA_Subset13_enr, file="./03_output_files/PCA_Subset13_enr.txt",sep = "\t",row.names = F)
```

## Subset14

The sample grouping was checked by h-clustering
```{r Subset14_Hclust}
romicsHclust(Subset14)
```

and then by PCA
```{r Subset14_PCA}
indPCAplot(Subset14,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset14,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset14<-romicsPCA(Subset14)
write.csv(PCA_Subset14$var$contrib,file="03_output_files/PCA_Subset14_contrib.csv")

universe <- rownames(PCA_Subset14$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset14$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset14$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset14_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset14_enr<-PCA_Subset14_enr[PCA_Subset14_enr$pval<0.05 & PCA_Subset14_enr$fold_change>1,]

datatable(PCA_Subset14_enr)

write.table(PCA_Subset14_enr, file="./03_output_files/PCA_Subset14_enr.txt",sep = "\t",row.names = F)
```

## Subset15

The sample grouping was checked by h-clustering
```{r Subset15_Hclust}
romicsHclust(Subset15)
```

and then by PCA
```{r Subset15_PCA}
indPCAplot(Subset15,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset15,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset15<-romicsPCA(Subset15)
write.csv(PCA_Subset15$var$contrib,file="03_output_files/PCA_Subset15_contrib.csv")

universe <- rownames(PCA_Subset15$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset15$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset15$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset15_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset15_enr<-PCA_Subset15_enr[PCA_Subset15_enr$pval<0.05 & PCA_Subset15_enr$fold_change>1,]

datatable(PCA_Subset15_enr)

write.table(PCA_Subset15_enr, file="./03_output_files/PCA_Subset15_enr.txt",sep = "\t",row.names = F)
```


## Subset16

The sample grouping was checked by h-clustering
```{r Subset16_Hclust}
romicsHclust(Subset16)
```

and then by PCA
```{r Subset16_PCA}
indPCAplot(Subset16,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset16,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset16<-romicsPCA(Subset16)
write.csv(PCA_Subset16$var$contrib,file="03_output_files/PCA_Subset16_contrib.csv")

universe <- rownames(PCA_Subset16$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset16$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset16$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset16_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset16_enr<-PCA_Subset16_enr[PCA_Subset16_enr$pval<0.05 & PCA_Subset16_enr$fold_change>1,]

datatable(PCA_Subset16_enr)

write.table(PCA_Subset16_enr, file="./03_output_files/PCA_Subset16_enr.txt",sep = "\t",row.names = F)
```

## Subset17

The sample grouping was checked by h-clustering
```{r Subset17_Hclust}
romicsHclust(Subset17)
```

and then by PCA
```{r Subset17_PCA}
indPCAplot(Subset17,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset17,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset17<-romicsPCA(Subset17)
write.csv(PCA_Subset17$var$contrib,file="03_output_files/PCA_Subset17_contrib.csv")

universe <- rownames(PCA_Subset17$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset17$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset17$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset17_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset17_enr<-PCA_Subset17_enr[PCA_Subset17_enr$pval<0.05 & PCA_Subset17_enr$fold_change>1,]

datatable(PCA_Subset17_enr)

write.table(PCA_Subset17_enr, file="./03_output_files/PCA_Subset17_enr.txt",sep = "\t",row.names = F)
```

## Subset18

The sample grouping was checked by h-clustering
```{r Subset18_Hclust}
romicsHclust(Subset18)
```

and then by PCA
```{r Subset18_PCA}
indPCAplot(Subset18,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset18,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset18<-romicsPCA(Subset18)
write.csv(PCA_Subset18$var$contrib,file="03_output_files/PCA_Subset18_contrib.csv")

universe <- rownames(PCA_Subset18$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset18$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset18$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset18_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset18_enr<-PCA_Subset18_enr[PCA_Subset18_enr$pval<0.05 & PCA_Subset18_enr$fold_change>1,]

datatable(PCA_Subset18_enr)

write.table(PCA_Subset18_enr, file="./03_output_files/PCA_Subset18_enr.txt",sep = "\t",row.names = F)
```

## Subset19

The sample grouping was checked by h-clustering
```{r Subset19_Hclust}
romicsHclust(Subset19)
```

and then by PCA
```{r Subset19_PCA}
indPCAplot(Subset19,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset19,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset19<-romicsPCA(Subset19)
write.csv(PCA_Subset19$var$contrib,file="03_output_files/PCA_Subset19_contrib.csv")

universe <- rownames(PCA_Subset19$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset19$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset19$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset19_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset19_enr<-PCA_Subset19_enr[PCA_Subset19_enr$pval<0.05 & PCA_Subset19_enr$fold_change>1,]

datatable(PCA_Subset19_enr)

write.table(PCA_Subset19_enr, file="./03_output_files/PCA_Subset19_enr.txt",sep = "\t",row.names = F)
```

## Subset20

The sample grouping was checked by h-clustering
```{r Subset20_Hclust}
romicsHclust(Subset20)
```

and then by PCA
```{r Subset20_PCA}
indPCAplot(Subset20,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset20,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset20<-romicsPCA(Subset20)
write.csv(PCA_Subset20$var$contrib,file="03_output_files/PCA_Subset20_contrib.csv")

universe <- rownames(PCA_Subset20$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset20$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset20$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset20_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset20_enr<-PCA_Subset20_enr[PCA_Subset20_enr$pval<0.05 & PCA_Subset20_enr$fold_change>1,]

datatable(PCA_Subset20_enr)

write.table(PCA_Subset20_enr, file="./03_output_files/PCA_Subset20_enr.txt",sep = "\t",row.names = F)
```

## Subset21

The sample grouping was checked by h-clustering
```{r Subset21_Hclust}
romicsHclust(Subset21)
```

and then by PCA
```{r Subset21_PCA}
indPCAplot(Subset21,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset21,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset21<-romicsPCA(Subset21)
write.csv(PCA_Subset21$var$contrib,file="03_output_files/PCA_Subset21_contrib.csv")

universe <- rownames(PCA_Subset21$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset21$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset21$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset21_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset21_enr<-PCA_Subset21_enr[PCA_Subset21_enr$pval<0.05 & PCA_Subset21_enr$fold_change>1,]

datatable(PCA_Subset21_enr)

write.table(PCA_Subset21_enr, file="./03_output_files/PCA_Subset21_enr.txt",sep = "\t",row.names = F)
```
## Subset22

The sample grouping was checked by h-clustering
```{r Subset22_Hclust}
romicsHclust(Subset22)
```

and then by PCA
```{r Subset22_PCA}
indPCAplot(Subset22,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset22,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset22<-romicsPCA(Subset22)
write.csv(PCA_Subset22$var$contrib,file="03_output_files/PCA_Subset22_contrib.csv")

universe <- rownames(PCA_Subset22$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset22$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset22$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

PC3_10percent <- PCA_Subset22$var$contrib[,3]
PC3_10percent <- PC3_10percent[order(PC3_10percent,decreasing = T)]
PC3_10percent <- names(PC3_10percent)[1:ceiling(length(PC3_10percent)/10)]


print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC3_10percent_enr_GO<-cbind(component="PC3(top10%)",UniProt_GO_EASE(PC3_10percent,universe))

PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))
PC3_10percent_enr_KEGG<-cbind(component="PC3(top10%)",UniProt_KEGG_EASE(PC3_10percent,universe))

PCA_Subset22_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG, PC3_10percent_enr_GO, PC3_10percent_enr_KEGG)

PCA_Subset22_enr<-PCA_Subset22_enr[PCA_Subset22_enr$pval<0.05 & PCA_Subset22_enr$fold_change>1,]

datatable(PCA_Subset22_enr)

write.table(PCA_Subset22_enr, file="./03_output_files/PCA_Subset22_enr.txt",sep = "\t",row.names = F)
```

For this group of Samples the sample clustering was evaluated
```{r Subset22_sampleClust}
Subset22_samples<-Subset22$data
Subset22_samples<-t(Subset22_samples)

print("First the distance between the samples will be evaluated")
distance <- get_dist(Subset22_samples)
fviz_dist(distance, gradient = list(low = "#00AFBB", mid = "white", high = "#FC4E07"))

print("K-mean will be use to establish the clusters for a number of clusters of 2,3,4,5,6, and 7")
k2Subset22<-kmeans(Subset22_samples,centers=2,nstart=25)
fviz_cluster(k2Subset22, data = Subset22_samples)

k3Subset22<-kmeans(Subset22_samples,centers=3,nstart=25)
fviz_cluster(k3Subset22, data = Subset22_samples)

k4Subset22<-kmeans(Subset22_samples,centers=4,nstart=25)
fviz_cluster(k4Subset22, data = Subset22_samples)

k5Subset22<-kmeans(Subset22_samples,centers=5,nstart=25)
fviz_cluster(k5Subset22, data = Subset22_samples)

k6Subset22<-kmeans(Subset22_samples,centers=6,nstart=25)
fviz_cluster(k6Subset22, data = Subset22_samples)

k7Subset22<-kmeans(Subset22_samples,centers=7,nstart=25)
fviz_cluster(k7Subset22, data = Subset22_samples)

k8Subset22<-kmeans(Subset22_samples,centers=8,nstart=25)
fviz_cluster(k8Subset22, data = Subset22_samples)

k9Subset22<-kmeans(Subset22_samples,centers=9,nstart=25)
fviz_cluster(k9Subset22, data = Subset22_samples)

groups22<- t(Subset22$metadata[5,])
groups22<- cbind(group=groups22,group_number=as.integer(as.factor(groups22[1:length(groups22)])))
groups22<- data.frame(cbind(groups22,rownames(Subset22_samples)))

print("The best number of clusters will be evaluated by the Elbow method")

fviz_nbclust(Subset22_samples, kmeans, method = "wss")

print("Using the Silhouette function")

fviz_nbclust(Subset22_samples, kmeans, method = "silhouette")

print("and using the Gap Statistics")

fviz_nbclust(Subset22_samples, kmeans, method = "gap_stat")

print("Finally for the 3 clusters corresponding to the three biological sample types intracluster and extracluster distances were calculated using the package clv")

clust_dist<-cls.scatt.data(Subset22_samples,k3Subset22$cluster)
group_dist<-cls.scatt.data(Subset22_samples,groups22$group_number)

print("Average intercluster distances are :")
clust_dist$intracls.average

print("Average extracluster_distances are :")
clust_dist$intercls.average

print("Average intragroups (AS(F adult)=c1, FSGS(M)=c2, FSGS(M)=c3, MN(F)=c4, MN(M)=c5, Ctrl=c6)) distances are :")
group_dist$intracls.average

print("Average intergroups (AS(F adult)=c1, FSGS(M)=c2, FSGS(M)=c3, MN(F)=c4, MN(M)=c5, Ctrl=c6)) distances are :")
group_dist$intercls.average
```

## Subset23

The sample grouping was checked by h-clustering
```{r Subset23_Hclust}
romicsHclust(Subset23)
```

and then by PCA
```{r Subset23_PCA}
indPCAplot(Subset23,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset23,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset23<-romicsPCA(Subset23)
write.csv(PCA_Subset23$var$contrib,file="03_output_files/PCA_Subset23_contrib.csv")

universe <- rownames(PCA_Subset23$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset23$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset23$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset23_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset23_enr<-PCA_Subset23_enr[PCA_Subset23_enr$pval<0.05 & PCA_Subset23_enr$fold_change>1,]

datatable(PCA_Subset23_enr)

write.table(PCA_Subset23_enr, file="./03_output_files/PCA_Subset23_enr.txt",sep = "\t",row.names = F)

```

## Subset24

The sample grouping was checked by h-clustering
```{r Subset24_Hclust}
romicsHclust(Subset24)
```

and then by PCA
```{r Subset24_PCA}
indPCAplot(Subset24,label = F,Xcomp = 1,Ycomp = 2,plotType = "individual")
indPCA3D(Subset24,Xcomp = 1,Ycomp = 2,Zcomp = 3)
PCA_Subset24<-romicsPCA(Subset24)
write.csv(PCA_Subset24$var$contrib,file="03_output_files/PCA_Subset24_contrib.csv")

universe <- rownames(PCA_Subset24$var$contrib)

print("For the PC1 and PC2 the top 10% transcript contributing to the separation were extracted")

PC1_10percent <- PCA_Subset24$var$contrib[,1]
PC1_10percent <- PC1_10percent[order(PC1_10percent,decreasing = T)]
PC1_10percent <- names(PC1_10percent)[1:ceiling(length(PC1_10percent)/10)]

PC2_10percent <- PCA_Subset24$var$contrib[,2]
PC2_10percent <- PC2_10percent[order(PC2_10percent,decreasing = T)]
PC2_10percent <- names(PC2_10percent)[1:ceiling(length(PC2_10percent)/10)]

print("Enrichment analysis were then performed")

PC1_10percent_enr_GO<-cbind(component="PC1(top10%)",UniProt_GO_EASE(PC1_10percent,universe))
PC2_10percent_enr_GO<-cbind(component="PC2(top10%)",UniProt_GO_EASE(PC2_10percent,universe))
PC1_10percent_enr_KEGG<-cbind(component="PC1(top10%)",UniProt_KEGG_EASE(PC1_10percent,universe))
PC2_10percent_enr_KEGG<-cbind(component="PC2(top10%)",UniProt_KEGG_EASE(PC2_10percent,universe))

PCA_Subset24_enr<- rbind(PC1_10percent_enr_GO, PC1_10percent_enr_KEGG, PC2_10percent_enr_GO, PC2_10percent_enr_KEGG)

PCA_Subset24_enr<-PCA_Subset24_enr[PCA_Subset24_enr$pval<0.05 & PCA_Subset24_enr$fold_change>1,]

datatable(PCA_Subset24_enr)

write.table(PCA_Subset24_enr, file="./03_output_files/PCA_Subset24_enr.txt",sep = "\t",row.names = F)
```

# Heatmap generation
For all the Subsets that had more than one group, an ANOVA was performed to identify the genes that were  expressed differentially and only those genes were plotted on the heatmap. For the other single groups, we did not applied an ANOVA pre-filter but performed the clustering on the whole data.

## Subset01
```{r Subset01_Heatmap}
Subset01<-romicsZscores(Subset01)

print("The Z-scores of Subset01 were used to evaluate the best number of clusters")
Subset01_z<-Subset01$statistics[,grepl("Z_scores",colnames(Subset01$statistics))]

fviz_nbclust(Subset01_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset01_z, kmeans, method = "wss")+geom_vline(xintercept = 5, linetype = 2)

romicsHeatmap(Subset01,variable_hclust_number = 6)
Subset01<-romicsVariableHclust(Subset01,clusters = 6,plot = F)

Subset01_enrichment<-romicsEnrichement(Subset01,cluster_filter = "hclust_clusters")
Subset01_enrichment<-Subset01_enrichment[Subset01_enrichment$pval<0.05&Subset01_enrichment$fold_change>1,]
```
## Subset02

```{r Subset02_Heatmap}
Subset02<-romicsZscores(Subset02)

print("The Z-scores of Subset02 were used to evaluate the best number of clusters")
Subset02_z<-Subset02$statistics[,grepl("Z_scores",colnames(Subset02$statistics))]

fviz_nbclust(Subset02_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset02_z, kmeans, method = "wss")+
geom_vline(xintercept = 5, linetype = 2)

romicsHeatmap(Subset02,variable_hclust_number = 5)
Subset02<-romicsVariableHclust(Subset02,clusters = 5,plot = F)

Subset02_enrichment<-romicsEnrichement(Subset02,cluster_filter = "hclust_clusters")
Subset02_enrichment<-Subset02_enrichment[Subset02_enrichment$pval<0.05&Subset02_enrichment$fold_change>1,]
```

## Subset03

```{r Subset03_Heatmap}
Subset03<-romicsANOVA(Subset03,factor = "disease_gen")
Subset03<-romicsZscores(Subset03)

print(paste0(sum(Subset03$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset03"))
pval<-data.frame(ids=rownames(Subset03$statistics), p=Subset03$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset03")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset03$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset03"))
pval<-data.frame(ids=rownames(Subset03$statistics), p=Subset03$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset03")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset03 were used to evaluate the best number of clusters to use for the Heatmap")
Subset03_z<-Subset03$statistics[Subset03$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset03$statistics))]


fviz_nbclust(Subset03_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset03_z, kmeans, method = "wss")+
geom_vline(xintercept = 2, linetype = 2)

romicsHeatmap(Subset03,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 2)
print("The clusters were then recorded in the statistics layer")
Subset03<-romicsVariableHclust(Subset03,ANOVA_filter = "padj",p=0.05,clusters = 2,plot = F)

Subset03_enrichment<-romicsEnrichement(Subset03,cluster_filter = "hclust_clusters")
Subset03_enrichment<-Subset03_enrichment[Subset03_enrichment$pval<0.05&Subset03_enrichment$fold_change>1,]
```

## Subset04

```{r Subset04_Heatmap}

Subset04<-romicsANOVA(Subset04,factor = "disease_gen")
Subset04<-romicsZscores(Subset04)

print(paste0(sum(Subset04$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset04"))
pval<-data.frame(ids=rownames(Subset04$statistics), p=Subset04$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset04")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset04$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset04"))
pval<-data.frame(ids=rownames(Subset04$statistics), p=Subset04$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset04")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset04 were used to evaluate the best number of clusters to use for the Heatmap")
Subset04_z<-Subset04$statistics[Subset04$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset04$statistics))]

fviz_nbclust(Subset04_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset04_z, kmeans, method = "wss")+
geom_vline(xintercept = 2, linetype = 2)

romicsHeatmap(Subset04,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 2)
print("The clusters were then recorded in the statistics layer")
Subset04<-romicsVariableHclust(Subset04,ANOVA_filter = "padj",p=0.05,clusters = 2,plot = F)

Subset04_enrichment<-romicsEnrichement(Subset04,cluster_filter = "hclust_clusters")
Subset04_enrichment<-Subset04_enrichment[Subset04_enrichment$pval<0.05&Subset04_enrichment$fold_change>1,]
```

## Subset05

```{r Subset05_Heatmap}
Subset05<-romicsANOVA(Subset05,factor = "disease_gen")
Subset05<-romicsZscores(Subset05)

print(paste0(sum(Subset05$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset05"))
pval<-data.frame(ids=rownames(Subset05$statistics), p=Subset05$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset05")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset05$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset05"))
pval<-data.frame(ids=rownames(Subset05$statistics), p=Subset05$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset05")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset05 were used to evaluate the best number of clusters to use for the Heatmap")
Subset05_z<-Subset05$statistics[Subset05$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset05$statistics))]


fviz_nbclust(Subset05_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset05_z, kmeans, method = "wss")+
geom_vline(xintercept = 2, linetype = 2)

romicsHeatmap(Subset05,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 2)
print("The clusters were then recorded in the statistics layer")
Subset05<-romicsVariableHclust(Subset05,ANOVA_filter = "padj",p=0.05,clusters = 2,plot = F)

Subset05_enrichment<-romicsEnrichement(Subset05,cluster_filter = "hclust_clusters")
Subset05_enrichment<-Subset05_enrichment[Subset05_enrichment$pval<0.05&Subset05_enrichment$fold_change>1,]
```

For this subset we performed a trajectory analysis using the package SLICER
```{r Subset05_Trajectory}
library("SLICER")
library("lle")

table_Subset05<-t(Subset05$data)
genes <- select_genes(table_Subset05)
k = select_k(table_Subset05[,genes], kmin=5)
traj_lle = lle(table_Subset05[,genes], m=2, k)$Y

print("the points of the trajectory are plotted")

traj_graph = conn_knn_graph(traj_lle,5)
ends = find_extreme_cells(traj_graph, traj_lle)
start = 1

print("The trajectory connections were plotted")

distances = graph_process_distance(traj_graph,embedding = traj_lle,start = 1)

traj_lle_for_graph = as.data.frame(traj_lle)
colnames(traj_lle_for_graph) = c("Manifold_dim_1","Manifold_dim_2")
traj_lle_for_graph$Manifold_dim_1 = as.numeric(traj_lle_for_graph$Manifold_dim_1)
traj_lle_for_graph$Manifold_dim_2 = as.numeric(traj_lle_for_graph$Manifold_dim_2)
traj_lle_for_graph<- cbind(ROI= rownames(table_Subset05), disease_gen= t(Subset05$metadata[5,]), traj_lle_for_graph, branches =  assign_branches(traj_graph,start = 1))

print("The plots is colored by condition")

ggplot(traj_lle_for_graph,aes(x=Manifold_dim_1,Manifold_dim_2,color=disease_gen))+ geom_point(size=1,)+ geom_text(label=traj_lle_for_graph$ROI)+theme_ROP()

DT::datatable(traj_lle_for_graph)
```

## Subset06

```{r Subset06_Heatmap}

Subset06<-romicsANOVA(Subset06,factor = "disease_gen")
Subset06<-romicsZscores(Subset06)

print(paste0(sum(Subset06$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset06"))
pval<-data.frame(ids=rownames(Subset06$statistics), p=Subset06$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset06")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset06$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset06"))
pval<-data.frame(ids=rownames(Subset06$statistics), p=Subset06$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset06")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset06 were used to evaluate the best number of clusters to use for the Heatmap")
Subset06_z<-Subset06$statistics[Subset06$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset06$statistics))]

fviz_nbclust(Subset06_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset06_z, kmeans, method = "wss")+
geom_vline(xintercept = 3, linetype = 2)

romicsHeatmap(Subset06,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 3)
print("The clusters were then recorded in the statistics layer")
Subset06<-romicsVariableHclust(Subset06,ANOVA_filter = "padj",p=0.05,clusters = 3,plot = F)

Subset06_enrichment<-romicsEnrichement(Subset06,cluster_filter = "hclust_clusters")
Subset06_enrichment<-Subset06_enrichment[Subset06_enrichment$pval<0.05&Subset06_enrichment$fold_change>1,]
```

For this subset we performed a pseudo-time analysis using the Banff classification pathology scores
```{r Subset06_Trend}
Subset06<-romicsTrend(Subset06,factor = "Total_score_plus_one",log_factor = F,type = "linear",p = 0.05)
pFrequencyPlot(Subset06)
romicsTrendHeatmap(Subset06)
linear_increasing<-rownames(Subset06$statistics)[Subset06$statistics$best_fitted_trend=="linear_increasing"]
linear_decreasing<-rownames(Subset06$statistics)[Subset06$statistics$best_fitted_trend=="linear_decreasing"]
universe<-rownames(Subset06$statistics)
linear_increasing_enr_GO<-cbind(trend="linear_increasing",UniProt_GO_EASE(linear_increasing,universe))
linear_decreasing_enr_GO<-cbind(trend="linear_decreasing",UniProt_GO_EASE(linear_decreasing,universe))
linear_increasing_enr_KEGG<-cbind(trend="linear_increasing",UniProt_KEGG_EASE(linear_increasing,universe))
linear_decreasing_enr_KEGG<-cbind(trend="linear_decreasing",UniProt_KEGG_EASE(linear_decreasing,universe))

trend_enr<-rbind(linear_increasing_enr_GO,linear_decreasing_enr_GO,linear_increasing_enr_KEGG, linear_decreasing_enr_KEGG)
trend_enr<-trend_enr[trend_enr$pval<0.05 & trend_enr$fold_change>1,]

results<- romicsExportData(Subset06,statistics = T)
write.csv(results,file="./03_output_files/Subset06_with_trend.csv")
write.table(trend_enr, file="./03_output_files/Subset06_enrichments_trend.txt",sep="\t")
```
For this subset we performed a trajectory analysis using the package SLICER
```{r Subset06_Trajectory}
library("SLICER")
library("lle")

table_Subset06<-t(Subset06$data)
genes <- select_genes(table_Subset06)
k = select_k(table_Subset06[,genes], kmin=5)
traj_lle = lle(table_Subset06[,genes], m=2, k)$Y

print("the points of the trajectory are plotted")

traj_graph = conn_knn_graph(traj_lle,5)
ends = find_extreme_cells(traj_graph, traj_lle)
start = 1

print("The trajectory connections were plotted")

distances = graph_process_distance(traj_graph,embedding = traj_lle,start = 1)

traj_lle_for_graph = as.data.frame(traj_lle)
colnames(traj_lle_for_graph) = c("Manifold_dim_1","Manifold_dim_2")
traj_lle_for_graph$Manifold_dim_1 = as.numeric(traj_lle_for_graph$Manifold_dim_1)
traj_lle_for_graph$Manifold_dim_2 = as.numeric(traj_lle_for_graph$Manifold_dim_2)
traj_lle_for_graph<- cbind(ROI= rownames(table_Subset06), disease_gen= t(Subset06$metadata[5,]), traj_lle_for_graph, branches =  assign_branches(traj_graph,start = 1))

print("The plots is colored by condition")

ggplot(traj_lle_for_graph,aes(x=Manifold_dim_1,Manifold_dim_2,color=disease_gen))+ geom_point(size=1,)+ geom_text(label=traj_lle_for_graph$ROI)+theme_ROP()

DT::datatable(traj_lle_for_graph)
```

## Subset07
```{r Subset07_Heatmap}
Subset07<-romicsZscores(Subset07)

print("The Z-scores of Subset07 were used to evaluate the best number of clusters")
Subset07_z<-Subset07$statistics[,grepl("Z_scores",colnames(Subset07$statistics))]

fviz_nbclust(Subset07_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset07_z, kmeans, method = "wss")+geom_vline(xintercept = 2, linetype = 5)

Subset07<-romicsZscores(Subset07)

romicsHeatmap(Subset07,variable_hclust_number = 2)
Subset07_2c<-romicsVariableHclust(Subset07,clusters = 2,plot = F)

romicsHeatmap(Subset07,variable_hclust_number = 4)
Subset07_4c<-romicsVariableHclust(Subset07,clusters = 4,plot = F)

Subset07_2c_enrichment<-romicsEnrichement(Subset07_2c,cluster_filter = "hclust_clusters")
Subset07_2c_enrichment<-Subset07_2c_enrichment[Subset07_2c_enrichment$pval<0.05&Subset07_2c_enrichment$fold_change>1,]

Subset07_4c_enrichment<-romicsEnrichement(Subset07_4c,cluster_filter = "hclust_clusters")
Subset07_4c_enrichment<-Subset07_4c_enrichment[Subset07_4c_enrichment$pval<0.05&Subset07_4c_enrichment$fold_change>1,]
```

For this subset we performed a trajectory analysis using the package SLICER
```{r Subset07_Trajectory}
library("SLICER")
library("lle")

table_Subset07<-t(Subset07$data)
genes <- select_genes(table_Subset07)
k = select_k(table_Subset07[,genes], kmin=5)
traj_lle = lle(table_Subset07[,genes], m=2, k)$Y

print("the points of the trajectory are plotted")

traj_graph = conn_knn_graph(traj_lle,5)
ends = find_extreme_cells(traj_graph, traj_lle)
start = 1

print("The trajectory connections were plotted")

distances = graph_process_distance(traj_graph,embedding = traj_lle,start = 1)

traj_lle_for_graph = as.data.frame(traj_lle)
colnames(traj_lle_for_graph) = c("Manifold_dim_1","Manifold_dim_2")
traj_lle_for_graph$Manifold_dim_1 = as.numeric(traj_lle_for_graph$Manifold_dim_1)
traj_lle_for_graph$Manifold_dim_2 = as.numeric(traj_lle_for_graph$Manifold_dim_2)
traj_lle_for_graph<- cbind(ROI= rownames(table_Subset07), disease_gen= t(Subset07$metadata[5,]), traj_lle_for_graph, branches =  assign_branches(traj_graph,start = 1))

print("The plots is colored by condition")

ggplot(traj_lle_for_graph,aes(x=Manifold_dim_1,Manifold_dim_2,color=disease_gen))+ geom_point(size=1,)+ geom_text(label=traj_lle_for_graph$ROI)+theme_ROP()

DT::datatable(traj_lle_for_graph)
```

## Subset08

```{r Subset08_Heatmap}
Subset08<-romicsANOVA(Subset08,factor = "disease_gen")
Subset08<-romicsZscores(Subset08)

print(paste0(sum(Subset08$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset08"))
pval<-data.frame(ids=rownames(Subset08$statistics), p=Subset08$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset08")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset08$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset08"))
pval<-data.frame(ids=rownames(Subset08$statistics), p=Subset08$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset08")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset08 were used to evaluate the best number of clusters to use for the Heatmap")
Subset08_z<-Subset08$statistics[Subset08$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset08$statistics))]

fviz_nbclust(Subset08_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset08_z, kmeans, method = "wss")+
geom_vline(xintercept = 2, linetype = 3)

romicsHeatmap(Subset08,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 2)
print("The clusters were then recorded in the statistics layer")
Subset08<-romicsVariableHclust(Subset08,ANOVA_filter = "padj",p=0.05,clusters = 2,plot = F)

Subset08_enrichment<-romicsEnrichement(Subset08,cluster_filter = "hclust_clusters")
Subset08_enrichment<-Subset08_enrichment[Subset08_enrichment$pval<0.05&Subset08_enrichment$fold_change>1,]

Subset08_NAST<-Subset08
```

For this subset we performed a pseudo-time analysis using the Banff classification pathology scores
```{r Subset08_Trend_NAST}
Subset08_NAST<-romicsTrend(Subset08_NAST,factor = "Total_score_Nast_plus_one",log_factor = F,type = "linear",p = 0.05)
pFrequencyPlot(Subset08_NAST)
romicsTrendHeatmap(Subset08_NAST)
linear_increasing<-rownames(Subset08_NAST$statistics)[Subset08_NAST$statistics$best_fitted_trend=="linear_increasing"]
linear_decreasing<-rownames(Subset08_NAST$statistics)[Subset08_NAST$statistics$best_fitted_trend=="linear_decreasing"]
universe<-rownames(Subset08_NAST$statistics)
linear_increasing_enr_GO<-cbind(trend="linear_increasing",UniProt_GO_EASE(linear_increasing,universe))
linear_decreasing_enr_GO<-cbind(trend="linear_decreasing",UniProt_GO_EASE(linear_decreasing,universe))
linear_increasing_enr_KEGG<-cbind(trend="linear_increasing",UniProt_KEGG_EASE(linear_increasing,universe))
linear_decreasing_enr_KEGG<-cbind(trend="linear_decreasing",UniProt_KEGG_EASE(linear_decreasing,universe))

trend_enr<-rbind(linear_increasing_enr_GO,linear_decreasing_enr_GO,linear_increasing_enr_KEGG, linear_decreasing_enr_KEGG)
trend_enr<-trend_enr[trend_enr$pval<0.05 & trend_enr$fold_change>1,]

results<- romicsExportData(Subset08_NAST,statistics = T)
write.csv(results,file="./03_output_files/Subset08_with_trend_NAST.csv")
write.table(trend_enr, file="./03_output_files/Subset08_enrichments_trend_NAST.txt",sep="\t")
```

For this subset we performed a trajectory analysis using the package SLICER
```{r Subset08_Trajectory}
library("SLICER")
library("lle")

table_Subset08<-t(Subset08$data)
genes <- select_genes(table_Subset08)
k = select_k(table_Subset08[,genes], kmin=5)
traj_lle = lle(table_Subset08[,genes], m=2, k)$Y

print("the points of the trajectory are plotted")

traj_graph = conn_knn_graph(traj_lle,5)
ends = find_extreme_cells(traj_graph, traj_lle)
start = 1

print("The trajectory connections were plotted")

distances = graph_process_distance(traj_graph,embedding = traj_lle,start = 1)

traj_lle_for_graph = as.data.frame(traj_lle)
colnames(traj_lle_for_graph) = c("Manifold_dim_1","Manifold_dim_2")
traj_lle_for_graph$Manifold_dim_1 = as.numeric(traj_lle_for_graph$Manifold_dim_1)
traj_lle_for_graph$Manifold_dim_2 = as.numeric(traj_lle_for_graph$Manifold_dim_2)
traj_lle_for_graph<- cbind(ROI= rownames(table_Subset08), disease_gen= t(Subset08$metadata[5,]), traj_lle_for_graph, branches =  assign_branches(traj_graph,start = 1))

print("The plots is colored by condition")

ggplot(traj_lle_for_graph,aes(x=Manifold_dim_1,Manifold_dim_2,color=disease_gen))+ geom_point(size=1,)+ geom_text(label=traj_lle_for_graph$ROI)+theme_ROP()

DT::datatable(traj_lle_for_graph)
```

## Subset09
```{r Subset09_Heatmap}
Subset09<-romicsZscores(Subset09)

print("The Z-scores of Subset09 were used to evaluate the best number of clusters")
Subset09_z<-Subset09$statistics[,grepl("Z_scores",colnames(Subset09$statistics))]

fviz_nbclust(Subset09_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset09_z, kmeans, method = "wss")+geom_vline(xintercept = 3, linetype = 5)

romicsHeatmap(Subset09,variable_hclust_number = 3)
Subset09_3c<-romicsVariableHclust(Subset09,clusters = 3,plot = F)

romicsHeatmap(Subset09,variable_hclust_number = 4)
Subset09_4c<-romicsVariableHclust(Subset09,clusters = 4,plot = F)

Subset09_3c_enrichment<-romicsEnrichement(Subset09_3c,cluster_filter = "hclust_clusters")
Subset09_3c_enrichment<-Subset09_3c_enrichment[Subset09_3c_enrichment$pval<0.05&Subset09_3c_enrichment$fold_change>1,]

Subset09_4c_enrichment<-romicsEnrichement(Subset09_4c,cluster_filter = "hclust_clusters")
Subset09_4c_enrichment<-Subset09_4c_enrichment[Subset09_4c_enrichment$pval<0.05&Subset09_4c_enrichment$fold_change>1,]
```

## Subset10
```{r Subset10_Heatmap}
Subset10<-romicsZscores(Subset10)

print("The Z-scores of Subset10 were used to evaluate the best number of clusters")
Subset10_z<-Subset10$statistics[,grepl("Z_scores",colnames(Subset10$statistics))]

fviz_nbclust(Subset10_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset10_z, kmeans, method = "wss")+geom_vline(xintercept = 4, linetype = 5)

romicsHeatmap(Subset10,variable_hclust_number = 4)
Subset10<-romicsVariableHclust(Subset10,clusters = 4,plot = F)

Subset10_enrichment<-romicsEnrichement(Subset10,cluster_filter = "hclust_clusters")
Subset10_enrichment<-Subset10_enrichment[Subset10_enrichment$pval<0.05&Subset10_enrichment$fold_change>1,]
```

## Subset11

```{r Subset11_Heatmap}
Subset11<-romicsANOVA(Subset11,factor = "disease_gen")
Subset11<-romicsZscores(Subset11)

print(paste0(sum(Subset11$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset11"))
pval<-data.frame(ids=rownames(Subset11$statistics), p=Subset11$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset11")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset11$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset11"))
pval<-data.frame(ids=rownames(Subset11$statistics), p=Subset11$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset11")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset11 were used to evaluate the best number of clusters to use for the Heatmap")
Subset11_z<-Subset11$statistics[Subset11$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset11$statistics))]


fviz_nbclust(Subset11_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset11_z, kmeans, method = "wss")+
geom_vline(xintercept = 2, linetype = 3)

romicsHeatmap(Subset11,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 2)
print("The clusters were then recorded in the statistics layer")
Subset11<-romicsVariableHclust(Subset11,ANOVA_filter = "padj",p=0.05,clusters = 2,plot = F)

Subset11_enrichment<-romicsEnrichement(Subset11,cluster_filter = "hclust_clusters")
Subset11_enrichment<-Subset11_enrichment[Subset11_enrichment$pval<0.05&Subset11_enrichment$fold_change>1,]
```

## Subset12

```{r Subset12_Heatmap}
Subset12<-romicsANOVA(Subset12,factor = "disease_gen")
Subset12<-romicsZscores(Subset12)

print(paste0(sum(Subset12$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset12"))
pval<-data.frame(ids=rownames(Subset12$statistics), p=Subset12$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset12")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset12$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset12"))
pval<-data.frame(ids=rownames(Subset12$statistics), p=Subset12$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset12")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset12 were used to evaluate the best number of clusters to use for the Heatmap")
Subset12_z<-Subset12$statistics[Subset12$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset12$statistics))]


fviz_nbclust(Subset12_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset12_z, kmeans, method = "wss")+
geom_vline(xintercept = 2, linetype = 3)

romicsHeatmap(Subset12,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 3)
print("The clusters were then recorded in the statistics layer")
Subset12<-romicsVariableHclust(Subset12,ANOVA_filter = "padj",p=0.05,clusters = 3,plot = F)

Subset12_enrichment<-romicsEnrichement(Subset12,cluster_filter = "hclust_clusters")
Subset12_enrichment<-Subset12_enrichment[Subset12_enrichment$pval<0.05&Subset12_enrichment$fold_change>1,]
```

## Subset13

```{r Subset13_Heatmap}

Subset13<-romicsANOVA(Subset13,factor = "disease_gen")
Subset13<-romicsZscores(Subset13)

print(paste0(sum(Subset13$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset13"))
pval<-data.frame(ids=rownames(Subset13$statistics), p=Subset13$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset13")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset13$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset13"))
pval<-data.frame(ids=rownames(Subset13$statistics), p=Subset13$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset13")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset13 were used to evaluate the best number of clusters to use for the Heatmap")
Subset13_z<-Subset13$statistics[Subset13$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset13$statistics))]

print("for FSGS no transcript was significantly modulated between male and females (padj<0.05)")
#fviz_nbclust(Subset13_z, kmeans, method = "silhouette", k.max = 10) + theme_bw() + ggtitle("The Silhouette Plot")
#fviz_nbclust(Subset13_z, kmeans, method = "wss")+
#geom_vline(xintercept = 2, linetype = 3)

#romicsHeatmap(Subset13,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 3)
#print("The clusters were then recorded in the statistics layer")
#Subset13<-romicsVariableHclust(Subset13,ANOVA_filter = "padj",p=0.05,clusters = 3,plot = F)
```
For this subset we performed a trajectory analysis using the package SLICER
```{r Subset13_Trajectory}
library("SLICER")
library("lle")

table_Subset13<-t(Subset13$data)
genes <- select_genes(table_Subset13)
k = select_k(table_Subset13[,genes], kmin=5)
traj_lle = lle(table_Subset13[,genes], m=2, k)$Y

print("the points of the trajectory are plotted")

traj_graph = conn_knn_graph(traj_lle,5)
ends = find_extreme_cells(traj_graph, traj_lle)
start = 1

print("The trajectory connections were plotted")

distances = graph_process_distance(traj_graph,embedding = traj_lle,start = 1)

traj_lle_for_graph = as.data.frame(traj_lle)
colnames(traj_lle_for_graph) = c("Manifold_dim_1","Manifold_dim_2")
traj_lle_for_graph$Manifold_dim_1 = as.numeric(traj_lle_for_graph$Manifold_dim_1)
traj_lle_for_graph$Manifold_dim_2 = as.numeric(traj_lle_for_graph$Manifold_dim_2)
traj_lle_for_graph<- cbind(ROI= rownames(table_Subset13), disease_gen= t(Subset13$metadata[5,]), traj_lle_for_graph, branches =  assign_branches(traj_graph,start = 1))

print("The plots is colored by condition")

ggplot(traj_lle_for_graph,aes(x=Manifold_dim_1,Manifold_dim_2,color=disease_gen))+ geom_point(size=1,)+ geom_text(label=traj_lle_for_graph$ROI)+theme_ROP()

DT::datatable(traj_lle_for_graph)
```
## Subset14

```{r FFSubset14_Heatmap}
Subset14<-romicsANOVA(Subset14,factor = "disease_gen")
Subset14<-romicsZscores(Subset14)

print(paste0(sum(Subset14$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset14"))
pval<-data.frame(ids=rownames(Subset14$statistics), p=Subset14$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset14")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset14$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset14"))
pval<-data.frame(ids=rownames(Subset14$statistics), p=Subset14$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset14")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset14 were used to evaluate the best number of clusters to use for the Heatmap")
Subset14_z<-Subset14$statistics[Subset14$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset14$statistics))]

fviz_nbclust(Subset14_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset14_z, kmeans, method = "wss")+
geom_vline(xintercept = 2, linetype = 3)

romicsHeatmap(Subset14,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 2)
print("The clusters were then recorded in the statistics layer")
Subset14_2c<-romicsVariableHclust(Subset14,ANOVA_filter = "padj",p=0.05,clusters = 2,plot = F)

romicsHeatmap(Subset14,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 3)
print("The clusters were then recorded in the statistics layer")
Subset14_3c<-romicsVariableHclust(Subset14,ANOVA_filter = "padj",p=0.05,clusters = 3,plot = F)


Subset14_2c_enrichment<-romicsEnrichement(Subset14_2c,cluster_filter = "hclust_clusters")
Subset14_2c_enrichment<-Subset14_2c_enrichment[Subset14_2c_enrichment$pval<0.05&Subset14_2c_enrichment$fold_change>1,]

Subset14_3c_enrichment<-romicsEnrichement(Subset14_3c,cluster_filter = "hclust_clusters")
Subset14_3c_enrichment<-Subset14_3c_enrichment[Subset14_3c_enrichment$pval<0.05&Subset14_3c_enrichment$fold_change>1,]
Subset14_NAST<-Subset14
```

For this subset we performed a pseudo-time analysis using the Banff classification pathology scores
```{r Subset14_Trend}
Subset14<-romicsTrend(Subset14,factor = "Total_score_plus_one",log_factor = F,type = "linear",p = 0.05)
pFrequencyPlot(Subset14)
romicsTrendHeatmap(Subset14)
linear_increasing<-rownames(Subset14$statistics)[Subset14$statistics$best_fitted_trend=="linear_increasing"]
linear_decreasing<-rownames(Subset14$statistics)[Subset14$statistics$best_fitted_trend=="linear_decreasing"]
universe<-rownames(Subset14$statistics)
linear_increasing_enr_GO<-cbind(trend="linear_increasing",UniProt_GO_EASE(linear_increasing,universe))
linear_decreasing_enr_GO<-cbind(trend="linear_decreasing",UniProt_GO_EASE(linear_decreasing,universe))
linear_increasing_enr_KEGG<-cbind(trend="linear_increasing",UniProt_KEGG_EASE(linear_increasing,universe))
linear_decreasing_enr_KEGG<-cbind(trend="linear_decreasing",UniProt_KEGG_EASE(linear_decreasing,universe))

trend_enr<-rbind(linear_increasing_enr_GO,linear_decreasing_enr_GO,linear_increasing_enr_KEGG, linear_decreasing_enr_KEGG)
trend_enr<-trend_enr[trend_enr$pval<0.05 & trend_enr$fold_change>1,]

results<- romicsExportData(Subset14,statistics = T)
write.csv(results,file="./03_output_files/Subset14_with_trend.csv")
write.table(trend_enr, file="./03_output_files/Subset14_enrichments_trend.txt",sep="\t")
```

For this subset we performed a pseudo-time analysis using the Banff classification pathology scores from NAST (2nd pathologist)
```{r Subset14_NAST_Trend}
Subset14_NAST<-romicsTrend(Subset14_NAST,factor = "Total_score_Nast_plus_one",log_factor = F,type = "linear",p = 0.05)
pFrequencyPlot(Subset14_NAST)
romicsTrendHeatmap(Subset14_NAST)
linear_increasing<-rownames(Subset14_NAST$statistics)[Subset14_NAST$statistics$best_fitted_trend=="linear_increasing"]
linear_decreasing<-rownames(Subset14_NAST$statistics)[Subset14_NAST$statistics$best_fitted_trend=="linear_decreasing"]
universe<-rownames(Subset14_NAST$statistics)
linear_increasing_enr_GO<-cbind(trend="linear_increasing",UniProt_GO_EASE(linear_increasing,universe))
linear_decreasing_enr_GO<-cbind(trend="linear_decreasing",UniProt_GO_EASE(linear_decreasing,universe))
linear_increasing_enr_KEGG<-cbind(trend="linear_increasing",UniProt_KEGG_EASE(linear_increasing,universe))
linear_decreasing_enr_KEGG<-cbind(trend="linear_decreasing",UniProt_KEGG_EASE(linear_decreasing,universe))

trend_enr<-rbind(linear_increasing_enr_GO,linear_decreasing_enr_GO,linear_increasing_enr_KEGG, linear_decreasing_enr_KEGG)
trend_enr<-trend_enr[trend_enr$pval<0.05 & trend_enr$fold_change>1,]

results<- romicsExportData(Subset14_NAST,statistics = T)
write.csv(results,file="./03_output_files/Subset14_with_trend_NAST.csv")
write.table(trend_enr, file="./03_output_files/Subset14_enrichments_trend_NAST.txt",sep="\t")
```

For this subset we performed a trajectory analysis using the package SLICER
```{r Subset14_Trajectory}
library("SLICER")
library("lle")

table_Subset14<-t(Subset14$data)
genes <- select_genes(table_Subset14)
k = select_k(table_Subset14[,genes], kmin=5)
traj_lle = lle(table_Subset14[,genes], m=2, k)$Y

print("the points of the trajectory are plotted")

traj_graph = conn_knn_graph(traj_lle,5)
ends = find_extreme_cells(traj_graph, traj_lle)
start = 1

print("The trajectory connections were plotted")

distances = graph_process_distance(traj_graph,embedding = traj_lle,start = 1)

traj_lle_for_graph = as.data.frame(traj_lle)
colnames(traj_lle_for_graph) = c("Manifold_dim_1","Manifold_dim_2")
traj_lle_for_graph$Manifold_dim_1 = as.numeric(traj_lle_for_graph$Manifold_dim_1)
traj_lle_for_graph$Manifold_dim_2 = as.numeric(traj_lle_for_graph$Manifold_dim_2)
traj_lle_for_graph<- cbind(ROI= rownames(table_Subset14), disease_gen= t(Subset14$metadata[5,]), traj_lle_for_graph, branches =  assign_branches(traj_graph,start = 1))

print("The plots is colored by condition")

ggplot(traj_lle_for_graph,aes(x=Manifold_dim_1,Manifold_dim_2,color=disease_gen))+ geom_point(size=1,)+ geom_text(label=traj_lle_for_graph$ROI)+theme_ROP()

DT::datatable(traj_lle_for_graph)
```

## Subset15
```{r Subset15_Heatmap}
Subset15<-romicsZscores(Subset15)

print("The Z-scores of Subset15 were used to evaluate the best number of clusters")
Subset15_z<-Subset15$statistics[,grepl("Z_scores",colnames(Subset15$statistics))]

fviz_nbclust(Subset15_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset15_z, kmeans, method = "wss")+geom_vline(xintercept = 5, linetype = 2)

romicsHeatmap(Subset15,variable_hclust_number = 5)
Subset15<-romicsVariableHclust(Subset15,clusters = 5,plot = F)

Subset15_enrichment<-romicsEnrichement(Subset15,cluster_filter = "hclust_clusters")
Subset15_enrichment<-Subset15_enrichment[Subset15_enrichment$pval<0.05&Subset15_enrichment$fold_change>1,]
```


## Subset16
```{r Subset16_Heatmap}
Subset16<-romicsZscores(Subset16)

print("The Z-scores of Subset16 were used to evaluate the best number of clusters")
Subset16_z<-Subset16$statistics[,grepl("Z_scores",colnames(Subset16$statistics))]

fviz_nbclust(Subset16_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset16_z, kmeans, method = "wss")+geom_vline(xintercept = 5, linetype = 2)

romicsHeatmap(Subset16,variable_hclust_number = 5)
Subset16<-romicsVariableHclust(Subset16,clusters = 5,plot = F)

Subset16_enrichment<-romicsEnrichement(Subset16,cluster_filter = "hclust_clusters")
Subset16_enrichment<-Subset16_enrichment[Subset16_enrichment$pval<0.05&Subset16_enrichment$fold_change>1,]
```
## Subset17

```{r Subset17_Heatmap}
Subset17<-romicsANOVA(Subset17,factor = "disease_gen")
Subset17<-romicsZscores(Subset17)

print(paste0(sum(Subset17$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset17"))
pval<-data.frame(ids=rownames(Subset17$statistics), p=Subset17$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset17")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset17$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset17"))
pval<-data.frame(ids=rownames(Subset17$statistics), p=Subset17$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset17")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset17 were used to evaluate the best number of clusters to use for the Heatmap")
Subset17_z<-Subset17$statistics[Subset17$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset17$statistics))]

fviz_nbclust(Subset17_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset17_z, kmeans, method = "wss")+
geom_vline(xintercept = 2, linetype = 3)

romicsHeatmap(Subset17,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 2)
print("The clusters were then recorded in the statistics layer")
Subset17_2c<-romicsVariableHclust(Subset17,ANOVA_filter = "padj",p=0.05,clusters = 2,plot = F)

romicsHeatmap(Subset17,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 3)
print("The clusters were then recorded in the statistics layer")
Subset17_3c<-romicsVariableHclust(Subset17,ANOVA_filter = "padj",p=0.05,clusters = 3,plot = F)

Subset17_2c_enrichment<-romicsEnrichement(Subset17_2c,cluster_filter = "hclust_clusters")
Subset17_2c_enrichment<-Subset17_2c_enrichment[Subset17_2c_enrichment$pval<0.05&Subset17_2c_enrichment$fold_change>1,]

Subset17_3c_enrichment<-romicsEnrichement(Subset17_3c,cluster_filter = "hclust_clusters")
Subset17_3c_enrichment<-Subset17_3c_enrichment[Subset17_3c_enrichment$pval<0.05&Subset17_3c_enrichment$fold_change>1,]
```


## Subset18
```{r Subset18_Heatmap}
Subset18<-romicsANOVA(Subset18,factor = "disease_gen")
Subset18<-romicsZscores(Subset18)

print(paste0(sum(Subset18$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset18"))
pval<-data.frame(ids=rownames(Subset18$statistics), p=Subset18$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset18")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset18$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset18"))
pval<-data.frame(ids=rownames(Subset18$statistics), p=Subset18$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset18")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset18 were used to evaluate the best number of clusters to use for the Heatmap")
Subset18_z<-Subset18$statistics[Subset18$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset18$statistics))]

fviz_nbclust(Subset18_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset18_z, kmeans, method = "wss")+
geom_vline(xintercept = 2, linetype = 3)

romicsHeatmap(Subset18,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 2)
print("The clusters were then recorded in the statistics layer")
Subset18<-romicsVariableHclust(Subset18,ANOVA_filter = "padj",p=0.05,clusters = 2,plot = F)

Subset18_enrichment<-romicsEnrichement(Subset18,cluster_filter = "hclust_clusters")
Subset18_enrichment<-Subset18_enrichment[Subset18_enrichment$pval<0.05&Subset18_enrichment$fold_change>1,]
```

## Subset19

```{r Subset19_Heatmap}
Subset19<-romicsANOVA(Subset19,factor = "disease_gen")
Subset19<-romicsZscores(Subset19)

print(paste0(sum(Subset19$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset19"))
pval<-data.frame(ids=rownames(Subset19$statistics), p=Subset19$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset19")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset19$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset19"))
pval<-data.frame(ids=rownames(Subset19$statistics), p=Subset19$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset19")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset19 were used to evaluate the best number of clusters to use for the Heatmap")
Subset19_z<-Subset19$statistics[Subset19$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset19$statistics))]

fviz_nbclust(Subset19_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset19_z, kmeans, method = "wss")+
geom_vline(xintercept = 2, linetype = 3)

romicsHeatmap(Subset19,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 2)
print("The clusters were then recorded in the statistics layer")
Subset19<-romicsVariableHclust(Subset19,ANOVA_filter = "padj",p=0.05,clusters = 2,plot = F)

Subset19_enrichment<-romicsEnrichement(Subset19,cluster_filter = "hclust_clusters")
Subset19_enrichment<-Subset19_enrichment[Subset19_enrichment$pval<0.05&Subset19_enrichment$fold_change>1,]
```

For this subset we performed a trajectory analysis using the package SLICER
```{r Subset19_Trajectory}
library("SLICER")
library("lle")

table_Subset19<-t(Subset19$data)
genes <- select_genes(table_Subset19)
k = select_k(table_Subset19[,genes], kmin=5)
traj_lle = lle(table_Subset19[,genes], m=2, k)$Y

print("the points of the trajectory are plotted")

traj_graph = conn_knn_graph(traj_lle,5)
ends = find_extreme_cells(traj_graph, traj_lle)
start = 1

print("The trajectory connections were plotted")

distances = graph_process_distance(traj_graph,embedding = traj_lle,start = 1)

traj_lle_for_graph = as.data.frame(traj_lle)
colnames(traj_lle_for_graph) = c("Manifold_dim_1","Manifold_dim_2")
traj_lle_for_graph$Manifold_dim_1 = as.numeric(traj_lle_for_graph$Manifold_dim_1)
traj_lle_for_graph$Manifold_dim_2 = as.numeric(traj_lle_for_graph$Manifold_dim_2)
traj_lle_for_graph<- cbind(ROI= rownames(table_Subset19), disease_gen= t(Subset19$metadata[5,]), traj_lle_for_graph, branches =  assign_branches(traj_graph,start = 1))

print("The plots is colored by condition")

ggplot(traj_lle_for_graph,aes(x=Manifold_dim_1,Manifold_dim_2,color=disease_gen))+ geom_point(size=1,)+ geom_text(label=traj_lle_for_graph$ROI)+theme_ROP()

DT::datatable(traj_lle_for_graph)
```
## Subset20

```{r Subset20_Heatmap}
Subset20<-romicsANOVA(Subset20,factor = "disease_gen")
Subset20<-romicsZscores(Subset20)

print(paste0(sum(Subset20$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset20"))
pval<-data.frame(ids=rownames(Subset20$statistics), p=Subset20$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset20")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset20$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset20"))
pval<-data.frame(ids=rownames(Subset20$statistics), p=Subset20$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset20")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset20 were used to evaluate the best number of clusters to use for the Heatmap")
Subset20_z<-Subset20$statistics[Subset20$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset20$statistics))]

fviz_nbclust(Subset20_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset20_z, kmeans, method = "wss")+
geom_vline(xintercept = 3, linetype = 3)

romicsHeatmap(Subset20,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 5)
print("The clusters were then recorded in the statistics layer")
Subset20<-romicsVariableHclust(Subset20,ANOVA_filter = "padj",p=0.05,clusters = 5,plot = F)

Subset20_enrichment<-romicsEnrichement(Subset20,cluster_filter = "hclust_clusters")
Subset20_enrichment<-Subset20_enrichment[Subset20_enrichment$pval<0.05&Subset20_enrichment$fold_change>1,]
Subset20_NAST<-Subset20
```

For this subset we performed a pseudo-time analysis using the Banff classification pathology scores
```{r Subset20_Trend}
Subset20<-romicsTrend(Subset20,factor = "Total_score_plus_one",log_factor = F,type = "linear",p = 0.05)
pFrequencyPlot(Subset20)
romicsTrendHeatmap(Subset20)
linear_increasing<-rownames(Subset20$statistics)[Subset20$statistics$best_fitted_trend=="linear_increasing"]
linear_decreasing<-rownames(Subset20$statistics)[Subset20$statistics$best_fitted_trend=="linear_decreasing"]
universe<-rownames(Subset20$statistics)
linear_increasing_enr_GO<-cbind(trend="linear_increasing",UniProt_GO_EASE(linear_increasing,universe))
linear_decreasing_enr_GO<-cbind(trend="linear_decreasing",UniProt_GO_EASE(linear_decreasing,universe))
linear_increasing_enr_KEGG<-cbind(trend="linear_increasing",UniProt_KEGG_EASE(linear_increasing,universe))
linear_decreasing_enr_KEGG<-cbind(trend="linear_decreasing",UniProt_KEGG_EASE(linear_decreasing,universe))

trend_enr<-rbind(linear_increasing_enr_GO,linear_decreasing_enr_GO,linear_increasing_enr_KEGG, linear_decreasing_enr_KEGG)
trend_enr<-trend_enr[trend_enr$pval<0.05 & trend_enr$fold_change>1,]

results<- romicsExportData(Subset20,statistics = T)
write.csv(results,file="./03_output_files/Subset20_with_trend.csv")
write.table(trend_enr, file="./03_output_files/Subset20_enrichments_trend.txt",sep="\t")
```


For this subset we performed a pseudo-time analysis using the Banff classification pathology scores
```{r Subset20_NAST_Trend}
Subset20_NAST<-romicsTrend(Subset20_NAST,factor = "Total_score_Nast_plus_one",log_factor = F,type = "linear",p = 0.05)
pFrequencyPlot(Subset20_NAST)
romicsTrendHeatmap(Subset20_NAST)
linear_increasing<-rownames(Subset20_NAST$statistics)[Subset20_NAST$statistics$best_fitted_trend=="linear_increasing"]
linear_decreasing<-rownames(Subset20_NAST$statistics)[Subset20_NAST$statistics$best_fitted_trend=="linear_decreasing"]
universe<-rownames(Subset20_NAST$statistics)
linear_increasing_enr_GO<-cbind(trend="linear_increasing",UniProt_GO_EASE(linear_increasing,universe))
linear_decreasing_enr_GO<-cbind(trend="linear_decreasing",UniProt_GO_EASE(linear_decreasing,universe))
linear_increasing_enr_KEGG<-cbind(trend="linear_increasing",UniProt_KEGG_EASE(linear_increasing,universe))
linear_decreasing_enr_KEGG<-cbind(trend="linear_decreasing",UniProt_KEGG_EASE(linear_decreasing,universe))

trend_enr<-rbind(linear_increasing_enr_GO,linear_decreasing_enr_GO,linear_increasing_enr_KEGG, linear_decreasing_enr_KEGG)
trend_enr<-trend_enr[trend_enr$pval<0.05 & trend_enr$fold_change>1,]

results<- romicsExportData(Subset20_NAST,statistics = T)
write.csv(results,file="./03_output_files/Subset20_with_trend_NAST.csv")
write.table(trend_enr, file="./03_output_files/Subset20_enrichments_trend_NAST.txt",sep="\t")
```

For this subset we performed a trajectory analysis using the package SLICER
```{r Subset20_Trajectory}
library("SLICER")
library("lle")

table_Subset20<-t(Subset20$data)
genes <- select_genes(table_Subset20)
k = select_k(table_Subset20[,genes], kmin=5)
traj_lle = lle(table_Subset20[,genes], m=2, k)$Y

print("the points of the trajectory are plotted")

traj_graph = conn_knn_graph(traj_lle,5)
ends = find_extreme_cells(traj_graph, traj_lle)
start = 1

print("The trajectory connections were plotted")

distances = graph_process_distance(traj_graph,embedding = traj_lle,start = 1)

traj_lle_for_graph = as.data.frame(traj_lle)
colnames(traj_lle_for_graph) = c("Manifold_dim_1","Manifold_dim_2")
traj_lle_for_graph$Manifold_dim_1 = as.numeric(traj_lle_for_graph$Manifold_dim_1)
traj_lle_for_graph$Manifold_dim_2 = as.numeric(traj_lle_for_graph$Manifold_dim_2)
traj_lle_for_graph<- cbind(ROI= rownames(table_Subset20), disease_gen= t(Subset20$metadata[5,]), traj_lle_for_graph, branches =  assign_branches(traj_graph,start = 1))

print("The plots is colored by condition")

ggplot(traj_lle_for_graph,aes(x=Manifold_dim_1,Manifold_dim_2,color=disease_gen))+ geom_point(size=1,)+ geom_text(label=traj_lle_for_graph$ROI)+theme_ROP()

DT::datatable(traj_lle_for_graph)
```

## Subset21

```{r Subset21_Heatmap}
Subset21<-romicsANOVA(Subset21,factor = "disease_gen")
Subset21<-romicsZscores(Subset21)

print(paste0(sum(Subset21$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset21"))
pval<-data.frame(ids=rownames(Subset21$statistics), p=Subset21$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset21")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset21$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset21"))
pval<-data.frame(ids=rownames(Subset21$statistics), p=Subset21$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset21")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset21 were used to evaluate the best number of clusters to use for the Heatmap")
Subset21_z<-Subset21$statistics[Subset21$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset21$statistics))]

fviz_nbclust(Subset21_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset21_z, kmeans, method = "wss")+
geom_vline(xintercept = 5, linetype = 3)

romicsHeatmap(Subset21,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 5)
print("The clusters were then recorded in the statistics layer")
Subset21<-romicsVariableHclust(Subset21,ANOVA_filter = "padj",p=0.05,clusters = 5,plot = F)

Subset21_enrichment<-romicsEnrichement(Subset21,cluster_filter = "hclust_clusters")
Subset21_enrichment<-Subset21_enrichment[Subset21_enrichment$pval<0.05&Subset21_enrichment$fold_change>1,]
```

For this subset we performed a trajectory analysis using the package SLICER
```{r Subset21_Trajectory}
library("SLICER")
library("lle")

table_Subset21<-t(Subset21$data)
genes <- select_genes(table_Subset21)
k = select_k(table_Subset21[,genes], kmin=5)
traj_lle = lle(table_Subset21[,genes], m=2, k)$Y

print("the points of the trajectory are plotted")

traj_graph = conn_knn_graph(traj_lle,5)
ends = find_extreme_cells(traj_graph, traj_lle)
start = 1

print("The trajectory connections were plotted")

distances = graph_process_distance(traj_graph,embedding = traj_lle,start = 1)

traj_lle_for_graph = as.data.frame(traj_lle)
colnames(traj_lle_for_graph) = c("Manifold_dim_1","Manifold_dim_2")
traj_lle_for_graph$Manifold_dim_1 = as.numeric(traj_lle_for_graph$Manifold_dim_1)
traj_lle_for_graph$Manifold_dim_2 = as.numeric(traj_lle_for_graph$Manifold_dim_2)
traj_lle_for_graph<- cbind(ROI= rownames(table_Subset21), disease_gen= t(Subset21$metadata[5,]), traj_lle_for_graph, branches =  assign_branches(traj_graph,start = 1))

print("The plots is colored by condition")

ggplot(traj_lle_for_graph,aes(x=Manifold_dim_1,Manifold_dim_2,color=disease_gen))+ geom_point(size=1,)+ geom_text(label=traj_lle_for_graph$ROI)+theme_ROP()

DT::datatable(traj_lle_for_graph)
```


## Subset22

```{r Subset22_Heatmap}
Subset22<-romicsANOVA(Subset22,factor = "disease_gen")
Subset22<-romicsZscores(Subset22)

print(paste0(sum(Subset22$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset22"))
pval<-data.frame(ids=rownames(Subset22$statistics), p=Subset22$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset22")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset22$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset22"))
pval<-data.frame(ids=rownames(Subset22$statistics), p=Subset22$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset22")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset22 were used to evaluate the best number of clusters to use for the Heatmap")
Subset22_z<-Subset22$statistics[Subset22$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset22$statistics))]

fviz_nbclust(Subset22_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset22_z, kmeans, method = "wss")+
geom_vline(xintercept = 5, linetype = 3)

romicsHeatmap(Subset22,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 11)
print("The clusters were then recorded in the statistics layer")
Subset22<-romicsVariableHclust(Subset22,ANOVA_filter = "padj",p=0.05,clusters = 11,plot = F)

Subset22_enrichment<-romicsEnrichement(Subset22,cluster_filter = "hclust_clusters")
Subset22_enrichment<-Subset22_enrichment[Subset22_enrichment$pval<0.05&Subset22_enrichment$fold_change>1,]
Subset22_NAST<-Subset22
```

For this subset we performed a pseudo-time analysis using the Banff classification pathology scores
```{r Subset22_Trend}
Subset22<-romicsTrend(Subset22,factor = "Total_score_plus_one",log_factor = F,type = "linear",p = 0.05)
pFrequencyPlot(Subset22)
romicsTrendHeatmap(Subset22)
linear_increasing<-rownames(Subset22$statistics)[Subset22$statistics$best_fitted_trend=="linear_increasing"]
linear_decreasing<-rownames(Subset22$statistics)[Subset22$statistics$best_fitted_trend=="linear_decreasing"]
universe<-rownames(Subset22$statistics)
linear_increasing_enr_GO<-cbind(trend="linear_increasing",UniProt_GO_EASE(linear_increasing,universe))
linear_decreasing_enr_GO<-cbind(trend="linear_decreasing",UniProt_GO_EASE(linear_decreasing,universe))
linear_increasing_enr_KEGG<-cbind(trend="linear_increasing",UniProt_KEGG_EASE(linear_increasing,universe))
linear_decreasing_enr_KEGG<-cbind(trend="linear_decreasing",UniProt_KEGG_EASE(linear_decreasing,universe))

trend_enr<-rbind(linear_increasing_enr_GO,linear_decreasing_enr_GO,linear_increasing_enr_KEGG, linear_decreasing_enr_KEGG)
trend_enr<-trend_enr[trend_enr$pval<0.05 & trend_enr$fold_change>1,]

results<- romicsExportData(Subset22,statistics = T)
write.csv(results,file="./03_output_files/Subset22_with_trend.csv")
write.table(trend_enr, file="./03_output_files/Subset22_enrichments_trend.txt",sep="\t")
```

For this subset we performed a pseudo-time analysis using the Banff classification pathology scores from NAST (2nd pathologist)
```{r Subset22_NAST_Pseudotime}
Subset22_NAST<-romicsTrend(Subset22_NAST,factor = "Total_score_Nast_plus_one",log_factor = F,type = "linear",p = 0.05)
pFrequencyPlot(Subset22_NAST)
romicsTrendHeatmap(Subset22_NAST)
linear_increasing<-rownames(Subset22_NAST$statistics)[Subset22_NAST$statistics$best_fitted_trend=="linear_increasing"]
linear_decreasing<-rownames(Subset22_NAST$statistics)[Subset22_NAST$statistics$best_fitted_trend=="linear_decreasing"]
universe<-rownames(Subset22_NAST$statistics)
linear_increasing_enr_GO<-cbind(trend="linear_increasing",UniProt_GO_EASE(linear_increasing,universe))
linear_decreasing_enr_GO<-cbind(trend="linear_decreasing",UniProt_GO_EASE(linear_decreasing,universe))
linear_increasing_enr_KEGG<-cbind(trend="linear_increasing",UniProt_KEGG_EASE(linear_increasing,universe))
linear_decreasing_enr_KEGG<-cbind(trend="linear_decreasing",UniProt_KEGG_EASE(linear_decreasing,universe))

trend_enr<-rbind(linear_increasing_enr_GO,linear_decreasing_enr_GO,linear_increasing_enr_KEGG, linear_decreasing_enr_KEGG)
trend_enr<-trend_enr[trend_enr$pval<0.05 & trend_enr$fold_change>1,]

results<- romicsExportData(Subset22_NAST,statistics = T)
write.csv(results,file="./03_output_files/Subset22_with_trend_NAST.csv")
write.table(trend_enr, file="./03_output_files/Subset22_enrichments_trend_NAST.txt",sep="\t")
```

For this subset we performed a trajectory analysis using the package SLICER
```{r Subset22_Trajectory}
library("SLICER")
library("lle")

table_Subset22<-t(Subset22$data)
genes <- select_genes(table_Subset22)
k = select_k(table_Subset22[,genes], kmin=5)
traj_lle = lle(table_Subset22[,genes], m=2, k)$Y

print("the points of the trajectory are plotted")

traj_graph = conn_knn_graph(traj_lle,5)
ends = find_extreme_cells(traj_graph, traj_lle)
start = 1

print("The trajectory connections were plotted")

distances = graph_process_distance(traj_graph,embedding = traj_lle,start = 1)

traj_lle_for_graph = as.data.frame(traj_lle)
colnames(traj_lle_for_graph) = c("Manifold_dim_1","Manifold_dim_2")
traj_lle_for_graph$Manifold_dim_1 = as.numeric(traj_lle_for_graph$Manifold_dim_1)
traj_lle_for_graph$Manifold_dim_2 = as.numeric(traj_lle_for_graph$Manifold_dim_2)
traj_lle_for_graph<- cbind(ROI= rownames(table_Subset22), disease_gen= t(Subset22$metadata[5,]), traj_lle_for_graph, branches =  assign_branches(traj_graph,start = 1))

print("The plots is colored by condition")

ggplot(traj_lle_for_graph,aes(x=Manifold_dim_1,Manifold_dim_2,color=disease_gen))+ geom_point(size=1,)+ geom_text(label=traj_lle_for_graph$ROI)+theme_ROP()

DT::datatable(traj_lle_for_graph)
```


## Subset23

```{r Subset23_Heatmap}
Subset23<-romicsANOVA(Subset23,factor = "disease_gen")
Subset23<-romicsZscores(Subset23)

print(paste0(sum(Subset23$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset23"))
pval<-data.frame(ids=rownames(Subset23$statistics), p=Subset23$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset23")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset23$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset23"))
pval<-data.frame(ids=rownames(Subset23$statistics), p=Subset23$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset23")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset23 were used to evaluate the best number of clusters to use for the Heatmap")
Subset23_z<-Subset23$statistics[Subset23$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset23$statistics))]


fviz_nbclust(Subset23_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset23_z, kmeans, method = "wss")+
geom_vline(xintercept = 4, linetype = 3)

romicsHeatmap(Subset23,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 4)
print("The clusters were then recorded in the statistics layer")
Subset23<-romicsVariableHclust(Subset23,ANOVA_filter = "padj",p=0.05,clusters = 4,plot = F)

Subset23_enrichment<-romicsEnrichement(Subset23,cluster_filter = "hclust_clusters")
Subset23_enrichment<-Subset23_enrichment[Subset23_enrichment$pval<0.05&Subset23_enrichment$fold_change>1,]
```

## Subset24

```{r Subset24_Heatmap}
Subset24<-romicsANOVA(Subset24,factor = "disease_gen")
Subset24<-romicsZscores(Subset24)

print(paste0(sum(Subset24$statistics$ANOVA_p<0.05)," transcripts had an ANOVA p<0.05 in the Subset24"))
pval<-data.frame(ids=rownames(Subset24$statistics), p=Subset24$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA p frequency plot Subset24")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print(paste0(sum(Subset24$statistics$ANOVA_padj<0.05)," transcripts had an ANOVA padj<0.05 in the Subset24"))
pval<-data.frame(ids=rownames(Subset24$statistics), p=Subset24$statistics$ANOVA_p)
ggplot(pval, aes(p)) + geom_histogram(binwidth = 0.01)+theme_ROP()+ggtitle("ANOVA padj frequency plot Subset24")+geom_vline(xintercept=0.05,linetype="dashed", color = "red")

print("The Z-scores of the transcript modulated in Subset24 were used to evaluate the best number of clusters to use for the Heatmap")
Subset24_z<-Subset24$statistics[Subset24$statistics$ANOVA_padj<0.05,grepl("Z_scores",colnames(Subset24$statistics))]

fviz_nbclust(Subset24_z, kmeans, method = "silhouette", k.max = 15) + theme_bw() + ggtitle("The Silhouette Plot")
fviz_nbclust(Subset24_z, kmeans, method = "wss")+
geom_vline(xintercept = 2, linetype = 3)

romicsHeatmap(Subset24,ANOVA_filter = "padj",p=0.05,variable_hclust_number = 2)
print("The clusters were then recorded in the statistics layer")
Subset24<-romicsVariableHclust(Subset24,ANOVA_filter = "padj",p=0.05,clusters = 5,plot = F)

Subset24_enrichment<-romicsEnrichement(Subset24,cluster_filter = "hclust_clusters")
Subset24_enrichment<-Subset24_enrichment[Subset24_enrichment$pval<0.05&Subset24_enrichment$fold_change>1,]

```

# T.test 
For all the Subsets with at least 2 groups the paired t.tests were performed and volcano plots were generated for each comparison

```{r Subset03_ttest}
print("Subset03")
Subset03<-romicsTtest(Subset03)
romicsVolcano(Subset03)
```

```{r Subset04_ttest}
print("Subset04")
Subset04<-romicsTtest(Subset04)
romicsVolcano(Subset04)
```

```{r Subset05_ttest}
print("Subset05")
Subset05<-romicsTtest(Subset05)
romicsVolcano(Subset05)
```

```{r Subset06_ttest}
print("Subset06")
Subset06<-romicsTtest(Subset06,reverse_order = T)
romicsVolcano(Subset06,p_type = "padj",plot = 2,min_fold_change = 0)
romicsVolcano(Subset06,p_type = "padj",plot = 3,min_fold_change = 0)

low_in_alport_f<-rownames(Subset06$statistics[Subset06$statistics$`Alport(F)(Young)_Glomeruli_vs_Normal (Youth)_Glomeruli_Ttest_padj`<0.05 & Subset06$statistics$`log(Alport(F)(Young)_Glomeruli/Normal (Youth)_Glomeruli)`< 0,])
high_in_alport_f<-rownames(Subset06$statistics[Subset06$statistics$`Alport(F)(Young)_Glomeruli_vs_Normal (Youth)_Glomeruli_Ttest_padj`<0.05 & Subset06$statistics$`log(Alport(F)(Young)_Glomeruli/Normal (Youth)_Glomeruli)`> 0,])
univ <- rownames(Subset06$statistics)

low_in_alport_m<-rownames(Subset06$statistics[Subset06$statistics$`Alport(M)_Glomeruli_vs_Normal (Youth)_Glomeruli_Ttest_padj`<0.05 & Subset06$statistics$`log(Alport(M)_Glomeruli/Normal (Youth)_Glomeruli)`< 0,])
high_in_alport_m<-rownames(Subset06$statistics[Subset06$statistics$`Alport(M)_Glomeruli_vs_Normal (Youth)_Glomeruli_Ttest_padj`<0.05 & Subset06$statistics$`log(Alport(M)_Glomeruli/Normal (Youth)_Glomeruli)`> 0,])

univ <- rownames(Subset06$statistics)

low_in_alport_f_enr_GO<-cbind(enriched_for="low_in_alport_f",type="GO",UniProt_GO_Fisher(low_in_alport_f,univ))
low_in_alport_f_enr_KEGG<-cbind(enriched_for="low_in_alport_f",type="KEGG",UniProt_KEGG_Fisher(low_in_alport_f,univ))
low_in_alport_m_enr_GO<-cbind(enriched_for="low_in_alport_m",type="GO",UniProt_GO_Fisher(low_in_alport_m,univ))
low_in_alport_m_enr_KEGG<-cbind(enriched_for="low_in_alport_m",type="KEGG",UniProt_KEGG_Fisher(low_in_alport_m,univ))
high_in_alport_f_enr_GO<-cbind(enriched_for="high_in_alport_f",type="GO",UniProt_GO_Fisher(high_in_alport_f,univ))
high_in_alport_f_enr_KEGG<-cbind(enriched_for="high_in_alport_f",type="KEGG",UniProt_KEGG_Fisher(high_in_alport_f,univ))
high_in_alport_m_enr_GO<-cbind(enriched_for="high_in_alport_m",type="GO",UniProt_GO_Fisher(high_in_alport_m,univ))
high_in_alport_m_enr_KEGG<-cbind(enriched_for="high_in_alport_m",type="KEGG",UniProt_KEGG_Fisher(high_in_alport_m,univ))

enr_ttests_Subset06<- rbind(low_in_alport_f_enr_GO,low_in_alport_f_enr_KEGG,low_in_alport_m_enr_GO,low_in_alport_m_enr_KEGG,high_in_alport_f_enr_GO,high_in_alport_f_enr_KEGG,high_in_alport_m_enr_GO,high_in_alport_m_enr_KEGG)

enr_ttests_Subset06<- enr_ttests_Subset06[enr_ttests_Subset06$pval<0.05 &enr_ttests_Subset06$fold_change>1,]
write.table(enr_ttests_Subset06,file = "03_output_files/enr_ttests_Subset06.txt",sep="\t",row.names = F)
```

```{r Subset08_ttest}
print("Subset08")
Subset08<-romicsTtest(Subset08,reverse_order = T)
romicsVolcano(Subset08,min_fold_change = 0)

low_in_alport_f<-rownames(Subset08$statistics[Subset08$statistics$`Alport(F)(Adult)_Glomeruli_vs_Normal (Adult)_Glomeruli_Ttest_padj`<0.05 & Subset08$statistics$`log(Alport(F)(Adult)_Glomeruli/Normal (Adult)_Glomeruli)`< 0,])
high_in_alport_f<-rownames(Subset08$statistics[Subset08$statistics$`Alport(F)(Adult)_Glomeruli_vs_Normal (Adult)_Glomeruli_Ttest_padj`<0.05 & Subset08$statistics$`log(Alport(F)(Adult)_Glomeruli/Normal (Adult)_Glomeruli)`> 0,])
univ <- rownames(Subset08$statistics)

low_in_alport_f_enr_GO<-cbind(enriched_for="low_in_alport_f",type="GO",UniProt_GO_Fisher(low_in_alport_f,univ))
low_in_alport_f_enr_KEGG<-cbind(enriched_for="low_in_alport_f",type="KEGG",UniProt_KEGG_Fisher(low_in_alport_f,univ))
high_in_alport_f_enr_GO<-cbind(enriched_for="high_in_alport_f",type="GO",UniProt_GO_Fisher(high_in_alport_f,univ))
high_in_alport_f_enr_KEGG<-cbind(enriched_for="high_in_alport_f",type="KEGG",UniProt_KEGG_Fisher(high_in_alport_f,univ))

enr_ttests_Subset08<- rbind(low_in_alport_f_enr_GO,low_in_alport_f_enr_KEGG,high_in_alport_f_enr_GO,high_in_alport_f_enr_KEGG)

enr_ttests_Subset08<- enr_ttests_Subset08[enr_ttests_Subset08$pval<0.05 &enr_ttests_Subset08$fold_change>1,]
write.table(enr_ttests_Subset08,file = "03_output_files/enr_ttests_Subset08.txt",sep="\t",row.names = F)
```

```{r Subset11_ttest}
print("Subset11")
Subset11<-romicsTtest(Subset11)
```

```{r Subset12_ttest}
print("Subset12")
Subset12<-romicsTtest(Subset12)
```

```{r Subset13_ttest}
print("Subset13")
Subset13<-romicsTtest(Subset13)
```

```{r Subset14_ttest}
print("Subset14_2c")
Subset14_2c<-romicsTtest(Subset14_2c,reverse_order = T)

print("Subset14_3c")
Subset14_3c<-romicsTtest(Subset14_3c,reverse_order = T)

romicsVolcano(Subset14_3c,p_type = "padj",plot = 2,min_fold_change = 0)
romicsVolcano(Subset14_3c,p_type = "padj",plot = 3,min_fold_change = 0)

low_in_FSGS_f<-rownames(Subset14_3c$statistics[Subset14_3c$statistics$`FSGS(F)_Glomeruli_vs_Normal (Adult)_Glomeruli_Ttest_padj`<0.05 & Subset14_3c$statistics$`log(FSGS(F)_Glomeruli/Normal (Adult)_Glomeruli)`< 0,])
high_in_FSGS_f<-rownames(Subset14_3c$statistics[Subset14_3c$statistics$`FSGS(F)_Glomeruli_vs_Normal (Adult)_Glomeruli_Ttest_padj`<0.05 & Subset14_3c$statistics$`log(FSGS(F)_Glomeruli/Normal (Adult)_Glomeruli)`> 0,])
univ <- rownames(Subset14_3c$statistics)
low_in_FSGS_m<-rownames(Subset14_3c$statistics[Subset14_3c$statistics$`FSGS (M)_Glomeruli_vs_Normal (Adult)_Glomeruli_Ttest_padj`<0.05 & Subset14_3c$statistics$`log(FSGS (M)_Glomeruli/Normal (Adult)_Glomeruli)`< 0,])
high_in_FSGS_m<-rownames(Subset14_3c$statistics[Subset14_3c$statistics$`FSGS (M)_Glomeruli_vs_Normal (Adult)_Glomeruli_Ttest_padj`<0.05 & Subset14_3c$statistics$`log(FSGS (M)_Glomeruli/Normal (Adult)_Glomeruli)`> 0,])
univ <- rownames(Subset14_3c$statistics)

low_in_FSGS_f_enr_GO<-cbind(enriched_for="low_in_FSGS_f",type="GO",UniProt_GO_Fisher(low_in_FSGS_f,univ))
low_in_FSGS_f_enr_KEGG<-cbind(enriched_for="low_in_FSGS_f",type="KEGG",UniProt_KEGG_Fisher(low_in_FSGS_f,univ))
low_in_FSGS_m_enr_GO<-cbind(enriched_for="low_in_FSGS_m",type="GO",UniProt_GO_Fisher(low_in_FSGS_m,univ))
low_in_FSGS_m_enr_KEGG<-cbind(enriched_for="low_in_FSGS_m",type="KEGG",UniProt_KEGG_Fisher(low_in_FSGS_m,univ))
high_in_FSGS_f_enr_GO<-cbind(enriched_for="high_in_FSGS_f",type="GO",UniProt_GO_Fisher(high_in_FSGS_f,univ))
high_in_FSGS_f_enr_KEGG<-cbind(enriched_for="high_in_FSGS_f",type="KEGG",UniProt_KEGG_Fisher(high_in_FSGS_f,univ))
high_in_FSGS_m_enr_GO<-cbind(enriched_for="high_in_FSGS_m",type="GO",UniProt_GO_Fisher(high_in_FSGS_m,univ))
high_in_FSGS_m_enr_KEGG<-cbind(enriched_for="high_in_FSGS_m",type="KEGG",UniProt_KEGG_Fisher(high_in_FSGS_m,univ))

enr_ttests_Subset14_3c<- rbind(low_in_FSGS_f_enr_GO,low_in_FSGS_f_enr_KEGG,low_in_FSGS_m_enr_GO,low_in_FSGS_m_enr_KEGG,high_in_FSGS_f_enr_GO,high_in_FSGS_f_enr_KEGG,high_in_FSGS_m_enr_GO,high_in_FSGS_m_enr_KEGG)

enr_ttests_Subset14_3c<- enr_ttests_Subset14_3c[enr_ttests_Subset14_3c$pval<0.05 &enr_ttests_Subset14_3c$fold_change>1,]
write.table(enr_ttests_Subset14_3c,file = "03_output_files/enr_ttests_Subset14_3c.txt",sep="\t",row.names = F)
```

```{r Subset17_ttest}
print("Subset17_2c")
Subset17_2c<-romicsTtest(Subset17_2c)
print("Subset17_3c")
Subset17_3c<-romicsTtest(Subset17_3c)
```

```{r Subset18_ttest}
print("Subset18")
Subset18<-romicsTtest(Subset18)
```

```{r Subset19_ttest}
print("Subset19")
Subset19<-romicsTtest(Subset19)
```

```{r Subset20_ttest}
print("Subset20")
Subset20<-romicsTtest(Subset20)
```

```{r Subset21_ttest}
print("Subset21")
Subset21<-romicsTtest(Subset21)
```

```{r Subset22_ttest}
print("Subset22")
Subset22<-romicsTtest(Subset22)
```

```{r Subset23_ttest}
print("Subset23")
Subset23<-romicsTtest(Subset23)
```

```{r Subset24_ttest}
print("Subset24")
Subset24<-romicsTtest(Subset24)
```

#data export
Finally the data was exported
```{r}
romics_rna_gt_exp<-romicsExportData(romics_rna_gt,statistics = T)
romics_rna_gt_exp<-cbind(IDs,romics_rna_gt_exp[1:ncol(romics_rna_gt_exp)])
write.csv(romics_rna_gt_exp,file = "./03_output_files/Glomeruli_and_tubule_results.csv")

Subset01_exp<-romicsExportData(Subset01,statistics = T)
Subset01_exp<-cbind(IDs,Subset01_exp[1:ncol(Subset01_exp)])
write.csv(Subset01_exp,file = "./03_output_files/Subset01_results.csv")

Subset02_exp<-romicsExportData(Subset02,statistics = T)
Subset02_exp<-cbind(IDs,Subset02_exp[1:ncol(Subset02_exp)])
write.csv(Subset02_exp,file = "./03_output_files/Subset02_results.csv")

Subset03_exp<-romicsExportData(Subset03,statistics = T)
Subset03_exp<-cbind(IDs,Subset03_exp[1:ncol(Subset03_exp)])
write.csv(Subset03_exp,file = "./03_output_files/Subset03_results.csv")

Subset04_exp<-romicsExportData(Subset04,statistics = T)
Subset04_exp<-cbind(IDs,Subset04_exp[1:ncol(Subset04_exp)])
write.csv(Subset04_exp,file = "./03_output_files/Subset04_results.csv")

Subset05_exp<-romicsExportData(Subset05,statistics = T)
Subset05_exp<-cbind(IDs,Subset05_exp[1:ncol(Subset05_exp)])
write.csv(Subset05_exp,file = "./03_output_files/Subset05_results.csv")

Subset06_exp<-romicsExportData(Subset06,statistics = T)
Subset06_exp<-cbind(IDs,Subset06_exp[1:ncol(Subset06_exp)])
write.csv(Subset06_exp,file = "./03_output_files/Subset06_results.csv")

Subset07_2c_exp<-romicsExportData(Subset07_2c,statistics = T)
Subset07_2c_exp<-cbind(IDs,Subset07_2c_exp[1:ncol(Subset07_2c_exp)])
write.csv(Subset07_2c_exp,file = "./03_output_files/Subset07_2c_results.csv")

Subset07_4c_exp<-romicsExportData(Subset07_4c,statistics = T)
Subset07_4c_exp<-cbind(IDs,Subset07_4c_exp[1:ncol(Subset07_4c_exp)])
write.csv(Subset07_4c_exp,file = "./03_output_files/Subset07_4c_results.csv")

Subset08_exp<-romicsExportData(Subset08,statistics = T)
Subset08_exp<-cbind(IDs,Subset08_exp[1:ncol(Subset08_exp)])
write.csv(Subset08_exp,file = "./03_output_files/Subset08_results.csv")

Subset09_3c_exp<-romicsExportData(Subset09_3c,statistics = T)
Subset09_3c_exp<-cbind(IDs,Subset09_3c_exp[1:ncol(Subset09_3c_exp)])
write.csv(Subset09_3c_exp,file = "./03_output_files/Subset09_3c_results.csv")

Subset09_4c_exp<-romicsExportData(Subset09_4c,statistics = T)
Subset09_4c_exp<-cbind(IDs,Subset09_4c_exp[1:ncol(Subset09_4c_exp)])
write.csv(Subset09_4c_exp,file = "./03_output_files/Subset09_4c_results.csv")


Subset10_exp<-romicsExportData(Subset10,statistics = T)
Subset10_exp<-cbind(IDs,Subset10_exp[1:ncol(Subset10_exp)])
write.csv(Subset10_exp,file = "./03_output_files/Subset10_results.csv")

Subset11_exp<-romicsExportData(Subset11,statistics = T)
Subset11_exp<-cbind(IDs,Subset11_exp[1:ncol(Subset11_exp)])
write.csv(Subset11_exp,file = "./03_output_files/Subset11_results.csv")

Subset12_exp<-romicsExportData(Subset12,statistics = T)
Subset12_exp<-cbind(IDs,Subset12_exp[1:ncol(Subset12_exp)])
write.csv(Subset12_exp,file = "./03_output_files/Subset12_results.csv")

Subset13_exp<-romicsExportData(Subset13,statistics = T)
Subset13_exp<-cbind(IDs,Subset13_exp[1:ncol(Subset13_exp)])
write.csv(Subset13_exp,file = "./03_output_files/Subset13_results.csv")

Subset14_2c_exp<-romicsExportData(Subset14_2c,statistics = T)
Subset14_2c_exp<-cbind(IDs,Subset14_2c_exp[1:ncol(Subset14_2c_exp)])
write.csv(Subset14_2c_exp,file = "./03_output_files/Subset14_2c_results.csv")

Subset14_3c_exp<-romicsExportData(Subset14_3c,statistics = T)
Subset14_3c_exp<-cbind(IDs,Subset14_3c_exp[1:ncol(Subset14_3c_exp)])
write.csv(Subset14_3c_exp,file = "./03_output_files/Subset14_3c_results.csv")

Subset15_exp<-romicsExportData(Subset15,statistics = T)
Subset15_exp<-cbind(IDs,Subset15_exp[1:ncol(Subset15_exp)])
write.csv(Subset15_exp,file = "./03_output_files/Subset15_results.csv")

Subset16_exp<-romicsExportData(Subset16,statistics = T)
Subset16_exp<-cbind(IDs,Subset16_exp[1:ncol(Subset16_exp)])
write.csv(Subset16_exp,file = "./03_output_files/Subset16_results.csv")

Subset17_2c_exp<-romicsExportData(Subset17_2c,statistics = T)
Subset17_2c_exp<-cbind(IDs,Subset17_2c_exp[1:ncol(Subset17_2c_exp)])
write.csv(Subset17_2c_exp,file = "./03_output_files/Subset17_2c_results.csv")

Subset17_3c_exp<-romicsExportData(Subset17_3c,statistics = T)
Subset17_3c_exp<-cbind(IDs,Subset17_3c_exp[1:ncol(Subset17_3c_exp)])
write.csv(Subset17_3c_exp,file = "./03_output_files/Subset17_3c_results.csv")

Subset18_exp<-romicsExportData(Subset18,statistics = T)
Subset18_exp<-cbind(IDs,Subset18_exp[1:ncol(Subset18_exp)])
write.csv(Subset18_exp,file = "./03_output_files/Subset18_results.csv")

Subset19_exp<-romicsExportData(Subset19,statistics = T)
Subset19_exp<-cbind(IDs,Subset19_exp[1:ncol(Subset19_exp)])
write.csv(Subset19_exp,file = "./03_output_files/Subset19_results.csv")

Subset20_exp<-romicsExportData(Subset20,statistics = T)
Subset20_exp<-cbind(IDs,Subset20_exp[1:ncol(Subset20_exp)])
write.csv(Subset20_exp,file = "./03_output_files/Subset20_results.csv")

Subset21_exp<-romicsExportData(Subset21,statistics = T)
Subset21_exp<-cbind(IDs,Subset21_exp[1:ncol(Subset21_exp)])
write.csv(Subset21_exp,file = "./03_output_files/Subset21_results.csv")

Subset22_exp<-romicsExportData(Subset22,statistics = T)
Subset22_exp<-cbind(IDs,Subset22_exp[1:ncol(Subset22_exp)])
write.csv(Subset22_exp,file = "./03_output_files/Subset22_results.csv")

Subset23_exp<-romicsExportData(Subset23,statistics = T)
Subset23_exp<-cbind(IDs,Subset23_exp[1:ncol(Subset23_exp)])
write.csv(Subset23_exp,file = "./03_output_files/Subset23_results.csv")

Subset24_exp<-romicsExportData(Subset24,statistics = T)
Subset24_exp<-cbind(IDs,Subset24_exp[1:ncol(Subset24_exp)])
write.csv(Subset24_exp,file = "./03_output_files/Subset24_results.csv")

write.table(enrichment_rna_gt, file="./03_output_files/enrichment_rna_gt.txt",sep="\t",row.names = F)
write.table(Subset01_enrichment, file="./03_output_files/Subset01_enrichment.txt",sep="\t",row.names = F)
write.table(Subset02_enrichment, file="./03_output_files/Subset02_enrichment.txt",sep="\t",row.names = F)
write.table(Subset03_enrichment, file="./03_output_files/Subset03_enrichment.txt",sep="\t",row.names = F)
write.table(Subset04_enrichment, file="./03_output_files/Subset04_enrichment.txt",sep="\t",row.names = F)
write.table(Subset05_enrichment, file="./03_output_files/Subset05_enrichment.txt",sep="\t",row.names = F)
write.table(Subset06_enrichment, file="./03_output_files/Subset06_enrichment.txt",sep="\t",row.names = F)
write.table(Subset07_2c_enrichment, file="./03_output_files/Subset07_2c_enrichment.txt",sep="\t",row.names = F)
write.table(Subset07_4c_enrichment, file="./03_output_files/Subset07_4c_enrichment.txt",sep="\t",row.names = F)
write.table(Subset08_enrichment, file="./03_output_files/Subset08_enrichment.txt",sep="\t",row.names = F)
write.table(Subset09_3c_enrichment, file="./03_output_files/Subset09_3c_enrichment.txt",sep="\t",row.names = F)
write.table(Subset09_4c_enrichment, file="./03_output_files/Subset09_4c_enrichment.txt",sep="\t",row.names = F)
write.table(Subset10_enrichment, file="./03_output_files/Subset10_enrichment.txt",sep="\t",row.names = F)
write.table(Subset11_enrichment, file="./03_output_files/Subset11_enrichment.txt",sep="\t",row.names = F)
write.table(Subset12_enrichment, file="./03_output_files/Subset12_enrichment.txt",sep="\t",row.names = F)
#write.table(Subset13_enrichment, file="./03_output_files/Subset13_enrichment.txt",sep="\t",row.names = F)
write.table(Subset14_2c_enrichment, file="./03_output_files/Subset14_2c_enrichment.txt",sep="\t",row.names = F)
write.table(Subset14_3c_enrichment, file="./03_output_files/Subset14_3c_enrichment.txt",sep="\t",row.names = F)
write.table(Subset15_enrichment, file="./03_output_files/Subset15_enrichment.txt",sep="\t",row.names = F)
write.table(Subset16_enrichment, file="./03_output_files/Subset16_enrichment.txt",sep="\t",row.names = F)
write.table(Subset17_2c_enrichment, file="./03_output_files/Subset17_2c_enrichment.txt",sep="\t",row.names = F)
write.table(Subset17_3c_enrichment, file="./03_output_files/Subset17_3c_enrichment.txt",sep="\t",row.names = F)
write.table(Subset18_enrichment, file="./03_output_files/Subset18_enrichment.txt",sep="\t",row.names = F)
write.table(Subset19_enrichment, file="./03_output_files/Subset19_enrichment.txt",sep="\t",row.names = F)
write.table(Subset20_enrichment, file="./03_output_files/Subset20_enrichment.txt",sep="\t",row.names = F)
write.table(Subset21_enrichment, file="./03_output_files/Subset21_enrichment.txt",sep="\t",row.names = F)
write.table(Subset22_enrichment, file="./03_output_files/Subset22_enrichment.txt",sep="\t",row.names = F)
write.table(Subset23_enrichment, file="./03_output_files/Subset23_enrichment.txt",sep="\t",row.names = F)
write.table(Subset24_enrichment, file="./03_output_files/Subset24_enrichment.txt",sep="\t",row.names = F)
```


# Correlation with Glomerular disease related proteins

For a set of 5 genes related to glomerular proteins (WT1, EHD3, NPHS1, NPHS2, CD2AP, PODXL,and PDGFRB) we performed correlation analysis to identify genes that were correlated and in their expression to the expression of these genes.

## Subset06
The Subset06 was plotted for each of the proteins and correlations were calculated, correlations with a cutoff of 0.5 were use to perform enrichment of correlated and anticorrelated genes
```{r plot_Subset06}
data<-Subset06$data

WT1<-data[17410,]
EHD3<-data[4622,]
NPHS1<-data[10252,]
NPHS2<-data[10253,]
CD2AP<-data[2529,]
PDGFRB<-data[11312,]
PECAM1<-data[11364,]
VWF<-data[17259,]
ACTA2<-data[198,]

table_genes=data.frame(WT1=t(WT1),EHD3=t(EHD3),NPHS1=t(NPHS1),NPHS2=t(NPHS2),CD2AP=t(CD2AP),PDGFRB=t(PDGFRB), PECAM1=t(PECAM1),VWF=t(VWF),ACTA2=t(ACTA2),condition=t(Subset06$metadata[5,]))

colnames(table_genes)= c("WT1","EHD3","NPHS1","NPHS2","CD2AP","PDGFRB","PECAM1","VWF","ACTA2","Condition")

ggplot(table_genes,aes(x=WT1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("WT1")
ggplot(table_genes,aes(x=EHD3,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("EHD3")
ggplot(table_genes,aes(x=NPHS1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS1")
ggplot(table_genes,aes(x=NPHS2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS2")
ggplot(table_genes,aes(x=CD2AP,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("CD2AP")
ggplot(table_genes,aes(x=PDGFRB,y=Condition,fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PDGFRB")
ggplot(table_genes,aes(x=PECAM1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PECAM")
ggplot(table_genes,aes(x=VWF,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("VWF")
ggplot(table_genes,aes(x=ACTA2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("ACTA2")

#the correlation cutoff is set
cutoff<-0.5

print(paste0("The correlation cutoff was set to ", cutoff))

#WT1
print("Correlation to WT1 were calculated")

cor_WT1<-numeric()
for(i in 1:nrow(data)){
  cor_WT1[i]<-cor(as.numeric(WT1),as.numeric(data[i,]),method = "pearson")
}
cor_WT1<-data.frame(Pearson_correlation=as.numeric(cor_WT1))
rownames(cor_WT1)<-rownames(data)
ggplot(cor_WT1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "WT1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for WT1")

universe<-rownames(cor_WT1)
correlated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation < -1*cutoff]

WT1_correlated_enr_GO<-cbind(gene="WT1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
WT1_anticorrelated_enr_GO<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
WT1_correlated_enr_KEGG<-cbind(gene="WT1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
WT1_anticorrelated_enr_KEGG<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

WT1_enr<-rbind(WT1_correlated_enr_GO,WT1_anticorrelated_enr_GO,WT1_correlated_enr_KEGG,WT1_anticorrelated_enr_KEGG)

write.csv(cor_WT1,file = "03_output_files/Correlations_WT1_Subset06.csv")

#EHD3
print("Correlation to EHD3 were calculated")

cor_EHD3<-numeric()
for(i in 1:nrow(data)){
  cor_EHD3[i]<-cor(as.numeric(EHD3),as.numeric(data[i,]),method = "pearson")
}
cor_EHD3<-data.frame(Pearson_correlation=as.numeric(cor_EHD3))
rownames(cor_EHD3)<-rownames(data)
ggplot(cor_EHD3,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "EHD3 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for EHD3")

universe<-rownames(cor_EHD3)
correlated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation < -1*cutoff]

EHD3_correlated_enr_GO<-cbind(gene="EHD3",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
EHD3_anticorrelated_enr_GO<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
EHD3_correlated_enr_KEGG<-cbind(gene="EHD3",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
EHD3_anticorrelated_enr_KEGG<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

EHD3_enr<-rbind(EHD3_correlated_enr_GO,EHD3_anticorrelated_enr_GO,EHD3_correlated_enr_KEGG,EHD3_anticorrelated_enr_KEGG)

write.csv(cor_EHD3,file = "03_output_files/Correlations_EHD3_Subset06.csv")

#NPHS1
print("Correlation to NPHS1 were calculated")

cor_NPHS1<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS1[i]<-cor(as.numeric(NPHS1),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS1<-data.frame(Pearson_correlation=as.numeric(cor_NPHS1))
rownames(cor_NPHS1)<-rownames(data)
ggplot(cor_NPHS1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS1")

universe<-rownames(cor_NPHS1)
correlated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation < -1*cutoff]

NPHS1_correlated_enr_GO<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS1_anticorrelated_enr_GO<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS1_correlated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS1_anticorrelated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS1_enr<-rbind(NPHS1_correlated_enr_GO,NPHS1_anticorrelated_enr_GO,NPHS1_correlated_enr_KEGG,NPHS1_anticorrelated_enr_KEGG)

write.csv(cor_NPHS1,file = "03_output_files/Correlations_NPHS1_Subset06.csv")

#NPHS2
print("Correlation to NPHS2 were calculated")

cor_NPHS2<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS2[i]<-cor(as.numeric(NPHS2),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS2<-data.frame(Pearson_correlation=as.numeric(cor_NPHS2))
rownames(cor_NPHS2)<-rownames(data)
ggplot(cor_NPHS2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS2")

universe<-rownames(cor_NPHS2)
correlated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation < -1*cutoff]

NPHS2_correlated_enr_GO<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS2_anticorrelated_enr_GO<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS2_correlated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS2_anticorrelated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS2_enr<-rbind(NPHS2_correlated_enr_GO,NPHS2_anticorrelated_enr_GO,NPHS2_correlated_enr_KEGG,NPHS2_anticorrelated_enr_KEGG)

write.csv(cor_NPHS2,file = "03_output_files/Correlations_NPHS2_Subset06.csv")

#CD2AP
print("Correlation to CD2AP were calculated")

cor_CD2AP<-numeric()
for(i in 1:nrow(data)){
  cor_CD2AP[i]<-cor(as.numeric(CD2AP),as.numeric(data[i,]),method = "pearson")
}
cor_CD2AP<-data.frame(Pearson_correlation=as.numeric(cor_CD2AP))
rownames(cor_CD2AP)<-rownames(data)
ggplot(cor_CD2AP,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "CD2AP correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for CD2AP")

universe<-rownames(cor_CD2AP)
correlated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation < -1*cutoff]

CD2AP_correlated_enr_GO<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
CD2AP_anticorrelated_enr_GO<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
CD2AP_correlated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
CD2AP_anticorrelated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

CD2AP_enr<-rbind(CD2AP_correlated_enr_GO,CD2AP_anticorrelated_enr_GO,CD2AP_correlated_enr_KEGG,CD2AP_anticorrelated_enr_KEGG)

write.csv(cor_CD2AP,file = "03_output_files/Correlations_CD2AP_Subset06.csv")

#PDGFRB
print("Correlation to PDGFRB were calculated")

cor_PDGFRB<-numeric()
for(i in 1:nrow(data)){
  cor_PDGFRB[i]<-cor(as.numeric(PDGFRB),as.numeric(data[i,]),method = "pearson")
}
cor_PDGFRB<-data.frame(Pearson_correlation=as.numeric(cor_PDGFRB))
rownames(cor_PDGFRB)<-rownames(data)
ggplot(cor_PDGFRB,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PDGFRB correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PDGFRB")

universe<-rownames(cor_PDGFRB)
correlated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation < -1*cutoff]

PDGFRB_correlated_enr_GO<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_GO<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PDGFRB_correlated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PDGFRB_enr<-rbind(PDGFRB_correlated_enr_GO,PDGFRB_anticorrelated_enr_GO,PDGFRB_correlated_enr_KEGG,PDGFRB_anticorrelated_enr_KEGG)

write.csv(cor_PDGFRB,file = "03_output_files/Correlations_PDGFRB_Subset06.csv")


#PECAM1
print("Correlation to PECAM1 were calculated")

cor_PECAM1<-numeric()
for(i in 1:nrow(data)){
  cor_PECAM1[i]<-cor(as.numeric(PECAM1),as.numeric(data[i,]),method = "pearson")
}
cor_PECAM1<-data.frame(Pearson_correlation=as.numeric(cor_PECAM1))
rownames(cor_PECAM1)<-rownames(data)
ggplot(cor_PECAM1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PECAM1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PECAM1")

universe<-rownames(cor_PECAM1)
correlated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation < -1*cutoff]

PECAM1_correlated_enr_GO<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PECAM1_anticorrelated_enr_GO<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PECAM1_correlated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PECAM1_anticorrelated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PECAM1_enr<-rbind(PECAM1_correlated_enr_GO,PECAM1_anticorrelated_enr_GO,PECAM1_correlated_enr_KEGG,PECAM1_anticorrelated_enr_KEGG)

write.csv(cor_PECAM1,file = "03_output_files/Correlations_PECAM1_Subset06.csv")

#VWF
print("Correlation to VWF were calculated")

cor_VWF<-numeric()
for(i in 1:nrow(data)){
  cor_VWF[i]<-cor(as.numeric(VWF),as.numeric(data[i,]),method = "pearson")
}
cor_VWF<-data.frame(Pearson_correlation=as.numeric(cor_VWF))
rownames(cor_VWF)<-rownames(data)
ggplot(cor_VWF,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "VWF correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for VWF")

universe<-rownames(cor_VWF)
correlated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation < -1*cutoff]

VWF_correlated_enr_GO<-cbind(gene="VWF",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
VWF_anticorrelated_enr_GO<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
VWF_correlated_enr_KEGG<-cbind(gene="VWF",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
VWF_anticorrelated_enr_KEGG<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

VWF_enr<-rbind(VWF_correlated_enr_GO,VWF_anticorrelated_enr_GO,VWF_correlated_enr_KEGG,VWF_anticorrelated_enr_KEGG)

write.csv(cor_VWF,file = "03_output_files/Correlations_VWF_Subset06.csv")

#ACTA2
print("Correlation to ACTA2 were calculated")

cor_ACTA2<-numeric()
for(i in 1:nrow(data)){
  cor_ACTA2[i]<-cor(as.numeric(ACTA2),as.numeric(data[i,]),method = "pearson")
}
cor_ACTA2<-data.frame(Pearson_correlation=as.numeric(cor_ACTA2))
rownames(cor_ACTA2)<-rownames(data)
ggplot(cor_ACTA2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "ACTA2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for ACTA2")

universe<-rownames(cor_ACTA2)
correlated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation < -1*cutoff]

ACTA2_correlated_enr_GO<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
ACTA2_anticorrelated_enr_GO<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
ACTA2_correlated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
ACTA2_anticorrelated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

ACTA2_enr<-rbind(ACTA2_correlated_enr_GO,ACTA2_anticorrelated_enr_GO,ACTA2_correlated_enr_KEGG,ACTA2_anticorrelated_enr_KEGG)

write.csv(cor_ACTA2,file = "03_output_files/Correlations_ACTA2_Subset06.csv")

gene_correlation_enrichments<-rbind(WT1_enr, EHD3_enr,NPHS1_enr,NPHS2_enr,CD2AP_enr,PDGFRB_enr,PECAM1_enr, VWF_enr,ACTA2_enr)
gene_correlation_enrichments<-gene_correlation_enrichments[gene_correlation_enrichments$pval<0.05&gene_correlation_enrichments$fold_change>1,]

write.table(gene_correlation_enrichments,file = "03_output_files/gene_correlation_enrichments_Subset06.txt",sep = "\t",row.names = F)
```

## Subset07
The Subset07 was plotted for each of the proteins and correlations were calculated, correlations with a cutoff of 0.5 were use to perform enrichment of correlated and anticorrelated genes
```{r plot_Subset07}
data<-Subset07$data

WT1<-data[17410,]
EHD3<-data[4622,]
NPHS1<-data[10252,]
NPHS2<-data[10253,]
CD2AP<-data[2529,]
PDGFRB<-data[11312,]
PECAM1<-data[11364,]
VWF<-data[17259,]
ACTA2<-data[198,]

table_genes=data.frame(WT1=t(WT1),EHD3=t(EHD3),NPHS1=t(NPHS1),NPHS2=t(NPHS2),CD2AP=t(CD2AP),PDGFRB=t(PDGFRB), PECAM1=t(PECAM1),VWF=t(VWF),ACTA2=t(ACTA2),condition=t(Subset07$metadata[5,]))

colnames(table_genes)= c("WT1","EHD3","NPHS1","NPHS2","CD2AP","PDGFRB","PECAM1","VWF","ACTA2","Condition")

ggplot(table_genes,aes(x=WT1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("WT1")
ggplot(table_genes,aes(x=EHD3,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("EHD3")
ggplot(table_genes,aes(x=NPHS1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS1")
ggplot(table_genes,aes(x=NPHS2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS2")
ggplot(table_genes,aes(x=CD2AP,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("CD2AP")
ggplot(table_genes,aes(x=PDGFRB,y=Condition,fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PDGFRB")
ggplot(table_genes,aes(x=PECAM1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PECAM")
ggplot(table_genes,aes(x=VWF,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("VWF")
ggplot(table_genes,aes(x=ACTA2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("ACTA2")

#the correlation cutoff is set
cutoff<-0.5

print(paste0("The correlation cutoff was set to ", cutoff))

#WT1
print("Correlation to WT1 were calculated")

cor_WT1<-numeric()
for(i in 1:nrow(data)){
  cor_WT1[i]<-cor(as.numeric(WT1),as.numeric(data[i,]),method = "pearson")
}
cor_WT1<-data.frame(Pearson_correlation=as.numeric(cor_WT1))
rownames(cor_WT1)<-rownames(data)
ggplot(cor_WT1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "WT1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for WT1")

universe<-rownames(cor_WT1)
correlated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation < -1*cutoff]

WT1_correlated_enr_GO<-cbind(gene="WT1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
WT1_anticorrelated_enr_GO<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
WT1_correlated_enr_KEGG<-cbind(gene="WT1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
WT1_anticorrelated_enr_KEGG<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

WT1_enr<-rbind(WT1_correlated_enr_GO,WT1_anticorrelated_enr_GO,WT1_correlated_enr_KEGG,WT1_anticorrelated_enr_KEGG)

write.csv(cor_WT1,file = "03_output_files/Correlations_WT1_Subset07.csv")

#EHD3
print("Correlation to EHD3 were calculated")

cor_EHD3<-numeric()
for(i in 1:nrow(data)){
  cor_EHD3[i]<-cor(as.numeric(EHD3),as.numeric(data[i,]),method = "pearson")
}
cor_EHD3<-data.frame(Pearson_correlation=as.numeric(cor_EHD3))
rownames(cor_EHD3)<-rownames(data)
ggplot(cor_EHD3,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "EHD3 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for EHD3")

universe<-rownames(cor_EHD3)
correlated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation < -1*cutoff]

EHD3_correlated_enr_GO<-cbind(gene="EHD3",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
EHD3_anticorrelated_enr_GO<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
EHD3_correlated_enr_KEGG<-cbind(gene="EHD3",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
EHD3_anticorrelated_enr_KEGG<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

EHD3_enr<-rbind(EHD3_correlated_enr_GO,EHD3_anticorrelated_enr_GO,EHD3_correlated_enr_KEGG,EHD3_anticorrelated_enr_KEGG)

write.csv(cor_EHD3,file = "03_output_files/Correlations_EHD3_Subset07.csv")

#NPHS1
print("Correlation to NPHS1 were calculated")

cor_NPHS1<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS1[i]<-cor(as.numeric(NPHS1),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS1<-data.frame(Pearson_correlation=as.numeric(cor_NPHS1))
rownames(cor_NPHS1)<-rownames(data)
ggplot(cor_NPHS1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS1")

universe<-rownames(cor_NPHS1)
correlated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation < -1*cutoff]

NPHS1_correlated_enr_GO<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS1_anticorrelated_enr_GO<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS1_correlated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS1_anticorrelated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS1_enr<-rbind(NPHS1_correlated_enr_GO,NPHS1_anticorrelated_enr_GO,NPHS1_correlated_enr_KEGG,NPHS1_anticorrelated_enr_KEGG)

write.csv(cor_NPHS1,file = "03_output_files/Correlations_NPHS1_Subset07.csv")

#NPHS2
print("Correlation to NPHS2 were calculated")

cor_NPHS2<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS2[i]<-cor(as.numeric(NPHS2),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS2<-data.frame(Pearson_correlation=as.numeric(cor_NPHS2))
rownames(cor_NPHS2)<-rownames(data)
ggplot(cor_NPHS2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS2")

universe<-rownames(cor_NPHS2)
correlated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation < -1*cutoff]

NPHS2_correlated_enr_GO<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS2_anticorrelated_enr_GO<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS2_correlated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS2_anticorrelated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS2_enr<-rbind(NPHS2_correlated_enr_GO,NPHS2_anticorrelated_enr_GO,NPHS2_correlated_enr_KEGG,NPHS2_anticorrelated_enr_KEGG)

write.csv(cor_NPHS2,file = "03_output_files/Correlations_NPHS2_Subset07.csv")

#CD2AP
print("Correlation to CD2AP were calculated")

cor_CD2AP<-numeric()
for(i in 1:nrow(data)){
  cor_CD2AP[i]<-cor(as.numeric(CD2AP),as.numeric(data[i,]),method = "pearson")
}
cor_CD2AP<-data.frame(Pearson_correlation=as.numeric(cor_CD2AP))
rownames(cor_CD2AP)<-rownames(data)
ggplot(cor_CD2AP,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "CD2AP correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for CD2AP")

universe<-rownames(cor_CD2AP)
correlated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation < -1*cutoff]

CD2AP_correlated_enr_GO<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
CD2AP_anticorrelated_enr_GO<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
CD2AP_correlated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
CD2AP_anticorrelated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

CD2AP_enr<-rbind(CD2AP_correlated_enr_GO,CD2AP_anticorrelated_enr_GO,CD2AP_correlated_enr_KEGG,CD2AP_anticorrelated_enr_KEGG)

write.csv(cor_CD2AP,file = "03_output_files/Correlations_CD2AP_Subset07.csv")

#PDGFRB
print("Correlation to PDGFRB were calculated")

cor_PDGFRB<-numeric()
for(i in 1:nrow(data)){
  cor_PDGFRB[i]<-cor(as.numeric(PDGFRB),as.numeric(data[i,]),method = "pearson")
}
cor_PDGFRB<-data.frame(Pearson_correlation=as.numeric(cor_PDGFRB))
rownames(cor_PDGFRB)<-rownames(data)
ggplot(cor_PDGFRB,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PDGFRB correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PDGFRB")

universe<-rownames(cor_PDGFRB)
correlated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation < -1*cutoff]

PDGFRB_correlated_enr_GO<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_GO<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PDGFRB_correlated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PDGFRB_enr<-rbind(PDGFRB_correlated_enr_GO,PDGFRB_anticorrelated_enr_GO,PDGFRB_correlated_enr_KEGG,PDGFRB_anticorrelated_enr_KEGG)

write.csv(cor_PDGFRB,file = "03_output_files/Correlations_PDGFRB_Subset07.csv")


#PECAM1
print("Correlation to PECAM1 were calculated")

cor_PECAM1<-numeric()
for(i in 1:nrow(data)){
  cor_PECAM1[i]<-cor(as.numeric(PECAM1),as.numeric(data[i,]),method = "pearson")
}
cor_PECAM1<-data.frame(Pearson_correlation=as.numeric(cor_PECAM1))
rownames(cor_PECAM1)<-rownames(data)
ggplot(cor_PECAM1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PECAM1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PECAM1")

universe<-rownames(cor_PECAM1)
correlated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation < -1*cutoff]

PECAM1_correlated_enr_GO<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PECAM1_anticorrelated_enr_GO<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PECAM1_correlated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PECAM1_anticorrelated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PECAM1_enr<-rbind(PECAM1_correlated_enr_GO,PECAM1_anticorrelated_enr_GO,PECAM1_correlated_enr_KEGG,PECAM1_anticorrelated_enr_KEGG)

write.csv(cor_PECAM1,file = "03_output_files/Correlations_PECAM1_Subset07.csv")

#VWF
print("Correlation to VWF were calculated")

cor_VWF<-numeric()
for(i in 1:nrow(data)){
  cor_VWF[i]<-cor(as.numeric(VWF),as.numeric(data[i,]),method = "pearson")
}
cor_VWF<-data.frame(Pearson_correlation=as.numeric(cor_VWF))
rownames(cor_VWF)<-rownames(data)
ggplot(cor_VWF,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "VWF correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for VWF")

universe<-rownames(cor_VWF)
correlated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation < -1*cutoff]

VWF_correlated_enr_GO<-cbind(gene="VWF",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
VWF_anticorrelated_enr_GO<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
VWF_correlated_enr_KEGG<-cbind(gene="VWF",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
VWF_anticorrelated_enr_KEGG<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

VWF_enr<-rbind(VWF_correlated_enr_GO,VWF_anticorrelated_enr_GO,VWF_correlated_enr_KEGG,VWF_anticorrelated_enr_KEGG)

write.csv(cor_VWF,file = "03_output_files/Correlations_VWF_Subset07.csv")

#ACTA2
print("Correlation to ACTA2 were calculated")

cor_ACTA2<-numeric()
for(i in 1:nrow(data)){
  cor_ACTA2[i]<-cor(as.numeric(ACTA2),as.numeric(data[i,]),method = "pearson")
}
cor_ACTA2<-data.frame(Pearson_correlation=as.numeric(cor_ACTA2))
rownames(cor_ACTA2)<-rownames(data)
ggplot(cor_ACTA2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "ACTA2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for ACTA2")

universe<-rownames(cor_ACTA2)
correlated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation < -1*cutoff]

ACTA2_correlated_enr_GO<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
ACTA2_anticorrelated_enr_GO<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
ACTA2_correlated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
ACTA2_anticorrelated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

ACTA2_enr<-rbind(ACTA2_correlated_enr_GO,ACTA2_anticorrelated_enr_GO,ACTA2_correlated_enr_KEGG,ACTA2_anticorrelated_enr_KEGG)

write.csv(cor_ACTA2,file = "03_output_files/Correlations_ACTA2_Subset07.csv")

gene_correlation_enrichments<-rbind(WT1_enr, EHD3_enr,NPHS1_enr,NPHS2_enr,CD2AP_enr,PDGFRB_enr,PECAM1_enr, VWF_enr,ACTA2_enr)
gene_correlation_enrichments<-gene_correlation_enrichments[gene_correlation_enrichments$pval<0.05&gene_correlation_enrichments$fold_change>1,]

write.table(gene_correlation_enrichments,file = "03_output_files/gene_correlation_enrichments_Subset07.txt",sep = "\t",row.names = F)
```


## Subset08
The Subset08 was plotted for each of the proteins and correlations were calculated, correlations with a cutoff of 0.5 were use to perform enrichment of correlated and anticorrelated genes
```{r plot_Subset08}
data<-Subset08$data

WT1<-data[17410,]
EHD3<-data[4622,]
NPHS1<-data[10252,]
NPHS2<-data[10253,]
CD2AP<-data[2529,]
PDGFRB<-data[11312,]
PECAM1<-data[11364,]
VWF<-data[17259,]
ACTA2<-data[198,]

table_genes=data.frame(WT1=t(WT1),EHD3=t(EHD3),NPHS1=t(NPHS1),NPHS2=t(NPHS2),CD2AP=t(CD2AP),PDGFRB=t(PDGFRB), PECAM1=t(PECAM1),VWF=t(VWF),ACTA2=t(ACTA2),condition=t(Subset08$metadata[5,]))

colnames(table_genes)= c("WT1","EHD3","NPHS1","NPHS2","CD2AP","PDGFRB","PECAM1","VWF","ACTA2","Condition")

ggplot(table_genes,aes(x=WT1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("WT1")
ggplot(table_genes,aes(x=EHD3,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("EHD3")
ggplot(table_genes,aes(x=NPHS1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS1")
ggplot(table_genes,aes(x=NPHS2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS2")
ggplot(table_genes,aes(x=CD2AP,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("CD2AP")
ggplot(table_genes,aes(x=PDGFRB,y=Condition,fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PDGFRB")
ggplot(table_genes,aes(x=PECAM1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PECAM")
ggplot(table_genes,aes(x=VWF,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("VWF")
ggplot(table_genes,aes(x=ACTA2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("ACTA2")

#the correlation cutoff is set
cutoff<-0.5

print(paste0("The correlation cutoff was set to ", cutoff))

#WT1
print("Correlation to WT1 were calculated")

cor_WT1<-numeric()
for(i in 1:nrow(data)){
  cor_WT1[i]<-cor(as.numeric(WT1),as.numeric(data[i,]),method = "pearson")
}
cor_WT1<-data.frame(Pearson_correlation=as.numeric(cor_WT1))
rownames(cor_WT1)<-rownames(data)
ggplot(cor_WT1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "WT1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for WT1")

universe<-rownames(cor_WT1)
correlated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation < -1*cutoff]

WT1_correlated_enr_GO<-cbind(gene="WT1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
WT1_anticorrelated_enr_GO<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
WT1_correlated_enr_KEGG<-cbind(gene="WT1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
WT1_anticorrelated_enr_KEGG<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

WT1_enr<-rbind(WT1_correlated_enr_GO,WT1_anticorrelated_enr_GO,WT1_correlated_enr_KEGG,WT1_anticorrelated_enr_KEGG)

write.csv(cor_WT1,file = "03_output_files/Correlations_WT1_Subset08.csv")

#EHD3
print("Correlation to EHD3 were calculated")

cor_EHD3<-numeric()
for(i in 1:nrow(data)){
  cor_EHD3[i]<-cor(as.numeric(EHD3),as.numeric(data[i,]),method = "pearson")
}
cor_EHD3<-data.frame(Pearson_correlation=as.numeric(cor_EHD3))
rownames(cor_EHD3)<-rownames(data)
ggplot(cor_EHD3,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "EHD3 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for EHD3")

universe<-rownames(cor_EHD3)
correlated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation < -1*cutoff]

EHD3_correlated_enr_GO<-cbind(gene="EHD3",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
EHD3_anticorrelated_enr_GO<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
EHD3_correlated_enr_KEGG<-cbind(gene="EHD3",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
EHD3_anticorrelated_enr_KEGG<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

EHD3_enr<-rbind(EHD3_correlated_enr_GO,EHD3_anticorrelated_enr_GO,EHD3_correlated_enr_KEGG,EHD3_anticorrelated_enr_KEGG)

write.csv(cor_EHD3,file = "03_output_files/Correlations_EHD3_Subset08.csv")

#NPHS1
print("Correlation to NPHS1 were calculated")

cor_NPHS1<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS1[i]<-cor(as.numeric(NPHS1),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS1<-data.frame(Pearson_correlation=as.numeric(cor_NPHS1))
rownames(cor_NPHS1)<-rownames(data)
ggplot(cor_NPHS1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS1")

universe<-rownames(cor_NPHS1)
correlated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation < -1*cutoff]

NPHS1_correlated_enr_GO<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS1_anticorrelated_enr_GO<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS1_correlated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS1_anticorrelated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS1_enr<-rbind(NPHS1_correlated_enr_GO,NPHS1_anticorrelated_enr_GO,NPHS1_correlated_enr_KEGG,NPHS1_anticorrelated_enr_KEGG)

write.csv(cor_NPHS1,file = "03_output_files/Correlations_NPHS1_Subset08.csv")

#NPHS2
print("Correlation to NPHS2 were calculated")

cor_NPHS2<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS2[i]<-cor(as.numeric(NPHS2),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS2<-data.frame(Pearson_correlation=as.numeric(cor_NPHS2))
rownames(cor_NPHS2)<-rownames(data)
ggplot(cor_NPHS2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS2")

universe<-rownames(cor_NPHS2)
correlated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation < -1*cutoff]

NPHS2_correlated_enr_GO<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS2_anticorrelated_enr_GO<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS2_correlated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS2_anticorrelated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS2_enr<-rbind(NPHS2_correlated_enr_GO,NPHS2_anticorrelated_enr_GO,NPHS2_correlated_enr_KEGG,NPHS2_anticorrelated_enr_KEGG)

write.csv(cor_NPHS2,file = "03_output_files/Correlations_NPHS2_Subset08.csv")

#CD2AP
print("Correlation to CD2AP were calculated")

cor_CD2AP<-numeric()
for(i in 1:nrow(data)){
  cor_CD2AP[i]<-cor(as.numeric(CD2AP),as.numeric(data[i,]),method = "pearson")
}
cor_CD2AP<-data.frame(Pearson_correlation=as.numeric(cor_CD2AP))
rownames(cor_CD2AP)<-rownames(data)
ggplot(cor_CD2AP,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "CD2AP correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for CD2AP")

universe<-rownames(cor_CD2AP)
correlated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation < -1*cutoff]

CD2AP_correlated_enr_GO<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
CD2AP_anticorrelated_enr_GO<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
CD2AP_correlated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
CD2AP_anticorrelated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

CD2AP_enr<-rbind(CD2AP_correlated_enr_GO,CD2AP_anticorrelated_enr_GO,CD2AP_correlated_enr_KEGG,CD2AP_anticorrelated_enr_KEGG)

write.csv(cor_CD2AP,file = "03_output_files/Correlations_CD2AP_Subset08.csv")

#PDGFRB
print("Correlation to PDGFRB were calculated")

cor_PDGFRB<-numeric()
for(i in 1:nrow(data)){
  cor_PDGFRB[i]<-cor(as.numeric(PDGFRB),as.numeric(data[i,]),method = "pearson")
}
cor_PDGFRB<-data.frame(Pearson_correlation=as.numeric(cor_PDGFRB))
rownames(cor_PDGFRB)<-rownames(data)
ggplot(cor_PDGFRB,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PDGFRB correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PDGFRB")

universe<-rownames(cor_PDGFRB)
correlated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation < -1*cutoff]

PDGFRB_correlated_enr_GO<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_GO<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PDGFRB_correlated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PDGFRB_enr<-rbind(PDGFRB_correlated_enr_GO,PDGFRB_anticorrelated_enr_GO,PDGFRB_correlated_enr_KEGG,PDGFRB_anticorrelated_enr_KEGG)

write.csv(cor_PDGFRB,file = "03_output_files/Correlations_PDGFRB_Subset08.csv")


#PECAM1
print("Correlation to PECAM1 were calculated")

cor_PECAM1<-numeric()
for(i in 1:nrow(data)){
  cor_PECAM1[i]<-cor(as.numeric(PECAM1),as.numeric(data[i,]),method = "pearson")
}
cor_PECAM1<-data.frame(Pearson_correlation=as.numeric(cor_PECAM1))
rownames(cor_PECAM1)<-rownames(data)
ggplot(cor_PECAM1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PECAM1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PECAM1")

universe<-rownames(cor_PECAM1)
correlated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation < -1*cutoff]

PECAM1_correlated_enr_GO<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PECAM1_anticorrelated_enr_GO<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PECAM1_correlated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PECAM1_anticorrelated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PECAM1_enr<-rbind(PECAM1_correlated_enr_GO,PECAM1_anticorrelated_enr_GO,PECAM1_correlated_enr_KEGG,PECAM1_anticorrelated_enr_KEGG)

write.csv(cor_PECAM1,file = "03_output_files/Correlations_PECAM1_Subset08.csv")

#VWF
print("Correlation to VWF were calculated")

cor_VWF<-numeric()
for(i in 1:nrow(data)){
  cor_VWF[i]<-cor(as.numeric(VWF),as.numeric(data[i,]),method = "pearson")
}
cor_VWF<-data.frame(Pearson_correlation=as.numeric(cor_VWF))
rownames(cor_VWF)<-rownames(data)
ggplot(cor_VWF,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "VWF correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for VWF")

universe<-rownames(cor_VWF)
correlated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation < -1*cutoff]

VWF_correlated_enr_GO<-cbind(gene="VWF",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
VWF_anticorrelated_enr_GO<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
VWF_correlated_enr_KEGG<-cbind(gene="VWF",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
VWF_anticorrelated_enr_KEGG<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

VWF_enr<-rbind(VWF_correlated_enr_GO,VWF_anticorrelated_enr_GO,VWF_correlated_enr_KEGG,VWF_anticorrelated_enr_KEGG)

write.csv(cor_VWF,file = "03_output_files/Correlations_VWF_Subset08.csv")

#ACTA2
print("Correlation to ACTA2 were calculated")

cor_ACTA2<-numeric()
for(i in 1:nrow(data)){
  cor_ACTA2[i]<-cor(as.numeric(ACTA2),as.numeric(data[i,]),method = "pearson")
}
cor_ACTA2<-data.frame(Pearson_correlation=as.numeric(cor_ACTA2))
rownames(cor_ACTA2)<-rownames(data)
ggplot(cor_ACTA2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "ACTA2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for ACTA2")

universe<-rownames(cor_ACTA2)
correlated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation < -1*cutoff]

ACTA2_correlated_enr_GO<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
ACTA2_anticorrelated_enr_GO<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
ACTA2_correlated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
ACTA2_anticorrelated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

ACTA2_enr<-rbind(ACTA2_correlated_enr_GO,ACTA2_anticorrelated_enr_GO,ACTA2_correlated_enr_KEGG,ACTA2_anticorrelated_enr_KEGG)

write.csv(cor_ACTA2,file = "03_output_files/Correlations_ACTA2_Subset08.csv")

gene_correlation_enrichments<-rbind(WT1_enr, EHD3_enr,NPHS1_enr,NPHS2_enr,CD2AP_enr,PDGFRB_enr,PECAM1_enr, VWF_enr,ACTA2_enr)
gene_correlation_enrichments<-gene_correlation_enrichments[gene_correlation_enrichments$pval<0.05&gene_correlation_enrichments$fold_change>1,]

write.table(gene_correlation_enrichments,file = "03_output_files/gene_correlation_enrichments_Subset08.txt",sep = "\t",row.names = F)
```

## Subset14
The Subset14 was plotted for each of the proteins and correlations were calculated, correlations with a cutoff of 0.5 were use to perform enrichment of correlated and anticorrelated genes
```{r plot_Subset14}
data<-Subset14$data

WT1<-data[17410,]
EHD3<-data[4622,]
NPHS1<-data[10252,]
NPHS2<-data[10253,]
CD2AP<-data[2529,]
PDGFRB<-data[11312,]
PECAM1<-data[11364,]
VWF<-data[17259,]
ACTA2<-data[198,]

table_genes=data.frame(WT1=t(WT1),EHD3=t(EHD3),NPHS1=t(NPHS1),NPHS2=t(NPHS2),CD2AP=t(CD2AP),PDGFRB=t(PDGFRB), PECAM1=t(PECAM1),VWF=t(VWF),ACTA2=t(ACTA2),condition=t(Subset14$metadata[5,]))

colnames(table_genes)= c("WT1","EHD3","NPHS1","NPHS2","CD2AP","PDGFRB","PECAM1","VWF","ACTA2","Condition")

ggplot(table_genes,aes(x=WT1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("WT1")
ggplot(table_genes,aes(x=EHD3,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("EHD3")
ggplot(table_genes,aes(x=NPHS1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS1")
ggplot(table_genes,aes(x=NPHS2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS2")
ggplot(table_genes,aes(x=CD2AP,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("CD2AP")
ggplot(table_genes,aes(x=PDGFRB,y=Condition,fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PDGFRB")
ggplot(table_genes,aes(x=PECAM1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PECAM")
ggplot(table_genes,aes(x=VWF,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("VWF")
ggplot(table_genes,aes(x=ACTA2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("ACTA2")

#the correlation cutoff is set
cutoff<-0.5

print(paste0("The correlation cutoff was set to ", cutoff))

#WT1
print("Correlation to WT1 were calculated")

cor_WT1<-numeric()
for(i in 1:nrow(data)){
  cor_WT1[i]<-cor(as.numeric(WT1),as.numeric(data[i,]),method = "pearson")
}
cor_WT1<-data.frame(Pearson_correlation=as.numeric(cor_WT1))
rownames(cor_WT1)<-rownames(data)
ggplot(cor_WT1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "WT1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for WT1")

universe<-rownames(cor_WT1)
correlated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation < -1*cutoff]

WT1_correlated_enr_GO<-cbind(gene="WT1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
WT1_anticorrelated_enr_GO<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
WT1_correlated_enr_KEGG<-cbind(gene="WT1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
WT1_anticorrelated_enr_KEGG<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

WT1_enr<-rbind(WT1_correlated_enr_GO,WT1_anticorrelated_enr_GO,WT1_correlated_enr_KEGG,WT1_anticorrelated_enr_KEGG)

write.csv(cor_WT1,file = "03_output_files/Correlations_WT1_Subset14.csv")

#EHD3
print("Correlation to EHD3 were calculated")

cor_EHD3<-numeric()
for(i in 1:nrow(data)){
  cor_EHD3[i]<-cor(as.numeric(EHD3),as.numeric(data[i,]),method = "pearson")
}
cor_EHD3<-data.frame(Pearson_correlation=as.numeric(cor_EHD3))
rownames(cor_EHD3)<-rownames(data)
ggplot(cor_EHD3,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "EHD3 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for EHD3")

universe<-rownames(cor_EHD3)
correlated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation < -1*cutoff]

EHD3_correlated_enr_GO<-cbind(gene="EHD3",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
EHD3_anticorrelated_enr_GO<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
EHD3_correlated_enr_KEGG<-cbind(gene="EHD3",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
EHD3_anticorrelated_enr_KEGG<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

EHD3_enr<-rbind(EHD3_correlated_enr_GO,EHD3_anticorrelated_enr_GO,EHD3_correlated_enr_KEGG,EHD3_anticorrelated_enr_KEGG)

write.csv(cor_EHD3,file = "03_output_files/Correlations_EHD3_Subset14.csv")

#NPHS1
print("Correlation to NPHS1 were calculated")

cor_NPHS1<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS1[i]<-cor(as.numeric(NPHS1),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS1<-data.frame(Pearson_correlation=as.numeric(cor_NPHS1))
rownames(cor_NPHS1)<-rownames(data)
ggplot(cor_NPHS1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS1")

universe<-rownames(cor_NPHS1)
correlated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation < -1*cutoff]

NPHS1_correlated_enr_GO<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS1_anticorrelated_enr_GO<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS1_correlated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS1_anticorrelated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS1_enr<-rbind(NPHS1_correlated_enr_GO,NPHS1_anticorrelated_enr_GO,NPHS1_correlated_enr_KEGG,NPHS1_anticorrelated_enr_KEGG)

write.csv(cor_NPHS1,file = "03_output_files/Correlations_NPHS1_Subset14.csv")

#NPHS2
print("Correlation to NPHS2 were calculated")

cor_NPHS2<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS2[i]<-cor(as.numeric(NPHS2),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS2<-data.frame(Pearson_correlation=as.numeric(cor_NPHS2))
rownames(cor_NPHS2)<-rownames(data)
ggplot(cor_NPHS2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS2")

universe<-rownames(cor_NPHS2)
correlated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation < -1*cutoff]

NPHS2_correlated_enr_GO<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS2_anticorrelated_enr_GO<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS2_correlated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS2_anticorrelated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS2_enr<-rbind(NPHS2_correlated_enr_GO,NPHS2_anticorrelated_enr_GO,NPHS2_correlated_enr_KEGG,NPHS2_anticorrelated_enr_KEGG)

write.csv(cor_NPHS2,file = "03_output_files/Correlations_NPHS2_Subset14.csv")

#CD2AP
print("Correlation to CD2AP were calculated")

cor_CD2AP<-numeric()
for(i in 1:nrow(data)){
  cor_CD2AP[i]<-cor(as.numeric(CD2AP),as.numeric(data[i,]),method = "pearson")
}
cor_CD2AP<-data.frame(Pearson_correlation=as.numeric(cor_CD2AP))
rownames(cor_CD2AP)<-rownames(data)
ggplot(cor_CD2AP,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "CD2AP correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for CD2AP")

universe<-rownames(cor_CD2AP)
correlated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation < -1*cutoff]

CD2AP_correlated_enr_GO<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
CD2AP_anticorrelated_enr_GO<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
CD2AP_correlated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
CD2AP_anticorrelated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

CD2AP_enr<-rbind(CD2AP_correlated_enr_GO,CD2AP_anticorrelated_enr_GO,CD2AP_correlated_enr_KEGG,CD2AP_anticorrelated_enr_KEGG)

write.csv(cor_CD2AP,file = "03_output_files/Correlations_CD2AP_Subset14.csv")

#PDGFRB
print("Correlation to PDGFRB were calculated")

cor_PDGFRB<-numeric()
for(i in 1:nrow(data)){
  cor_PDGFRB[i]<-cor(as.numeric(PDGFRB),as.numeric(data[i,]),method = "pearson")
}
cor_PDGFRB<-data.frame(Pearson_correlation=as.numeric(cor_PDGFRB))
rownames(cor_PDGFRB)<-rownames(data)
ggplot(cor_PDGFRB,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PDGFRB correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PDGFRB")

universe<-rownames(cor_PDGFRB)
correlated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation < -1*cutoff]

PDGFRB_correlated_enr_GO<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_GO<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PDGFRB_correlated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PDGFRB_enr<-rbind(PDGFRB_correlated_enr_GO,PDGFRB_anticorrelated_enr_GO,PDGFRB_correlated_enr_KEGG,PDGFRB_anticorrelated_enr_KEGG)

write.csv(cor_PDGFRB,file = "03_output_files/Correlations_PDGFRB_Subset14.csv")


#PECAM1
print("Correlation to PECAM1 were calculated")

cor_PECAM1<-numeric()
for(i in 1:nrow(data)){
  cor_PECAM1[i]<-cor(as.numeric(PECAM1),as.numeric(data[i,]),method = "pearson")
}
cor_PECAM1<-data.frame(Pearson_correlation=as.numeric(cor_PECAM1))
rownames(cor_PECAM1)<-rownames(data)
ggplot(cor_PECAM1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PECAM1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PECAM1")

universe<-rownames(cor_PECAM1)
correlated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation < -1*cutoff]

PECAM1_correlated_enr_GO<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PECAM1_anticorrelated_enr_GO<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PECAM1_correlated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PECAM1_anticorrelated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PECAM1_enr<-rbind(PECAM1_correlated_enr_GO,PECAM1_anticorrelated_enr_GO,PECAM1_correlated_enr_KEGG,PECAM1_anticorrelated_enr_KEGG)

write.csv(cor_PECAM1,file = "03_output_files/Correlations_PECAM1_Subset14.csv")

#VWF
print("Correlation to VWF were calculated")

cor_VWF<-numeric()
for(i in 1:nrow(data)){
  cor_VWF[i]<-cor(as.numeric(VWF),as.numeric(data[i,]),method = "pearson")
}
cor_VWF<-data.frame(Pearson_correlation=as.numeric(cor_VWF))
rownames(cor_VWF)<-rownames(data)
ggplot(cor_VWF,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "VWF correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for VWF")

universe<-rownames(cor_VWF)
correlated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation < -1*cutoff]

VWF_correlated_enr_GO<-cbind(gene="VWF",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
VWF_anticorrelated_enr_GO<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
VWF_correlated_enr_KEGG<-cbind(gene="VWF",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
VWF_anticorrelated_enr_KEGG<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

VWF_enr<-rbind(VWF_correlated_enr_GO,VWF_anticorrelated_enr_GO,VWF_correlated_enr_KEGG,VWF_anticorrelated_enr_KEGG)

write.csv(cor_VWF,file = "03_output_files/Correlations_VWF_Subset14.csv")

#ACTA2
print("Correlation to ACTA2 were calculated")

cor_ACTA2<-numeric()
for(i in 1:nrow(data)){
  cor_ACTA2[i]<-cor(as.numeric(ACTA2),as.numeric(data[i,]),method = "pearson")
}
cor_ACTA2<-data.frame(Pearson_correlation=as.numeric(cor_ACTA2))
rownames(cor_ACTA2)<-rownames(data)
ggplot(cor_ACTA2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "ACTA2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for ACTA2")

universe<-rownames(cor_ACTA2)
correlated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation < -1*cutoff]

ACTA2_correlated_enr_GO<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
ACTA2_anticorrelated_enr_GO<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
ACTA2_correlated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
ACTA2_anticorrelated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

ACTA2_enr<-rbind(ACTA2_correlated_enr_GO,ACTA2_anticorrelated_enr_GO,ACTA2_correlated_enr_KEGG,ACTA2_anticorrelated_enr_KEGG)

write.csv(cor_ACTA2,file = "03_output_files/Correlations_ACTA2_Subset14.csv")

gene_correlation_enrichments<-rbind(WT1_enr, EHD3_enr,NPHS1_enr,NPHS2_enr,CD2AP_enr,PDGFRB_enr,PECAM1_enr, VWF_enr,ACTA2_enr)
gene_correlation_enrichments<-gene_correlation_enrichments[gene_correlation_enrichments$pval<0.05&gene_correlation_enrichments$fold_change>1,]

write.table(gene_correlation_enrichments,file = "03_output_files/gene_correlation_enrichments_Subset14.txt",sep = "\t",row.names = F)
```

## Subset19
The Subset19 was plotted for each of the proteins and correlations were calculated, correlations with a cutoff of 0.5 were use to perform enrichment of correlated and anticorrelated genes
```{r plot_Subset19}
data<-Subset19$data

WT1<-data[17410,]
EHD3<-data[4622,]
NPHS1<-data[10252,]
NPHS2<-data[10253,]
CD2AP<-data[2529,]
PDGFRB<-data[11312,]
PECAM1<-data[11364,]
VWF<-data[17259,]
ACTA2<-data[198,]

table_genes=data.frame(WT1=t(WT1),EHD3=t(EHD3),NPHS1=t(NPHS1),NPHS2=t(NPHS2),CD2AP=t(CD2AP),PDGFRB=t(PDGFRB), PECAM1=t(PECAM1),VWF=t(VWF),ACTA2=t(ACTA2),condition=t(Subset19$metadata[5,]))

colnames(table_genes)= c("WT1","EHD3","NPHS1","NPHS2","CD2AP","PDGFRB","PECAM1","VWF","ACTA2","Condition")

ggplot(table_genes,aes(x=WT1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("WT1")
ggplot(table_genes,aes(x=EHD3,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("EHD3")
ggplot(table_genes,aes(x=NPHS1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS1")
ggplot(table_genes,aes(x=NPHS2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS2")
ggplot(table_genes,aes(x=CD2AP,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("CD2AP")
ggplot(table_genes,aes(x=PDGFRB,y=Condition,fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PDGFRB")
ggplot(table_genes,aes(x=PECAM1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PECAM")
ggplot(table_genes,aes(x=VWF,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("VWF")
ggplot(table_genes,aes(x=ACTA2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("ACTA2")

#the correlation cutoff is set
cutoff<-0.5

print(paste0("The correlation cutoff was set to ", cutoff))

#WT1
print("Correlation to WT1 were calculated")

cor_WT1<-numeric()
for(i in 1:nrow(data)){
  cor_WT1[i]<-cor(as.numeric(WT1),as.numeric(data[i,]),method = "pearson")
}
cor_WT1<-data.frame(Pearson_correlation=as.numeric(cor_WT1))
rownames(cor_WT1)<-rownames(data)
ggplot(cor_WT1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "WT1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for WT1")

universe<-rownames(cor_WT1)
correlated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation < -1*cutoff]

WT1_correlated_enr_GO<-cbind(gene="WT1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
WT1_anticorrelated_enr_GO<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
WT1_correlated_enr_KEGG<-cbind(gene="WT1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
WT1_anticorrelated_enr_KEGG<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

WT1_enr<-rbind(WT1_correlated_enr_GO,WT1_anticorrelated_enr_GO,WT1_correlated_enr_KEGG,WT1_anticorrelated_enr_KEGG)

write.csv(cor_WT1,file = "03_output_files/Correlations_WT1_Subset19.csv")

#EHD3
print("Correlation to EHD3 were calculated")

cor_EHD3<-numeric()
for(i in 1:nrow(data)){
  cor_EHD3[i]<-cor(as.numeric(EHD3),as.numeric(data[i,]),method = "pearson")
}
cor_EHD3<-data.frame(Pearson_correlation=as.numeric(cor_EHD3))
rownames(cor_EHD3)<-rownames(data)
ggplot(cor_EHD3,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "EHD3 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for EHD3")

universe<-rownames(cor_EHD3)
correlated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation < -1*cutoff]

EHD3_correlated_enr_GO<-cbind(gene="EHD3",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
EHD3_anticorrelated_enr_GO<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
EHD3_correlated_enr_KEGG<-cbind(gene="EHD3",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
EHD3_anticorrelated_enr_KEGG<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

EHD3_enr<-rbind(EHD3_correlated_enr_GO,EHD3_anticorrelated_enr_GO,EHD3_correlated_enr_KEGG,EHD3_anticorrelated_enr_KEGG)

write.csv(cor_EHD3,file = "03_output_files/Correlations_EHD3_Subset19.csv")

#NPHS1
print("Correlation to NPHS1 were calculated")

cor_NPHS1<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS1[i]<-cor(as.numeric(NPHS1),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS1<-data.frame(Pearson_correlation=as.numeric(cor_NPHS1))
rownames(cor_NPHS1)<-rownames(data)
ggplot(cor_NPHS1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS1")

universe<-rownames(cor_NPHS1)
correlated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation < -1*cutoff]

NPHS1_correlated_enr_GO<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS1_anticorrelated_enr_GO<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS1_correlated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS1_anticorrelated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS1_enr<-rbind(NPHS1_correlated_enr_GO,NPHS1_anticorrelated_enr_GO,NPHS1_correlated_enr_KEGG,NPHS1_anticorrelated_enr_KEGG)

write.csv(cor_NPHS1,file = "03_output_files/Correlations_NPHS1_Subset19.csv")

#NPHS2
print("Correlation to NPHS2 were calculated")

cor_NPHS2<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS2[i]<-cor(as.numeric(NPHS2),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS2<-data.frame(Pearson_correlation=as.numeric(cor_NPHS2))
rownames(cor_NPHS2)<-rownames(data)
ggplot(cor_NPHS2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS2")

universe<-rownames(cor_NPHS2)
correlated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation < -1*cutoff]

NPHS2_correlated_enr_GO<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS2_anticorrelated_enr_GO<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS2_correlated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS2_anticorrelated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS2_enr<-rbind(NPHS2_correlated_enr_GO,NPHS2_anticorrelated_enr_GO,NPHS2_correlated_enr_KEGG,NPHS2_anticorrelated_enr_KEGG)

write.csv(cor_NPHS2,file = "03_output_files/Correlations_NPHS2_Subset19.csv")

#CD2AP
print("Correlation to CD2AP were calculated")

cor_CD2AP<-numeric()
for(i in 1:nrow(data)){
  cor_CD2AP[i]<-cor(as.numeric(CD2AP),as.numeric(data[i,]),method = "pearson")
}
cor_CD2AP<-data.frame(Pearson_correlation=as.numeric(cor_CD2AP))
rownames(cor_CD2AP)<-rownames(data)
ggplot(cor_CD2AP,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "CD2AP correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for CD2AP")

universe<-rownames(cor_CD2AP)
correlated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation < -1*cutoff]

CD2AP_correlated_enr_GO<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
CD2AP_anticorrelated_enr_GO<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
CD2AP_correlated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
CD2AP_anticorrelated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

CD2AP_enr<-rbind(CD2AP_correlated_enr_GO,CD2AP_anticorrelated_enr_GO,CD2AP_correlated_enr_KEGG,CD2AP_anticorrelated_enr_KEGG)

write.csv(cor_CD2AP,file = "03_output_files/Correlations_CD2AP_Subset19.csv")

#PDGFRB
print("Correlation to PDGFRB were calculated")

cor_PDGFRB<-numeric()
for(i in 1:nrow(data)){
  cor_PDGFRB[i]<-cor(as.numeric(PDGFRB),as.numeric(data[i,]),method = "pearson")
}
cor_PDGFRB<-data.frame(Pearson_correlation=as.numeric(cor_PDGFRB))
rownames(cor_PDGFRB)<-rownames(data)
ggplot(cor_PDGFRB,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PDGFRB correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PDGFRB")

universe<-rownames(cor_PDGFRB)
correlated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation < -1*cutoff]

PDGFRB_correlated_enr_GO<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_GO<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PDGFRB_correlated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PDGFRB_enr<-rbind(PDGFRB_correlated_enr_GO,PDGFRB_anticorrelated_enr_GO,PDGFRB_correlated_enr_KEGG,PDGFRB_anticorrelated_enr_KEGG)

write.csv(cor_PDGFRB,file = "03_output_files/Correlations_PDGFRB_Subset19.csv")


#PECAM1
print("Correlation to PECAM1 were calculated")

cor_PECAM1<-numeric()
for(i in 1:nrow(data)){
  cor_PECAM1[i]<-cor(as.numeric(PECAM1),as.numeric(data[i,]),method = "pearson")
}
cor_PECAM1<-data.frame(Pearson_correlation=as.numeric(cor_PECAM1))
rownames(cor_PECAM1)<-rownames(data)
ggplot(cor_PECAM1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PECAM1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PECAM1")

universe<-rownames(cor_PECAM1)
correlated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation < -1*cutoff]

PECAM1_correlated_enr_GO<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PECAM1_anticorrelated_enr_GO<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PECAM1_correlated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PECAM1_anticorrelated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PECAM1_enr<-rbind(PECAM1_correlated_enr_GO,PECAM1_anticorrelated_enr_GO,PECAM1_correlated_enr_KEGG,PECAM1_anticorrelated_enr_KEGG)

write.csv(cor_PECAM1,file = "03_output_files/Correlations_PECAM1_Subset19.csv")

#VWF
print("Correlation to VWF were calculated")

cor_VWF<-numeric()
for(i in 1:nrow(data)){
  cor_VWF[i]<-cor(as.numeric(VWF),as.numeric(data[i,]),method = "pearson")
}
cor_VWF<-data.frame(Pearson_correlation=as.numeric(cor_VWF))
rownames(cor_VWF)<-rownames(data)
ggplot(cor_VWF,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "VWF correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for VWF")

universe<-rownames(cor_VWF)
correlated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation < -1*cutoff]

VWF_correlated_enr_GO<-cbind(gene="VWF",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
VWF_anticorrelated_enr_GO<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
VWF_correlated_enr_KEGG<-cbind(gene="VWF",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
VWF_anticorrelated_enr_KEGG<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

VWF_enr<-rbind(VWF_correlated_enr_GO,VWF_anticorrelated_enr_GO,VWF_correlated_enr_KEGG,VWF_anticorrelated_enr_KEGG)

write.csv(cor_VWF,file = "03_output_files/Correlations_VWF_Subset19.csv")

#ACTA2
print("Correlation to ACTA2 were calculated")

cor_ACTA2<-numeric()
for(i in 1:nrow(data)){
  cor_ACTA2[i]<-cor(as.numeric(ACTA2),as.numeric(data[i,]),method = "pearson")
}
cor_ACTA2<-data.frame(Pearson_correlation=as.numeric(cor_ACTA2))
rownames(cor_ACTA2)<-rownames(data)
ggplot(cor_ACTA2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "ACTA2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for ACTA2")

universe<-rownames(cor_ACTA2)
correlated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation < -1*cutoff]

ACTA2_correlated_enr_GO<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
ACTA2_anticorrelated_enr_GO<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
ACTA2_correlated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
ACTA2_anticorrelated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

ACTA2_enr<-rbind(ACTA2_correlated_enr_GO,ACTA2_anticorrelated_enr_GO,ACTA2_correlated_enr_KEGG,ACTA2_anticorrelated_enr_KEGG)

write.csv(cor_ACTA2,file = "03_output_files/Correlations_ACTA2_Subset19.csv")

gene_correlation_enrichments<-rbind(WT1_enr, EHD3_enr,NPHS1_enr,NPHS2_enr,CD2AP_enr,PDGFRB_enr,PECAM1_enr, VWF_enr,ACTA2_enr)
gene_correlation_enrichments<-gene_correlation_enrichments[gene_correlation_enrichments$pval<0.05&gene_correlation_enrichments$fold_change>1,]

write.table(gene_correlation_enrichments,file = "03_output_files/gene_correlation_enrichments_Subset19.txt",sep = "\t",row.names = F)
```

## Subset20
The Subset20 was plotted for each of the proteins and correlations were calculated, correlations with a cutoff of 0.5 were use to perform enrichment of correlated and anticorrelated genes
```{r plot_Subset20}
data<-Subset20$data

WT1<-data[17410,]
EHD3<-data[4622,]
NPHS1<-data[10252,]
NPHS2<-data[10253,]
CD2AP<-data[2529,]
PDGFRB<-data[11312,]
PECAM1<-data[11364,]
VWF<-data[17259,]
ACTA2<-data[198,]

table_genes=data.frame(WT1=t(WT1),EHD3=t(EHD3),NPHS1=t(NPHS1),NPHS2=t(NPHS2),CD2AP=t(CD2AP),PDGFRB=t(PDGFRB), PECAM1=t(PECAM1),VWF=t(VWF),ACTA2=t(ACTA2),condition=t(Subset20$metadata[5,]))

colnames(table_genes)= c("WT1","EHD3","NPHS1","NPHS2","CD2AP","PDGFRB","PECAM1","VWF","ACTA2","Condition")

ggplot(table_genes,aes(x=WT1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("WT1")
ggplot(table_genes,aes(x=EHD3,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("EHD3")
ggplot(table_genes,aes(x=NPHS1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS1")
ggplot(table_genes,aes(x=NPHS2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS2")
ggplot(table_genes,aes(x=CD2AP,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("CD2AP")
ggplot(table_genes,aes(x=PDGFRB,y=Condition,fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PDGFRB")
ggplot(table_genes,aes(x=PECAM1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PECAM")
ggplot(table_genes,aes(x=VWF,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("VWF")
ggplot(table_genes,aes(x=ACTA2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("ACTA2")

#the correlation cutoff is set
cutoff<-0.5

print(paste0("The correlation cutoff was set to ", cutoff))

#WT1
print("Correlation to WT1 were calculated")

cor_WT1<-numeric()
for(i in 1:nrow(data)){
  cor_WT1[i]<-cor(as.numeric(WT1),as.numeric(data[i,]),method = "pearson")
}
cor_WT1<-data.frame(Pearson_correlation=as.numeric(cor_WT1))
rownames(cor_WT1)<-rownames(data)
ggplot(cor_WT1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "WT1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for WT1")

universe<-rownames(cor_WT1)
correlated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation < -1*cutoff]

WT1_correlated_enr_GO<-cbind(gene="WT1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
WT1_anticorrelated_enr_GO<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
WT1_correlated_enr_KEGG<-cbind(gene="WT1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
WT1_anticorrelated_enr_KEGG<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

WT1_enr<-rbind(WT1_correlated_enr_GO,WT1_anticorrelated_enr_GO,WT1_correlated_enr_KEGG,WT1_anticorrelated_enr_KEGG)

write.csv(cor_WT1,file = "03_output_files/Correlations_WT1_Subset20.csv")

#EHD3
print("Correlation to EHD3 were calculated")

cor_EHD3<-numeric()
for(i in 1:nrow(data)){
  cor_EHD3[i]<-cor(as.numeric(EHD3),as.numeric(data[i,]),method = "pearson")
}
cor_EHD3<-data.frame(Pearson_correlation=as.numeric(cor_EHD3))
rownames(cor_EHD3)<-rownames(data)
ggplot(cor_EHD3,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "EHD3 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for EHD3")

universe<-rownames(cor_EHD3)
correlated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation < -1*cutoff]

EHD3_correlated_enr_GO<-cbind(gene="EHD3",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
EHD3_anticorrelated_enr_GO<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
EHD3_correlated_enr_KEGG<-cbind(gene="EHD3",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
EHD3_anticorrelated_enr_KEGG<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

EHD3_enr<-rbind(EHD3_correlated_enr_GO,EHD3_anticorrelated_enr_GO,EHD3_correlated_enr_KEGG,EHD3_anticorrelated_enr_KEGG)

write.csv(cor_EHD3,file = "03_output_files/Correlations_EHD3_Subset20.csv")

#NPHS1
print("Correlation to NPHS1 were calculated")

cor_NPHS1<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS1[i]<-cor(as.numeric(NPHS1),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS1<-data.frame(Pearson_correlation=as.numeric(cor_NPHS1))
rownames(cor_NPHS1)<-rownames(data)
ggplot(cor_NPHS1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS1")

universe<-rownames(cor_NPHS1)
correlated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation < -1*cutoff]

NPHS1_correlated_enr_GO<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS1_anticorrelated_enr_GO<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS1_correlated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS1_anticorrelated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS1_enr<-rbind(NPHS1_correlated_enr_GO,NPHS1_anticorrelated_enr_GO,NPHS1_correlated_enr_KEGG,NPHS1_anticorrelated_enr_KEGG)

write.csv(cor_NPHS1,file = "03_output_files/Correlations_NPHS1_Subset20.csv")

#NPHS2
print("Correlation to NPHS2 were calculated")

cor_NPHS2<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS2[i]<-cor(as.numeric(NPHS2),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS2<-data.frame(Pearson_correlation=as.numeric(cor_NPHS2))
rownames(cor_NPHS2)<-rownames(data)
ggplot(cor_NPHS2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS2")

universe<-rownames(cor_NPHS2)
correlated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation < -1*cutoff]

NPHS2_correlated_enr_GO<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS2_anticorrelated_enr_GO<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS2_correlated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS2_anticorrelated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS2_enr<-rbind(NPHS2_correlated_enr_GO,NPHS2_anticorrelated_enr_GO,NPHS2_correlated_enr_KEGG,NPHS2_anticorrelated_enr_KEGG)

write.csv(cor_NPHS2,file = "03_output_files/Correlations_NPHS2_Subset20.csv")

#CD2AP
print("Correlation to CD2AP were calculated")

cor_CD2AP<-numeric()
for(i in 1:nrow(data)){
  cor_CD2AP[i]<-cor(as.numeric(CD2AP),as.numeric(data[i,]),method = "pearson")
}
cor_CD2AP<-data.frame(Pearson_correlation=as.numeric(cor_CD2AP))
rownames(cor_CD2AP)<-rownames(data)
ggplot(cor_CD2AP,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "CD2AP correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for CD2AP")

universe<-rownames(cor_CD2AP)
correlated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation < -1*cutoff]

CD2AP_correlated_enr_GO<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
CD2AP_anticorrelated_enr_GO<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
CD2AP_correlated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
CD2AP_anticorrelated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

CD2AP_enr<-rbind(CD2AP_correlated_enr_GO,CD2AP_anticorrelated_enr_GO,CD2AP_correlated_enr_KEGG,CD2AP_anticorrelated_enr_KEGG)

write.csv(cor_CD2AP,file = "03_output_files/Correlations_CD2AP_Subset20.csv")

#PDGFRB
print("Correlation to PDGFRB were calculated")

cor_PDGFRB<-numeric()
for(i in 1:nrow(data)){
  cor_PDGFRB[i]<-cor(as.numeric(PDGFRB),as.numeric(data[i,]),method = "pearson")
}
cor_PDGFRB<-data.frame(Pearson_correlation=as.numeric(cor_PDGFRB))
rownames(cor_PDGFRB)<-rownames(data)
ggplot(cor_PDGFRB,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PDGFRB correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PDGFRB")

universe<-rownames(cor_PDGFRB)
correlated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation < -1*cutoff]

PDGFRB_correlated_enr_GO<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_GO<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PDGFRB_correlated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PDGFRB_enr<-rbind(PDGFRB_correlated_enr_GO,PDGFRB_anticorrelated_enr_GO,PDGFRB_correlated_enr_KEGG,PDGFRB_anticorrelated_enr_KEGG)

write.csv(cor_PDGFRB,file = "03_output_files/Correlations_PDGFRB_Subset20.csv")


#PECAM1
print("Correlation to PECAM1 were calculated")

cor_PECAM1<-numeric()
for(i in 1:nrow(data)){
  cor_PECAM1[i]<-cor(as.numeric(PECAM1),as.numeric(data[i,]),method = "pearson")
}
cor_PECAM1<-data.frame(Pearson_correlation=as.numeric(cor_PECAM1))
rownames(cor_PECAM1)<-rownames(data)
ggplot(cor_PECAM1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PECAM1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PECAM1")

universe<-rownames(cor_PECAM1)
correlated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation < -1*cutoff]

PECAM1_correlated_enr_GO<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PECAM1_anticorrelated_enr_GO<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PECAM1_correlated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PECAM1_anticorrelated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PECAM1_enr<-rbind(PECAM1_correlated_enr_GO,PECAM1_anticorrelated_enr_GO,PECAM1_correlated_enr_KEGG,PECAM1_anticorrelated_enr_KEGG)

write.csv(cor_PECAM1,file = "03_output_files/Correlations_PECAM1_Subset20.csv")

#VWF
print("Correlation to VWF were calculated")

cor_VWF<-numeric()
for(i in 1:nrow(data)){
  cor_VWF[i]<-cor(as.numeric(VWF),as.numeric(data[i,]),method = "pearson")
}
cor_VWF<-data.frame(Pearson_correlation=as.numeric(cor_VWF))
rownames(cor_VWF)<-rownames(data)
ggplot(cor_VWF,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "VWF correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for VWF")

universe<-rownames(cor_VWF)
correlated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation < -1*cutoff]

VWF_correlated_enr_GO<-cbind(gene="VWF",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
VWF_anticorrelated_enr_GO<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
VWF_correlated_enr_KEGG<-cbind(gene="VWF",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
VWF_anticorrelated_enr_KEGG<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

VWF_enr<-rbind(VWF_correlated_enr_GO,VWF_anticorrelated_enr_GO,VWF_correlated_enr_KEGG,VWF_anticorrelated_enr_KEGG)

write.csv(cor_VWF,file = "03_output_files/Correlations_VWF_Subset20.csv")

#ACTA2
print("Correlation to ACTA2 were calculated")

cor_ACTA2<-numeric()
for(i in 1:nrow(data)){
  cor_ACTA2[i]<-cor(as.numeric(ACTA2),as.numeric(data[i,]),method = "pearson")
}
cor_ACTA2<-data.frame(Pearson_correlation=as.numeric(cor_ACTA2))
rownames(cor_ACTA2)<-rownames(data)
ggplot(cor_ACTA2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "ACTA2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for ACTA2")

universe<-rownames(cor_ACTA2)
correlated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation < -1*cutoff]

ACTA2_correlated_enr_GO<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
ACTA2_anticorrelated_enr_GO<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
ACTA2_correlated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
ACTA2_anticorrelated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

ACTA2_enr<-rbind(ACTA2_correlated_enr_GO,ACTA2_anticorrelated_enr_GO,ACTA2_correlated_enr_KEGG,ACTA2_anticorrelated_enr_KEGG)

write.csv(cor_ACTA2,file = "03_output_files/Correlations_ACTA2_Subset20.csv")

gene_correlation_enrichments<-rbind(WT1_enr, EHD3_enr,NPHS1_enr,NPHS2_enr,CD2AP_enr,PDGFRB_enr,PECAM1_enr, VWF_enr,ACTA2_enr)
gene_correlation_enrichments<-gene_correlation_enrichments[gene_correlation_enrichments$pval<0.05&gene_correlation_enrichments$fold_change>1,]

write.table(gene_correlation_enrichments,file = "03_output_files/gene_correlation_enrichments_Subset20.txt",sep = "\t",row.names = F)
```


## Subset22
The Subset22 was plotted for each of the proteins and correlations were calculated, correlations with a cutoff of 0.5 were use to perform enrichment of correlated and anticorrelated genes
```{r plot_Subset22}
data<-Subset22$data

WT1<-data[17410,]
EHD3<-data[4622,]
NPHS1<-data[10252,]
NPHS2<-data[10253,]
CD2AP<-data[2529,]
PDGFRB<-data[11312,]
PECAM1<-data[11364,]
VWF<-data[17259,]
ACTA2<-data[198,]

table_genes=data.frame(WT1=t(WT1),EHD3=t(EHD3),NPHS1=t(NPHS1),NPHS2=t(NPHS2),CD2AP=t(CD2AP),PDGFRB=t(PDGFRB), PECAM1=t(PECAM1),VWF=t(VWF),ACTA2=t(ACTA2),condition=t(Subset22$metadata[5,]))

colnames(table_genes)= c("WT1","EHD3","NPHS1","NPHS2","CD2AP","PDGFRB","PECAM1","VWF","ACTA2","Condition")

ggplot(table_genes,aes(x=WT1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("WT1")
ggplot(table_genes,aes(x=EHD3,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("EHD3")
ggplot(table_genes,aes(x=NPHS1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS1")
ggplot(table_genes,aes(x=NPHS2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("NPHS2")
ggplot(table_genes,aes(x=CD2AP,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("CD2AP")
ggplot(table_genes,aes(x=PDGFRB,y=Condition,fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PDGFRB")
ggplot(table_genes,aes(x=PECAM1,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("PECAM")
ggplot(table_genes,aes(x=VWF,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("VWF")
ggplot(table_genes,aes(x=ACTA2,y=Condition, fill=Condition))+geom_boxplot()+geom_jitter()+ggtitle("ACTA2")

#the correlation cutoff is set
cutoff<-0.5

print(paste0("The correlation cutoff was set to ", cutoff))

#WT1
print("Correlation to WT1 were calculated")

cor_WT1<-numeric()
for(i in 1:nrow(data)){
  cor_WT1[i]<-cor(as.numeric(WT1),as.numeric(data[i,]),method = "pearson")
}
cor_WT1<-data.frame(Pearson_correlation=as.numeric(cor_WT1))
rownames(cor_WT1)<-rownames(data)
ggplot(cor_WT1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "WT1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for WT1")

universe<-rownames(cor_WT1)
correlated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_WT1)[cor_WT1$Pearson_correlation < -1*cutoff]

WT1_correlated_enr_GO<-cbind(gene="WT1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
WT1_anticorrelated_enr_GO<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
WT1_correlated_enr_KEGG<-cbind(gene="WT1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
WT1_anticorrelated_enr_KEGG<-cbind(gene="WT1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

WT1_enr<-rbind(WT1_correlated_enr_GO,WT1_anticorrelated_enr_GO,WT1_correlated_enr_KEGG,WT1_anticorrelated_enr_KEGG)

write.csv(cor_WT1,file = "03_output_files/Correlations_WT1_Subset22.csv")

#EHD3
print("Correlation to EHD3 were calculated")

cor_EHD3<-numeric()
for(i in 1:nrow(data)){
  cor_EHD3[i]<-cor(as.numeric(EHD3),as.numeric(data[i,]),method = "pearson")
}
cor_EHD3<-data.frame(Pearson_correlation=as.numeric(cor_EHD3))
rownames(cor_EHD3)<-rownames(data)
ggplot(cor_EHD3,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "EHD3 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for EHD3")

universe<-rownames(cor_EHD3)
correlated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_EHD3)[cor_EHD3$Pearson_correlation < -1*cutoff]

EHD3_correlated_enr_GO<-cbind(gene="EHD3",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
EHD3_anticorrelated_enr_GO<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
EHD3_correlated_enr_KEGG<-cbind(gene="EHD3",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
EHD3_anticorrelated_enr_KEGG<-cbind(gene="EHD3",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

EHD3_enr<-rbind(EHD3_correlated_enr_GO,EHD3_anticorrelated_enr_GO,EHD3_correlated_enr_KEGG,EHD3_anticorrelated_enr_KEGG)

write.csv(cor_EHD3,file = "03_output_files/Correlations_EHD3_Subset22.csv")

#NPHS1
print("Correlation to NPHS1 were calculated")

cor_NPHS1<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS1[i]<-cor(as.numeric(NPHS1),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS1<-data.frame(Pearson_correlation=as.numeric(cor_NPHS1))
rownames(cor_NPHS1)<-rownames(data)
ggplot(cor_NPHS1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS1")

universe<-rownames(cor_NPHS1)
correlated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS1)[cor_NPHS1$Pearson_correlation < -1*cutoff]

NPHS1_correlated_enr_GO<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS1_anticorrelated_enr_GO<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS1_correlated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS1_anticorrelated_enr_KEGG<-cbind(gene="NPHS1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS1_enr<-rbind(NPHS1_correlated_enr_GO,NPHS1_anticorrelated_enr_GO,NPHS1_correlated_enr_KEGG,NPHS1_anticorrelated_enr_KEGG)

write.csv(cor_NPHS1,file = "03_output_files/Correlations_NPHS1_Subset22.csv")

#NPHS2
print("Correlation to NPHS2 were calculated")

cor_NPHS2<-numeric()
for(i in 1:nrow(data)){
  cor_NPHS2[i]<-cor(as.numeric(NPHS2),as.numeric(data[i,]),method = "pearson")
}
cor_NPHS2<-data.frame(Pearson_correlation=as.numeric(cor_NPHS2))
rownames(cor_NPHS2)<-rownames(data)
ggplot(cor_NPHS2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "NPHS2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for NPHS2")

universe<-rownames(cor_NPHS2)
correlated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_NPHS2)[cor_NPHS2$Pearson_correlation < -1*cutoff]

NPHS2_correlated_enr_GO<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
NPHS2_anticorrelated_enr_GO<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
NPHS2_correlated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
NPHS2_anticorrelated_enr_KEGG<-cbind(gene="NPHS2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

NPHS2_enr<-rbind(NPHS2_correlated_enr_GO,NPHS2_anticorrelated_enr_GO,NPHS2_correlated_enr_KEGG,NPHS2_anticorrelated_enr_KEGG)

write.csv(cor_NPHS2,file = "03_output_files/Correlations_NPHS2_Subset22.csv")

#CD2AP
print("Correlation to CD2AP were calculated")

cor_CD2AP<-numeric()
for(i in 1:nrow(data)){
  cor_CD2AP[i]<-cor(as.numeric(CD2AP),as.numeric(data[i,]),method = "pearson")
}
cor_CD2AP<-data.frame(Pearson_correlation=as.numeric(cor_CD2AP))
rownames(cor_CD2AP)<-rownames(data)
ggplot(cor_CD2AP,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "CD2AP correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for CD2AP")

universe<-rownames(cor_CD2AP)
correlated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_CD2AP)[cor_CD2AP$Pearson_correlation < -1*cutoff]

CD2AP_correlated_enr_GO<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
CD2AP_anticorrelated_enr_GO<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
CD2AP_correlated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
CD2AP_anticorrelated_enr_KEGG<-cbind(gene="CD2AP",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

CD2AP_enr<-rbind(CD2AP_correlated_enr_GO,CD2AP_anticorrelated_enr_GO,CD2AP_correlated_enr_KEGG,CD2AP_anticorrelated_enr_KEGG)

write.csv(cor_CD2AP,file = "03_output_files/Correlations_CD2AP_Subset22.csv")

#PDGFRB
print("Correlation to PDGFRB were calculated")

cor_PDGFRB<-numeric()
for(i in 1:nrow(data)){
  cor_PDGFRB[i]<-cor(as.numeric(PDGFRB),as.numeric(data[i,]),method = "pearson")
}
cor_PDGFRB<-data.frame(Pearson_correlation=as.numeric(cor_PDGFRB))
rownames(cor_PDGFRB)<-rownames(data)
ggplot(cor_PDGFRB,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PDGFRB correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PDGFRB")

universe<-rownames(cor_PDGFRB)
correlated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PDGFRB)[cor_PDGFRB$Pearson_correlation < -1*cutoff]

PDGFRB_correlated_enr_GO<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_GO<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PDGFRB_correlated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PDGFRB_anticorrelated_enr_KEGG<-cbind(gene="PDGFRB",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PDGFRB_enr<-rbind(PDGFRB_correlated_enr_GO,PDGFRB_anticorrelated_enr_GO,PDGFRB_correlated_enr_KEGG,PDGFRB_anticorrelated_enr_KEGG)

write.csv(cor_PDGFRB,file = "03_output_files/Correlations_PDGFRB_Subset22.csv")


#PECAM1
print("Correlation to PECAM1 were calculated")

cor_PECAM1<-numeric()
for(i in 1:nrow(data)){
  cor_PECAM1[i]<-cor(as.numeric(PECAM1),as.numeric(data[i,]),method = "pearson")
}
cor_PECAM1<-data.frame(Pearson_correlation=as.numeric(cor_PECAM1))
rownames(cor_PECAM1)<-rownames(data)
ggplot(cor_PECAM1,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "PECAM1 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for PECAM1")

universe<-rownames(cor_PECAM1)
correlated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_PECAM1)[cor_PECAM1$Pearson_correlation < -1*cutoff]

PECAM1_correlated_enr_GO<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
PECAM1_anticorrelated_enr_GO<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
PECAM1_correlated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
PECAM1_anticorrelated_enr_KEGG<-cbind(gene="PECAM1",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

PECAM1_enr<-rbind(PECAM1_correlated_enr_GO,PECAM1_anticorrelated_enr_GO,PECAM1_correlated_enr_KEGG,PECAM1_anticorrelated_enr_KEGG)

write.csv(cor_PECAM1,file = "03_output_files/Correlations_PECAM1_Subset22.csv")

#VWF
print("Correlation to VWF were calculated")

cor_VWF<-numeric()
for(i in 1:nrow(data)){
  cor_VWF[i]<-cor(as.numeric(VWF),as.numeric(data[i,]),method = "pearson")
}
cor_VWF<-data.frame(Pearson_correlation=as.numeric(cor_VWF))
rownames(cor_VWF)<-rownames(data)
ggplot(cor_VWF,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "VWF correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for VWF")

universe<-rownames(cor_VWF)
correlated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_VWF)[cor_VWF$Pearson_correlation < -1*cutoff]

VWF_correlated_enr_GO<-cbind(gene="VWF",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
VWF_anticorrelated_enr_GO<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
VWF_correlated_enr_KEGG<-cbind(gene="VWF",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
VWF_anticorrelated_enr_KEGG<-cbind(gene="VWF",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

VWF_enr<-rbind(VWF_correlated_enr_GO,VWF_anticorrelated_enr_GO,VWF_correlated_enr_KEGG,VWF_anticorrelated_enr_KEGG)

write.csv(cor_VWF,file = "03_output_files/Correlations_VWF_Subset22.csv")

#ACTA2
print("Correlation to ACTA2 were calculated")

cor_ACTA2<-numeric()
for(i in 1:nrow(data)){
  cor_ACTA2[i]<-cor(as.numeric(ACTA2),as.numeric(data[i,]),method = "pearson")
}
cor_ACTA2<-data.frame(Pearson_correlation=as.numeric(cor_ACTA2))
rownames(cor_ACTA2)<-rownames(data)
ggplot(cor_ACTA2,aes(x=Pearson_correlation))+geom_histogram(bins = 100,fill="gray10",alpha=0.9)+ggtitle(label = "ACTA2 correlation frequency plot")+theme_ROP()+geom_vline(xintercept = cutoff, linetype="dotted",color="red")+ geom_vline(xintercept = -cutoff, linetype="dotted",color="blue")

print("Enrichment analysis were then performed for ACTA2")

universe<-rownames(cor_ACTA2)
correlated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation>cutoff]
anticorrelated<- rownames(cor_ACTA2)[cor_ACTA2$Pearson_correlation < -1*cutoff]

ACTA2_correlated_enr_GO<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_GO_EASE(correlated,universe))
ACTA2_anticorrelated_enr_GO<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_GO_EASE(anticorrelated,universe))
ACTA2_correlated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="correlated",UniProt_KEGG_EASE(correlated,universe))
ACTA2_anticorrelated_enr_KEGG<-cbind(gene="ACTA2",enriched_in="anticorrelated",UniProt_KEGG_EASE(anticorrelated,universe))

ACTA2_enr<-rbind(ACTA2_correlated_enr_GO,ACTA2_anticorrelated_enr_GO,ACTA2_correlated_enr_KEGG,ACTA2_anticorrelated_enr_KEGG)

write.csv(cor_ACTA2,file = "03_output_files/Correlations_ACTA2_Subset22.csv")

gene_correlation_enrichments<-rbind(WT1_enr, EHD3_enr,NPHS1_enr,NPHS2_enr,CD2AP_enr,PDGFRB_enr,PECAM1_enr, VWF_enr,ACTA2_enr)
gene_correlation_enrichments<-gene_correlation_enrichments[gene_correlation_enrichments$pval<0.05&gene_correlation_enrichments$fold_change>1,]

write.table(gene_correlation_enrichments,file = "03_output_files/gene_correlation_enrichments_Subset22.txt",sep = "\t",row.names = F)
```



# Venn Diagrams

The Venn diagram for the young donors was plotted and the GO and KEGG enrichments were performed and exported
```{r Venn_AS_young}
print("First the lists of transcripts modulated in each group were evaluated.")
AS_Young<- list(AS_M_up = rownames(Subset06$statistics)[Subset06$statistics$`Alport(M)_Glomeruli_vs_Normal (Youth)_Glomeruli_Ttest_padj`<0.05&Subset06$statistics$`log(Alport(M)_Glomeruli/Normal (Youth)_Glomeruli)`>0],
AS_F_up = rownames(Subset06$statistics)[Subset06$statistics$`Alport(F)(Young)_Glomeruli_vs_Normal (Youth)_Glomeruli_Ttest_padj`<0.05&Subset06$statistics$`log(Alport(F)(Young)_Glomeruli/Normal (Youth)_Glomeruli)`>0],
AS_M_down = rownames(Subset06$statistics)[Subset06$statistics$`Alport(M)_Glomeruli_vs_Normal (Youth)_Glomeruli_Ttest_padj`<0.05&Subset06$statistics$`log(Alport(M)_Glomeruli/Normal (Youth)_Glomeruli)`<0],
AS_F_down = rownames(Subset06$statistics)[Subset06$statistics$`Alport(F)(Young)_Glomeruli_vs_Normal (Youth)_Glomeruli_Ttest_padj`<0.05&Subset06$statistics$`log(Alport(F)(Young)_Glomeruli/Normal (Youth)_Glomeruli)`<0]
)

universe<-rownames(Subset06$statistics)

print("The Venn diagrams were plotted using ggven.")

library("ggvenn")
ggvenn(
  AS_Young, 
  fill_color = c("#0073C2FF", "#EFC000FF", "#868686FF", "#CD534CFF"),
  stroke_size = 0.5, set_name_size = 4
  )

print("For all the groups with more than 0 counts enrichment analysis were performed")

AS_M_UP_only<- setdiff(AS_Young$AS_M_up,c(AS_Young$AS_F_up,AS_Young$AS_M_down,AS_Young$AS_F_down))
AS_F_UP_only<- setdiff(AS_Young$AS_F_up,c(AS_Young$AS_M_up,AS_Young$AS_M_down,AS_Young$AS_F_down))
AS_M_DOWN_only<- setdiff(AS_Young$AS_M_down,c(AS_Young$AS_F_up,AS_Young$AS_F_down,AS_Young$AS_M_up))
AS_F_DOWN_only<- setdiff(AS_Young$AS_F_down,c(AS_Young$AS_M_up,AS_Young$AS_M_down,AS_Young$AS_F_up))

AS_M_and_F_up<- setdiff(intersect(AS_Young$AS_M_up,AS_Young$AS_F_up),c(AS_Young$AS_M_down,AS_Young$AS_F_down))
AS_M_and_F_down<- setdiff(intersect(AS_Young$AS_M_down,AS_Young$AS_F_down),c(AS_Young$AS_M_up,AS_Young$AS_F_up))

AS_M_up_F_down<-setdiff(intersect(AS_Young$AS_M_up,AS_Young$AS_F_down),c(AS_Young$AS_M_down,AS_Young$AS_F_up))

enrichments_Venn<- rbind(cbind(group="AS_M_UP_only",UniProt_GO_EASE(AS_M_UP_only,universe)),
                    cbind(group="AS_M_UP_only",UniProt_KEGG_EASE(AS_M_UP_only,universe)),
                    cbind(group="AS_F_UP_only",UniProt_GO_EASE(AS_F_UP_only,universe)),
                    cbind(group="AS_F_UP_only",UniProt_KEGG_EASE(AS_F_UP_only,universe)),
                    cbind(group="AS_M_DOWN_only",UniProt_GO_EASE(AS_M_DOWN_only,universe)),
                    cbind(group="AS_M_DOWN_only",UniProt_KEGG_EASE(AS_M_DOWN_only,universe)),
                    cbind(group="AS_F_DOWN_only",UniProt_GO_EASE(AS_F_DOWN_only,universe)),
                    cbind(group="AS_F_DOWN_only",UniProt_KEGG_EASE(AS_F_DOWN_only,universe)),
                    cbind(group="AS_M_and_F_up",UniProt_GO_EASE(AS_M_and_F_up,universe)),
                    cbind(group="AS_M_and_F_up",UniProt_KEGG_EASE(AS_M_and_F_up,universe)),
                    cbind(group="AS_M_and_F_down",UniProt_GO_EASE(AS_M_and_F_down,universe)),
                    cbind(group="AS_M_and_F_down",UniProt_KEGG_EASE(AS_M_and_F_down,universe)),
                    cbind(group="AS_M_up_F_down",UniProt_GO_EASE(AS_M_up_F_down,universe))#,
                    #cbind(group="AS_M_up_F_down",UniProt_KEGG_EASE(AS_M_up_F_down,universe))
                    )

enrichments_Venn<- enrichments_Venn[enrichments_Venn$pval<0.05&enrichments_Venn$fold_change>1,]

print("GO and KEGG terms with p<0.05 were exported")

write.table(enrichments_Venn,file = "04_Venn_diagram_enrichments/enr_VENN_AS_Young.txt",sep="\t",row.names = F)
write.csv(t(AS_Young),file="04_Venn_diagram_enrichments/AS_Young_significant.csv")

```

The Venn diagram for the FSGS donors was plotted and the GO and KEGG enrichment were performed and exported
```{r Venn_FSGS}
print("First the lists of transcripts modulated in each group were evaluated.")
FSGS_vs_control<- list(FSGS_M_up = rownames(Subset14_3c$statistics)[Subset14_3c$statistics$`FSGS (M)_Glomeruli_vs_Normal (Adult)_Glomeruli_Ttest_padj`<0.05&Subset14_3c$statistics$`log(FSGS (M)_Glomeruli/Normal (Adult)_Glomeruli)`>0],
                       FSGS_F_up = rownames(Subset14_3c$statistics)[Subset14_3c$statistics$`FSGS(F)_Glomeruli_vs_Normal (Adult)_Glomeruli_Ttest_padj`<0.05&Subset14_3c$statistics$`log(FSGS(F)_Glomeruli/Normal (Adult)_Glomeruli)`>0],
                       FSGS_M_down = rownames(Subset14_3c$statistics)[Subset14_3c$statistics$`FSGS (M)_Glomeruli_vs_Normal (Adult)_Glomeruli_Ttest_padj`<0.05&Subset14_3c$statistics$`log(FSGS (M)_Glomeruli/Normal (Adult)_Glomeruli)`<0],
                       FSGS_F_down = rownames(Subset14_3c$statistics)[Subset14_3c$statistics$`FSGS(F)_Glomeruli_vs_Normal (Adult)_Glomeruli_Ttest_padj`<0.05&Subset14_3c$statistics$`log(FSGS(F)_Glomeruli/Normal (Adult)_Glomeruli)`<0])


universe<-rownames(Subset14_3c$statistics)

print("The Venn diagrams were plotted using ggven.")

ggvenn(
 FSGS_vs_control, 
  fill_color = c("#0073C2FF", "#EFC000FF", "#868686FF", "#CD534CFF"),
  stroke_size = 0.5, set_name_size = 4
  )

print("For all the groups with more than 0 counts enrichment analysis were performed")

FSGS_M_UP_only<- setdiff(FSGS_vs_control$FSGS_M_up,c(FSGS_vs_control$FSGS_F_up,FSGS_vs_control$FSGS_M_down,FSGS_vs_control$FSGS_F_down))
FSGS_F_UP_only<- setdiff(FSGS_vs_control$FSGS_F_up,c(FSGS_vs_control$FSGS_M_up,FSGS_vs_control$FSGS_M_down,FSGS_vs_control$FSGS_F_down))
FSGS_M_DOWN_only<- setdiff(FSGS_vs_control$FSGS_M_down,c(FSGS_vs_control$FSGS_F_up,FSGS_vs_control$FSGS_F_down,FSGS_vs_control$FSGS_M_up))
FSGS_F_DOWN_only<- setdiff(FSGS_vs_control$FSGS_F_down,c(FSGS_vs_control$FSGS_M_up,FSGS_vs_control$FSGS_M_down,FSGS_vs_control$FSGS_F_up))

FSGS_M_and_F_up<- setdiff(intersect(FSGS_vs_control$FSGS_M_up,FSGS_vs_control$FSGS_F_up),c(FSGS_vs_control$FSGS_M_down,FSGS_vs_control$FSGS_F_down))
FSGS_M_and_F_down<- setdiff(intersect(FSGS_vs_control$FSGS_M_down,FSGS_vs_control$FSGS_F_down),c(FSGS_vs_control$FSGS_M_up,FSGS_vs_control$FSGS_F_up))
  
FSGS_M_down_and_F_up<- setdiff(intersect(FSGS_vs_control$FSGS_M_down,FSGS_vs_control$FSGS_F_up),c(FSGS_vs_control$FSGS_M_up,FSGS_vs_control$FSGS_F_down))

enrichments_Venn<- rbind(cbind(group="FSGS_M_UP_only",UniProt_GO_EASE(FSGS_M_UP_only,universe)),
                    cbind(group="FSGS_M_UP_only",UniProt_KEGG_EASE(FSGS_M_UP_only,universe)),
                    cbind(group="FSGS_F_UP_only",UniProt_GO_EASE(FSGS_F_UP_only,universe)),
                    cbind(group="FSGS_F_UP_only",UniProt_KEGG_EASE(FSGS_F_UP_only,universe)),
                    cbind(group="FSGS_M_DOWN_only",UniProt_GO_EASE(FSGS_M_DOWN_only,universe)),
                    cbind(group="FSGS_M_DOWN_only",UniProt_KEGG_EASE(FSGS_M_DOWN_only,universe)),
                    cbind(group="FSGS_F_DOWN_only",UniProt_GO_EASE(FSGS_F_DOWN_only,universe)),
                    cbind(group="FSGS_F_DOWN_only",UniProt_KEGG_EASE(FSGS_F_DOWN_only,universe)),
                    cbind(group="FSGS_M_and_F_up",UniProt_GO_EASE(FSGS_M_and_F_up,universe)),
                    cbind(group="FSGS_M_and_F_up",UniProt_KEGG_EASE(FSGS_M_and_F_up,universe)),
                    cbind(group="FSGS_M_and_F_down",UniProt_GO_EASE(FSGS_M_and_F_down,universe)),
                    cbind(group="FSGS_M_and_F_down",UniProt_KEGG_EASE(FSGS_M_and_F_down,universe)),
                    cbind(group="FSGS_M_down_and_F_up",UniProt_GO_EASE(FSGS_M_down_and_F_up,universe))#,
                    #cbind(group="FSGS_M_down_and_F_up",UniProt_KEGG_EASE(FSGS_M_down_and_F_up,universe))
                    )

enrichments_Venn_FSGS<- enrichments_Venn[enrichments_Venn$pval<0.05&enrichments_Venn$fold_change>1,]

print("GO and KEGG terms with p<0.05 were exported")
write.table(enrichments_Venn_FSGS,file = "03_output_files/enr_VENN_FSGS.txt",sep="\t",row.names = F)
```

The Venn diagram for the MN donors was plotted and the GO and KEGG enrichment were performed and exported
```{r Venn_MN}
print("First the lists of transcripts modulated in each group were evaluated.")
MN_vs_control<- list(MN_M_up = rownames(Subset20$statistics)[Subset20$statistics$`Normal (Adult)_Glomeruli_vs_MN(M)_Glomeruli_Ttest_padj`<0.05&Subset20$statistics$`log(Normal (Adult)_Glomeruli/MN(M)_Glomeruli)`<0],
                     MN_F_up = rownames(Subset20$statistics)[Subset20$statistics$`Normal (Adult)_Glomeruli_vs_MN(F)_Glomeruli_Ttest_padj`<0.05&Subset20$statistics$`log(Normal (Adult)_Glomeruli/MN(F)_Glomeruli)`<0],
                     MN_M_down = rownames(Subset20$statistics)[Subset20$statistics$`Normal (Adult)_Glomeruli_vs_MN(M)_Glomeruli_Ttest_padj`<0.05&Subset20$statistics$`log(Normal (Adult)_Glomeruli/MN(M)_Glomeruli)`>0], 
                     MN_F_down = rownames(Subset20$statistics)[Subset20$statistics$`Normal (Adult)_Glomeruli_vs_MN(F)_Glomeruli_Ttest_padj`<0.05&Subset20$statistics$`log(Normal (Adult)_Glomeruli/MN(F)_Glomeruli)`>0])

universe<-rownames(Subset20$statistics)

print("The Venn diagrams were plotted using ggven.")

ggvenn(
 MN_vs_control, 
  fill_color = c("#0073C2FF", "#EFC000FF", "#868686FF", "#CD534CFF"),
  stroke_size = 0.5, set_name_size = 4
  )

print("For all the groups with more than 0 counts enrichment analysis were performed")

MN_M_UP_only<- setdiff(MN_vs_control$MN_M_up,c(MN_vs_control$MN_F_up,MN_vs_control$MN_M_down,MN_vs_control$MN_F_down))
MN_F_UP_only<- setdiff(MN_vs_control$MN_F_up,c(MN_vs_control$MN_M_up,MN_vs_control$MN_M_down,MN_vs_control$MN_F_down))
MN_M_DOWN_only<- setdiff(MN_vs_control$MN_M_down,c(MN_vs_control$MN_F_up,MN_vs_control$MN_F_down,MN_vs_control$MN_M_up))
MN_F_DOWN_only<- setdiff(MN_vs_control$MN_F_down,c(MN_vs_control$MN_M_up,MN_vs_control$MN_M_down,MN_vs_control$MN_F_up))

MN_M_and_F_up<- setdiff(intersect(MN_vs_control$MN_M_up,MN_vs_control$MN_F_up),c(MN_vs_control$MN_M_down,MN_vs_control$MN_F_down))
MN_M_and_F_down<- setdiff(intersect(MN_vs_control$MN_M_down,MN_vs_control$MN_F_down),c(MN_vs_control$MN_M_up,MN_vs_control$MN_F_up))
  
MN_M_up_and_F_down<- setdiff(intersect(MN_vs_control$MN_M_up,MN_vs_control$MN_F_down),c(MN_vs_control$MN_M_down,MN_vs_control$MN_F_up))

enrichments_Venn<- rbind(cbind(group="MN_M_UP_only",UniProt_GO_EASE(MN_M_UP_only,universe)),
                    cbind(group="MN_M_UP_only",UniProt_KEGG_EASE(MN_M_UP_only,universe)),
                    cbind(group="MN_F_UP_only",UniProt_GO_EASE(MN_F_UP_only,universe)),
                    cbind(group="MN_F_UP_only",UniProt_KEGG_EASE(MN_F_UP_only,universe)),
                    cbind(group="MN_M_DOWN_only",UniProt_GO_EASE(MN_M_DOWN_only,universe)),
                    cbind(group="MN_M_DOWN_only",UniProt_KEGG_EASE(MN_M_DOWN_only,universe)),
                    cbind(group="MN_F_DOWN_only",UniProt_GO_EASE(MN_F_DOWN_only,universe)),
                    cbind(group="MN_F_DOWN_only",UniProt_KEGG_EASE(MN_F_DOWN_only,universe)),
                    cbind(group="MN_M_and_F_up",UniProt_GO_EASE(MN_M_and_F_up,universe)),
                    cbind(group="MN_M_and_F_up",UniProt_KEGG_EASE(MN_M_and_F_up,universe)),
                    cbind(group="MN_M_and_F_down",UniProt_GO_EASE(MN_M_and_F_down,universe)),
                    cbind(group="MN_M_and_F_down",UniProt_KEGG_EASE(MN_M_and_F_down,universe)),
                    cbind(group="MN_M_up_and_F_down",UniProt_GO_EASE(MN_M_up_and_F_down,universe)),
                    cbind(group="MN_M_up_and_F_down",UniProt_KEGG_EASE(MN_M_up_and_F_down,universe))
                    )

enrichments_Venn_MN<- enrichments_Venn[enrichments_Venn$pval<0.05&enrichments_Venn$fold_change>1,]

print("GO and KEGG terms with p<0.05 were exported")
write.table(enrichments_Venn_MN,file = "03_output_files/enr_VENN_MN.txt",sep="\t",row.names = F)
```

The Venn diagram for the MN donors was plotted and the GO and KEGG enrichment were performed and exported
```{r Venn_MN_AS_FSGS_modulated}
print("First the lists of transcripts modulated in each group were evaluated.")
AS_adult<-list(AS_F_up= rownames(Subset08$statistics)[Subset08$statistics$`Alport(F)(Adult)_Glomeruli_vs_Normal (Adult)_Glomeruli_Ttest_padj`<0.05&Subset08$statistics$`log(Alport(F)(Adult)_Glomeruli/Normal (Adult)_Glomeruli)`>0],
               AS_F_down= rownames(Subset08$statistics)[Subset08$statistics$`Alport(F)(Adult)_Glomeruli_vs_Normal (Adult)_Glomeruli_Ttest_padj`<0.05&Subset08$statistics$`log(Alport(F)(Adult)_Glomeruli/Normal (Adult)_Glomeruli)`<0])

AS_MN_FSGS<-c(AS_adult, MN_vs_control, FSGS_vs_control)
AS_MN_FSGS_F<-AS_MN_FSGS[names(AS_MN_FSGS) %in% c("AS_F_up","AS_F_down","MN_F_up","MN_F_down","FSGS_F_up","FSGS_F_down")]
AS_MN_FSGS_F_up<-AS_MN_FSGS[names(AS_MN_FSGS) %in% c("AS_F_up","MN_F_up","FSGS_F_up")]
AS_MN_FSGS_F_down<-AS_MN_FSGS[names(AS_MN_FSGS) %in% c("AS_F_down","MN_F_down","FSGS_F_down")]

universe<-rownames(Subset20$statistics)

print("The Venn diagrams were plotted using ggven.")

ggvenn(
AS_MN_FSGS_F_down, 
  fill_color = c("#0073C2FF", "#EFC000FF", "#868686FF", "#CD534CFF"),
  stroke_size = 0.5, set_name_size = 4
  )

ggvenn(
AS_MN_FSGS_F_up, 
  fill_color = c("#0073C2FF", "#EFC000FF", "#868686FF", "#CD534CFF"),
  stroke_size = 0.5, set_name_size = 4
  )

print("For all the groups with more than 0 counts enrichment analysis were performed")

AS_F_only_up <- setdiff(AS_MN_FSGS_F_up$AS_F_up,c(AS_MN_FSGS_F_up$MN_F_up,AS_MN_FSGS_F_up$FSGS_F_up))
MN_F_only_up <-setdiff(AS_MN_FSGS_F_up$MN_F_up,c(AS_MN_FSGS_F_up$AS_F_up,AS_MN_FSGS_F_up$FSGS_F_up))
FSGS_F_only_up <-setdiff(AS_MN_FSGS_F_up$FSGS_F_up,c(AS_MN_FSGS_F_up$AS_F_up,AS_MN_FSGS_F_up$MN_F_up))
AS_MN_up <-setdiff(intersect(AS_MN_FSGS_F_up$AS_F_up, AS_MN_FSGS_F_up$MN_F_up),AS_MN_FSGS_F_up$FSGS_F_up)
AS_FSGS_up <-setdiff(intersect(AS_MN_FSGS_F_up$AS_F_up, AS_MN_FSGS_F_up$FSGS_F_up),AS_MN_FSGS_F_up$MN_F_up)
MN_FSGS_up <-setdiff(intersect(AS_MN_FSGS_F_up$MN_F_up, AS_MN_FSGS_F_up$FSGS_F_up),AS_MN_FSGS_F_up$AS_F_up)
F_up_all_diseases <-intersect(AS_MN_FSGS_F_up$AS_F_up,intersect(AS_MN_FSGS_F_up$MN_F_up,AS_MN_FSGS_F_up$FSGS_F_up))


AS_F_only_down <- setdiff(AS_MN_FSGS_F_down$AS_F_down,c(AS_MN_FSGS_F_down$MN_F_down,AS_MN_FSGS_F_down$FSGS_F_down))
MN_F_only_down <-setdiff(AS_MN_FSGS_F_down$MN_F_down,c(AS_MN_FSGS_F_down$AS_F_down,AS_MN_FSGS_F_down$FSGS_F_down))
FSGS_F_only_down <-setdiff(AS_MN_FSGS_F_down$FSGS_F_down,c(AS_MN_FSGS_F_down$AS_F_down,AS_MN_FSGS_F_down$MN_F_down))
AS_MN_down <-setdiff(intersect(AS_MN_FSGS_F_down$AS_F_down, AS_MN_FSGS_F_down$MN_F_down),AS_MN_FSGS_F_down$FSGS_F_down)
AS_FSGS_down <-setdiff(intersect(AS_MN_FSGS_F_down$AS_F_down, AS_MN_FSGS_F_down$FSGS_F_down),AS_MN_FSGS_F_down$MN_F_down)
MN_FSGS_down <-setdiff(intersect(AS_MN_FSGS_F_down$MN_F_down, AS_MN_FSGS_F_down$FSGS_F_down),AS_MN_FSGS_F_down$AS_F_down)
F_down_all_diseases <-intersect(AS_MN_FSGS_F_down$AS_F_down,intersect(AS_MN_FSGS_F_down$MN_F_down,AS_MN_FSGS_F_down$FSGS_F_down))

enrichments_Venn_up_all_diseases<- rbind(cbind(group="AS_F_only_up",UniProt_GO_EASE(AS_F_only_up,universe)), 
                         cbind(group="AS_F_only_up",UniProt_KEGG_EASE(AS_F_only_up,universe)),
                         cbind(group="MN_F_only_up",UniProt_GO_EASE(MN_F_only_up,universe)), 
                         cbind(group="MN_F_only_up",UniProt_KEGG_EASE(MN_F_only_up,universe)),
                         cbind(group="FSGS_F_only_up",UniProt_GO_EASE(FSGS_F_only_up,universe)), 
                         cbind(group="FSGS_F_only_up",UniProt_KEGG_EASE(FSGS_F_only_up,universe)),
                         cbind(group="AS_MN_up",UniProt_GO_EASE(AS_MN_up,universe)), 
                         cbind(group="AS_MN_up",UniProt_KEGG_EASE(AS_MN_up,universe)),
                         cbind(group="AS_FSGS_up",UniProt_GO_EASE(AS_FSGS_up,universe)), 
                         cbind(group="AS_FSGS_up",UniProt_KEGG_EASE(AS_FSGS_up,universe)),
                         cbind(group="MN_FSGS_up",UniProt_GO_EASE(MN_FSGS_up,universe)), 
                         cbind(group="MN_FSGS_up",UniProt_KEGG_EASE(MN_FSGS_up,universe)),
                         cbind(group="F_up_all_diseases",UniProt_GO_EASE(F_up_all_diseases,universe)), 
                         cbind(group="F_up_all_diseases",UniProt_KEGG_EASE(F_up_all_diseases,universe))
                        )

enrichments_Venn_down_all_diseases<- rbind(cbind(group="AS_F_only_down",UniProt_GO_EASE(AS_F_only_down,universe)), 
                         cbind(group="AS_F_only_down",UniProt_KEGG_EASE(AS_F_only_down,universe)),
                         cbind(group="MN_F_only_down",UniProt_GO_EASE(MN_F_only_down,universe)), 
                         cbind(group="MN_F_only_down",UniProt_KEGG_EASE(MN_F_only_down,universe)),
                         cbind(group="FSGS_F_only_down",UniProt_GO_EASE(FSGS_F_only_down,universe)), 
                         cbind(group="FSGS_F_only_down",UniProt_KEGG_EASE(FSGS_F_only_down,universe)),
                         cbind(group="AS_MN_down",UniProt_GO_EASE(AS_MN_down,universe)), 
                         cbind(group="AS_MN_down",UniProt_KEGG_EASE(AS_MN_down,universe)),
                         cbind(group="AS_FSGS_down",UniProt_GO_EASE(AS_FSGS_down,universe)), 
                         cbind(group="AS_FSGS_down",UniProt_KEGG_EASE(AS_FSGS_down,universe)),
                         cbind(group="MN_FSGS_down",UniProt_GO_EASE(MN_FSGS_down,universe)), 
                         cbind(group="MN_FSGS_down",UniProt_KEGG_EASE(MN_FSGS_down,universe)),
                         cbind(group="F_down_all_diseases",UniProt_GO_EASE(F_down_all_diseases,universe)),
                         cbind(group="F_down_all_diseases",UniProt_KEGG_EASE(F_down_all_diseases,universe))
                        )

enrichments_Venn_up_all_diseases_F<- enrichments_Venn_up_all_diseases[enrichments_Venn_up_all_diseases$pval<0.05&enrichments_Venn_up_all_diseases$fold_change>1,]
enrichments_Venn_down_all_diseases_F<- enrichments_Venn_down_all_diseases[enrichments_Venn_down_all_diseases$pval<0.05&enrichments_Venn_down_all_diseases$fold_change>1,]

print("GO and KEGG terms with p<0.05 were exported")
write.table(enrichments_Venn_up_all_diseases,file = "03_output_files/enr_VENN_different_diseases_up.txt",sep="\t",row.names = F)
write.table(enrichments_Venn_up_all_diseases,file = "03_output_files/enr_VENN_different_diseases_down.txt",sep="\t",row.names = F)

```

